<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Integrate methods: PANDA, LIONESS, CONDOR, ALPACA, SAMBAR, and MONSTER into one workflow. â€¢ netZooR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/all.min.css" integrity="sha256-PbSmjxuVAzJ6FPvNYsrXygfGhNJYyZ2GktDbkMBqQZg=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/v4-shims.min.css" integrity="sha256-A6jcAdwFD48VMjlI3GDxUd+eCQa7/KWy6G9oe/ovaPA=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Integrate methods: PANDA, LIONESS, CONDOR, ALPACA, SAMBAR, and MONSTER into one workflow.">
<meta property="og:description" content="PANDA(Passing Attributes between Networks for Data Assimilation) is a message-passing model to gene regulatory network reconstruction. It integrates multiple sources of biological data, including protein-protein interaction, gene expression, and sequence motif information, in order to reconstruct genome-wide, condition-specific regulatory networks.[(Glass et al. 2013)]. LIONESS(Linear Interpolation to Obtain Network Estimates for Single Samples) is a method to estimate sample-specific regulatory networks by applying linear interpolation to the predictions made by existing aggregate network inference approaches. CONDOR(COmplex Network Description Of Regulators)is a bipartite community structure analysis tool of biological networks, especially eQTL networks, including a method for scoring nodes based on their modularity contribution.[(Platig et al. 2016). ALPACA(ALtered Partitions Across Community Architectures) is a method for comparing two genome-scale networks derived from different phenotypic states to identify condition-specific modules.[(Padi and Quackenbush 2018)]. This package integrates pypanda--the Python implementation of PANDA and LIONESS(https://github.com/davidvi/pypanda),the R implementation of CONDOR(https://github.com/jplatig/condor) and the R implementation of ALPACA (https://github.com/meghapadi/ALPACA) into one workflow. Each tool can be call in this package by one function, and the relevant output could be accessible in current R session for downstream analysis. ">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.6.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/ALPACA_MaghaP.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="articles/ApplicationinGTExData_DeborahW.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="articles/ApplicationwithTBdataset_TianW.html">Application with TB dataset</a>
    </li>
    <li>
      <a href="articles/MONSTER_DanielS.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="articles/SAMBAR_MariekeK.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="articles/lioness_limma_CamilaLR.html">Comparing LIONESS Regulatory Networks using limma</a>
    </li>
    <li>
      <a href="articles/pandaR_DanielS.html">An Introduction to the pandaR Package</a>
    </li>
    <li>
      <a href="articles/panda_CamilaLR.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">


<p>Linux<a href="https://travis-ci.org/netZoo/netZooR"><img src="https://travis-ci-job-status.herokuapp.com/badge/netZoo/netZooR/devel/linux" alt="LINUX"></a></p>
<p>Macos<a href="https://travis-ci.org/netZoo/netZooR"><img src="https://travis-ci-job-status.herokuapp.com/badge/netZoo/netZooR/devel/macos" alt="MAC"></a></p>
<div id="description" class="section level2">
<h2 class="hasAnchor">
<a href="#description" class="anchor"></a>Description</h2>
<p>netZooR is an R package to construct, analyse and plot gene regulatory networks.</p>
</div>
<div id="features" class="section level2">
<h2 class="hasAnchor">
<a href="#features" class="anchor"></a>Features</h2>
<p>netZooR currently integrates with:</p>
<ul>
<li><p><strong>PANDA</strong>(Passing Attributes between Networks for Data Assimilation)<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0064832">[Glass et al. 2013]</a>: construct gene regulatory network from gene expression data, protein-protein interaction data, and transcription factor binding motifs (TFBMs) data.</p></li>
<li><p><strong>CONDOR</strong>(COmplex Network Description Of Regulators)<a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005033">[Platig et al. 2016]</a>: analyse bipartite community structure analysis of biological networks.</p></li>
<li><p><strong>LIONESS</strong>(Linear Interpolation to Obtain Network Estimates for Single Samples)<a href="https://doi.org/10.1016/j.isci.2019.03.021">[Kuijjer et al. 2019]</a>: reconstruct single-sample gene regulatory networks.</p></li>
<li><p><strong>ALPACA</strong>(ALtered Partitions Across Community Architectures)<a href="https://www.nature.com/articles/s41540-018-0052-5">[Padi and Quackenbush 2018]</a>: compare two networks and identify changes in modular structure.</p></li>
<li><p><strong>SAMBAR</strong>(Subtyping Agglomerated Mutations By Annotation Relations)<a href="https://www.nature.com/articles/s41416-018-0109-7">[Kuijjer et al.]</a>: identify subtypes based on somatic mutation data.</p></li>
<li><p><strong>MONSTER</strong>(Modeling Network State Transitions from Expression and Regulatory data)<a href="https://doi.org/10.1186/s12918-017-0517-y">[Schlauch et al.]</a>: infer transcription factor drivers of cell state conditions at the gene regulatory network level.</p></li>
<li><p>Source protein-protein interaction network from <a href="https://string-db.org/">STRINGdb</a> based on a list of protein of interest.</p></li>
<li><p>Plot one PANDA network in <a href="https://cytoscape.org/">Cytoscape</a>.</p></li>
<li><p>Plot two differential PANDA networks in Cytoscape.</p></li>
</ul>
</div>
<div id="requirement-installation-and-basic-configuration-" class="section level2">
<h2 class="hasAnchor">
<a href="#requirement-installation-and-basic-configuration-" class="anchor"></a>Requirement, installation and basic configuration.</h2>
<p>Using this pacakage requires <a href="https://www.python.org/downloads/"><strong>Python</strong></a> (3.X) and some <a href="#required-python-libraries">Python libraries</a>, <a href="https://cran.r-project.org/"><strong>R</strong></a> (&gt;= 3.3.3), and stable <strong>Internet access</strong>.</p>
<p>Some plotting functions will require the <a href="https://cytoscape.org/"><strong>Cytoscape</strong></a> installed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install.packages("devtools") </span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(devtools)
<span class="co"># install netZooR pkg with vignettes, otherwise remove the "build_vignettes = TRUE" argument.</span>
devtools<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/devtools/man/reexports.html">install_github</a></span>(<span class="st">"netZoo/netZooR"</span>, <span class="dt">build_vignettes =</span> <span class="ot">TRUE</span>)
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(netZooR)</code></pre></div>
<p>For more details please refer to the <a href="https://netzoo.github.io/netZooR/">documentation website</a>.</p>
<p>This package will invoke the Python in R environment through reticulate package. Configure which version of Python to use if necessary, here in netZooR, Python 3.X is required. More details can be found <a href="https://cran.r-project.org/web/packages/reticulate/vignettes/versions.html">here</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#check your Python configuration and the specific version of Python in use currently</span>
<span class="kw">py_config</span>()

<span class="co"># reset to Python 3.X if necessary, like below:</span>
<span class="co"># use_python("/usr/local/bin/python3")</span></code></pre></div>
</div>
<div id="issues" class="section level2">
<h2 class="hasAnchor">
<a href="#issues" class="anchor"></a>Issues</h2>
<p>For <strong>data.table</strong> installation issue please refer to <a href="https://github.com/netZoo/netZooR/issues/40">issue #40</a>.</p>
<p>Please report any further issue to the <a href="https://github.com/netZoo/netZooR/issues">issues page</a>.</p>
</div>
<div id="tutorials" class="section level2">
<h2 class="hasAnchor">
<a href="#tutorials" class="anchor"></a>Tutorials</h2>
<p>Please refer to the top navigation bar <strong>Articles/</strong> for basic usage and application cases. Or use <code><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes("netZooR")</a></code> after installing package.</p>
</div>
<div id="contribution-and-development" class="section level2">
<h2 class="hasAnchor">
<a href="#contribution-and-development" class="anchor"></a>Contribution and Development</h2>
<p>Contributions are welcome! Instruction of how to contribute netZooR repository can be found <a href="https://netzoo.github.io/contribute/contribute/">here</a>. After adding new features or debugging, please re-build package and run CDM check by using below codes before pulling request to the devel branch.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(devtools)
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(pkgdown)
<span class="co"># document the description of function</span>
devtools<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/devtools/man/document.html">document</a></span>()
<span class="co"># build vignettes</span>
devtools<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/devtools/man/build_vignettes.html">build_vignettes</a></span>()
<span class="co"># build documentation website</span>
pkgdown<span class="op">::</span><span class="kw"><a href="https://pkgdown.r-lib.org/reference/build_site.html">build_site</a></span>()
<span class="co"># devtools::build()</span>

<span class="co"># CMD check, if passed all tests here, it means this package is ready to pull request to the devel branch. Otherwise, fix the bug before pulling request.</span>
devtools<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/devtools/man/check.html">check</a></span>()</code></pre></div>
<p>The master branch on github should always be in good shape, so please to pull request to the <strong>devel</strong> branch. If the contribution is specific to pandaR, please contribute to its seperate GitHub page by <a href="https://github.com/jnpaulson/pandaR">pull request</a>.</p>
</div>
<div id="license" class="section level2">
<h2 class="hasAnchor">
<a href="#license" class="anchor"></a>License</h2>
<p>The software is free and is licensed under the GNU General License v3.0, see the file <a href="LICENSE" class="uri">LICENSE</a> for details.</p>
</div>
<div id="code-of-conduct" class="section level2">
<h2 class="hasAnchor">
<a href="#code-of-conduct" class="anchor"></a>Code of conduct</h2>
<p>Please note that this project is released with a <a href="CONDUCT.html">Contributor Code of Conduct</a>. By participating in this project you agree to abide by its terms.</p>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Tian Wang <br><small class="roles"> Author, maintainer </small>  </li>
<li>John Platig <br><small class="roles"> Author </small>  </li>
<li>Marieke Kuijjer <br><small class="roles"> Author </small>  </li>
<li>Magha Padi <br><small class="roles"> Author </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Tian Wang, John Platig, Marieke Kuijjer, Magha Padi.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
