<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Application with TB dataset • netZooR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/all.min.css" integrity="sha256-PbSmjxuVAzJ6FPvNYsrXygfGhNJYyZ2GktDbkMBqQZg=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/v4-shims.min.css" integrity="sha256-A6jcAdwFD48VMjlI3GDxUd+eCQa7/KWy6G9oe/ovaPA=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Application with TB dataset">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.6.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/ApplicationwithTBdataset.html">Application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Application with TB dataset</h1>
                        <h4 class="author">Tian Wang</h4>
            
            <h4 class="date">2019-10-19</h4>
      
      
      <div class="hidden name"><code>ApplicationwithTBdataset.Rmd</code></div>

    </div>

    
    
<div id="netzoor" class="section level1">
<h1 class="hasAnchor">
<a href="#netzoor" class="anchor"></a>netZooR</h1>
<p>netZooR is an R package which consists of five main algorithms and is able to construct, analyse and plot gene regulatory networks.</p>
<ul>
<li><p><strong>PANDA</strong>(Passing Attributes between Networks for Data Assimilation) is a message-passing model to gene regulatory network reconstruction. It integrates multiple sources of biological data, including protein-protein interaction, gene expression, and transcription factor binding motifs information to reconstruct genome-wide, condition-specific regulatory networks.<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0064832">[Glass et al. 2013]</a></p></li>
<li><p><strong>LIONESS</strong>(Linear Interpolation to Obtain Network Estimates for Single Samples) is a method to estimate sample-specific regulatory networks by applying linear interpolation to the predictions made by existing aggregate network inference approaches.<a href="https://doi.org/10.1016/j.isci.2019.03.021">[LIONESS arxiv paper]</a></p></li>
<li><p><strong>CONDOR</strong> (COmplex Network Description Of Regulators) implements methods to cluster biapartite networks and estimatiing the contribution of each node to its community’s modularity.<a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005033">[Platig et al. 2016]</a></p></li>
<li><p><strong>ALPACA</strong>(ALtered Partitions Across Community Architectures) is a method to compare two genome-scale networks derived from different phenotypic states to identify condition-specific modules. <a href="https://www.nature.com/articles/s41540-018-0052-5">[Padi and Quackenbush 2018]</a></p></li>
<li><p><strong>SAMBAR</strong>(Subtyping Agglomerated Mutations By Annotation Relations) is a method to identify subtypes based on somatic mutation data.<a href="https://www.nature.com/articles/s41416-018-0109-7">[Kuijjer et al.]</a>.</p></li>
</ul>
<div id="table-of-contents" class="section level2">
<h2 class="hasAnchor">
<a href="#table-of-contents" class="anchor"></a>Table of Contents</h2>
<ul>
<li>
<a href="#getting-started">Getting Started</a>
<ul>
<li><a href="#prerequisites">Prerequisites</a></li>
<li><a href="#required-python-libraries">Required Python libraries</a></li>
<li><a href="#installing">Installing</a></li>
</ul>
</li>
<li>
<a href="#data-resources">Data Resources</a>
<ul>
<li><a href="#motif-mapping-data">Motif mapping data</a></li>
<li><a href="#PPI">PPI</a></li>
</ul>
</li>
<li>
<a href="#running-the-sample-tb-datasets">Running the sample TB datasets</a>
<ul>
<li><a href="#panda-algorithm">PANDA algorithm</a></li>
<li><a href="#panda-cytoscape-plotting">PANDA Cytoscape Plotting</a></li>
<li><a href="#lioness-algorithm">LIONESS Algorithm</a></li>
<li><a href="#condor-algorithm-and-plotting">CONDOR Algorithm and plotting</a></li>
<li><a href="#alpaca-algorithm">ALPACA Algorithm</a></li>
<li><a href="#sambar-algorithm">SAMBAR Algorithm</a></li>
<li><a href="#monster-algorithm">MONSTER Algorithm</a></li>
</ul>
</li>
<li><a href="#more-tutorials">More tutorials</a></li>
<li><a href="#note">Note</a></li>
</ul>
</div>
<div id="getting-started" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting Started</h2>
<div id="prerequisites" class="section level3">
<h3 class="hasAnchor">
<a href="#prerequisites" class="anchor"></a>Prerequisites</h3>
<p>Using this pacakage requires <a href="https://www.python.org/downloads/"><strong>Python</strong></a> (3.X) and some <a href="#required-python-libraries">Python libraries</a>, <a href="https://cran.r-project.org/"><strong>R</strong></a> (&gt;= 3.3.3), and stable <strong>Internet access</strong>.</p>
<p>Some plotting functions will require the <a href="https://cytoscape.org/"><strong>Cytoscape</strong></a> installed.</p>
</div>
<div id="required-python-libraries" class="section level3">
<h3 class="hasAnchor">
<a href="#required-python-libraries" class="anchor"></a>Required Python libraries</h3>
<p>How to install Python libraries depends varies from different platforms. More instructions could be find <a href="https://packaging.python.org/tutorials/installing-packages/">here</a>.</p>
<p>The following Python libraries (or packages) are required by running PANDA and LIONESS algorithms:</p>
<p>The required Python packages are: <a href="https://pandas.pydata.org/">pandas</a>, <a href="http://www.numpy.org/">numpy</a>, <a href="https://networkx.github.io/">networkx</a>, <a href="https://matplotlib.org/api/pyplot_api.html">matplotlib.pyplot</a>.</p>
</div>
<div id="installing" class="section level3">
<h3 class="hasAnchor">
<a href="#installing" class="anchor"></a>Installing</h3>
<p>This package could be downloaded via <code><a href="https://rdrr.io/pkg/devtools/man/reexports.html">install_github()</a></code> function from <code>devtools</code> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install.packages("devtools") </span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(devtools)
<span class="co"># install netZooR pkg with vignettes, otherwise remove the "build_vignettes = TRUE" argument.</span>
devtools<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/devtools/man/reexports.html">install_github</a></span>(<span class="st">"netZoo/netZooR"</span>, <span class="dt">build_vignettes =</span> <span class="ot">TRUE</span>)</code></pre></div>
</div>
</div>
<div id="data-resources" class="section level2">
<h2 class="hasAnchor">
<a href="#data-resources" class="anchor"></a>Data Resources</h2>
<div id="motif-data" class="section level3">
<h3 class="hasAnchor">
<a href="#motif-data" class="anchor"></a>Motif data</h3>
<p>Here is some <a href="https://sites.google.com/a/channing.harvard.edu/kimberlyglass/tools/resources">PANDA-ready transcription factor binding motifs data</a> by species.</p>
</div>
<div id="ppi" class="section level3">
<h3 class="hasAnchor">
<a href="#ppi" class="anchor"></a>PPI</h3>
<p>This package includes a function <code>source.PPI</code> may source a Protein-Protein Interactions (PPI) througt STRING database given a list of proteins of interest. The <a href="http://www.bioconductor.org/packages/release/bioc/html/STRINGdb.html">STRINGdb</a> is already loaded while loading netZooR.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># TF is a data frame with single column filled with TFs of Mycobacterium tuberculosis H37Rv.</span>
PPI &lt;-<span class="st"> </span><span class="kw"><a href="../reference/source.PPI.html">source.PPI</a></span>(TF, <span class="dt">species.index=</span><span class="dv">83332</span>, <span class="dt">score_threshold=</span><span class="dv">0</span>)</code></pre></div>
</div>
</div>
<div id="running-the-sample-tb-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#running-the-sample-tb-datasets" class="anchor"></a>Running the sample TB datasets</h2>
<p>Use search() to check all loaded package currently.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(netZooR)
<span class="kw"><a href="https://rdrr.io/r/base/search.html">search</a></span>()</code></pre></div>
<p>Accessing the help pages for the usage of core functions.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><a href="../reference/panda.py.html">?panda.py
?create.condor.object
?panda.to.condor.object
?lioness.py
?alpaca
?panda.to.alpaca
?sambar</a></code></pre></div>
<p>This package will invoke the Python in R environment through reticulate package. Configure which version of Python to use if necessary, here in netZooR, Python 3.X is required. More details can be found <a href="https://cran.r-project.org/web/packages/reticulate/vignettes/versions.html">here</a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#check your Python configuration and the specific version of Python in use currently</span>
<span class="kw">py_config</span>()

<span class="co"># reset to Python 3.X if necessary, like below:</span>
<span class="co"># use_python("/usr/local/bin/python3")</span></code></pre></div>
<p>Use example datasets within package to test this package. Refer to four input datasets files: one TB expression dataset control group , one TB expression dataset treated, one transcription factor binding motifs dataset, and one protein-protein interaction datasets in inst/extdat.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># retrieve the file path of these files</span>
treated_expression_file_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"expr4.txt"</span>, <span class="dt">package =</span> <span class="st">"netZooR"</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span>)
control_expression_file_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"expr10.txt"</span>, <span class="dt">package =</span> <span class="st">"netZooR"</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span>)
motif_file_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"chip.txt"</span>, <span class="dt">package =</span> <span class="st">"netZooR"</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span>)
ppi_file_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"ppi.txt"</span>, <span class="dt">package =</span> <span class="st">"netZooR"</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span>)</code></pre></div>
<div id="panda-algorithm" class="section level3">
<h3 class="hasAnchor">
<a href="#panda-algorithm" class="anchor"></a>PANDA algorithm</h3>
<p>Assign the above file paths to flag <code>e</code>(refers to “expression dataset”), <code>m</code>(refers to “motif dataset”), and <code>ppi</code>(refers to “PPI” dataset), respectively. Then set option <code>rm_missing</code> to <code>TRUE</code> to run <strong>PANDA</strong> to generate an aggregate network without unmatched TF and genes.</p>
<p>Repeat with control group.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">treated_all_panda_result &lt;-<span class="st"> </span><span class="kw"><a href="../reference/panda.py.html">panda.py</a></span>(<span class="dt">expr =</span> treated_expression_file_path, <span class="dt">motif =</span> motif_file_path, <span class="dt">ppi =</span> ppi_file_path,<span class="dt">mode_process=</span><span class="st">"legacy"</span>,  <span class="dt">rm_missing =</span> <span class="ot">TRUE</span> )
control_all_panda_result &lt;-<span class="st"> </span><span class="kw"><a href="../reference/panda.py.html">panda.py</a></span>(<span class="dt">expr =</span> control_expression_file_path, <span class="dt">motif =</span> motif_file_path, <span class="dt">ppi =</span> ppi_file_path,<span class="dt">mode_process=</span><span class="st">"legacy"</span>,  <span class="dt">rm_missing =</span> <span class="ot">TRUE</span> )</code></pre></div>
<p>Vector <code>treated_all_panda_result</code> and vector <code>control_all_panda_result</code> below are large lists with three elements: the entire PANDA network, indegree (“to” nodes) nodes and score, outdegree (“from” nodes) nodes and score. Use <code>$panda</code>,<code>$indegree</code> and <code>$outdegree</code> to access each list item resepctively.</p>
<p>Use <code>$panda</code>to access the entire PANDA network.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">treated_net &lt;-<span class="st"> </span>treated_all_panda_result<span class="op">$</span>panda
control_net &lt;-<span class="st"> </span>control_all_panda_result<span class="op">$</span>panda</code></pre></div>
</div>
<div id="panda-cytoscape-plotting" class="section level3">
<h3 class="hasAnchor">
<a href="#panda-cytoscape-plotting" class="anchor"></a>PANDA Cytoscape Plotting</h3>
<p>Cytoscape is an interactivity network visualization tool highly recommanded to explore the PANDA network. Before using this function <code>plot.panda.in.cytoscape</code>, please install and launch Cytoscape (3.6.1 or greater) and keep it running whenever using.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># select top 1000 edges in PANDA network by edge weight.</span>
panda.net &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(treated_net[<span class="kw"><a href="https://rdrr.io/r/base/order.html">order</a></span>(control_net<span class="op">$</span>force,<span class="dt">decreasing =</span> <span class="ot">TRUE</span>),], <span class="dv">1000</span>)
 
<span class="co"># run this function to create a network in Cytoscape.</span>
<span class="kw">plot.panda.in.cytoscape</span>(panda.net, <span class="dt">network.name=</span><span class="st">"PANDA"</span>)</code></pre></div>
</div>
<div id="lioness-algorithm" class="section level3">
<h3 class="hasAnchor">
<a href="#lioness-algorithm" class="anchor"></a>LIONESS Algorithm</h3>
<p>How to run LIONESS is mostly idential with method how to run PANDA in this package, unless the return values of <code>runLioness</code> is a data frame where first two columns represent TFs (regulators) and Genes (targets) while the rest columns represent each sample. each cell filled with estimated score calculated by LIONESS.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Run LIONESS algorithm for the first two samples</span>
<span class="co"># removing start_sample and end_sample to generate whole LIONESS network with all samples.</span>
control_lioness_result &lt;-<span class="st"> </span><span class="kw"><a href="../reference/lioness.py.html">lioness.py</a></span>(<span class="dt">expr =</span> control_expression_file_path, <span class="dt">motif =</span> motif_file_path, <span class="dt">ppi =</span> ppi_file_path, <span class="dt">rm_missing =</span> <span class="ot">TRUE</span>, <span class="dt">start_sample=</span><span class="dv">1</span>, <span class="dt">end_sample=</span><span class="dv">2</span>)</code></pre></div>
</div>
<div id="condor-algorithm-and-plotting" class="section level3">
<h3 class="hasAnchor">
<a href="#condor-algorithm-and-plotting" class="anchor"></a>CONDOR Algorithm and plotting</h3>
<p>PANDA network can simply be converted into condor.object by <code><a href="../reference/panda.to.condor.object.html">panda.to.condor.object(panda.net, threshold)</a></code> Defaults option <code>threshold</code> is the average of [median weight of non-prior edges] and [median weight of prior edges], all weights mentioned previous are transformationed with formula <code>w'=ln(e^w+1)</code> before calculating the median and average. But all the edges selected will remain the orginal weights calculated by PANDA.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">treated_condor_object &lt;-<span class="st"> </span><span class="kw"><a href="../reference/panda.to.condor.object.html">panda.to.condor.object</a></span>(treated_net, <span class="dt">threshold =</span> <span class="dv">0</span>)</code></pre></div>
<p>The communities structure can be plotted by igraph.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">treated_color_num &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(treated_condor_object<span class="op">$</span>red.memb<span class="op">$</span>com)
treated_color &lt;-<span class="st"> </span><span class="kw">viridis</span>(treated_color_num, <span class="dt">alpha =</span> <span class="dv">1</span>, <span class="dt">begin =</span> <span class="dv">0</span>, <span class="dt">end =</span> <span class="dv">1</span>, <span class="dt">direction =</span> <span class="dv">1</span>, <span class="dt">option =</span> <span class="st">"D"</span>)
<span class="kw"><a href="../reference/condor.plot.communities.html">condor.plot.communities</a></span>(treated_condor_object, <span class="dt">color_list=</span>treated_color, <span class="dt">point.size=</span><span class="fl">0.04</span>, <span class="dt">xlab=</span><span class="st">"Target"</span>, <span class="dt">ylab=</span><span class="st">"Regulator"</span>)</code></pre></div>
</div>
<div id="alpaca-algorithm" class="section level3">
<h3 class="hasAnchor">
<a href="#alpaca-algorithm" class="anchor"></a>ALPACA Algorithm</h3>
<p>ALPACA community structure can also be generated from two PANDA network by <code>panda.to.alpaca</code></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">alpaca&lt;-<span class="st"> </span><span class="kw"><a href="../reference/panda.to.alpaca.html">panda.to.alpaca</a></span>(treated_net, control_net, <span class="st">"./TB"</span>, <span class="dt">verbose=</span><span class="ot">TRUE</span>)</code></pre></div>
</div>
<div id="sambar-algorithm" class="section level3">
<h3 class="hasAnchor">
<a href="#sambar-algorithm" class="anchor"></a>SAMBAR Algorithm</h3>
<p>(working in progress)</p>
</div>
<div id="monster-algorithm" class="section level3">
<h3 class="hasAnchor">
<a href="#monster-algorithm" class="anchor"></a>MONSTER Algorithm</h3>
<p>(working in progress)</p>
</div>
</div>
<div id="more-tutorials" class="section level2">
<h2 class="hasAnchor">
<a href="#more-tutorials" class="anchor"></a>More tutorials</h2>
<p>Browse with <code><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes("netZooR")</a></code></p>
</div>
<div id="note" class="section level2">
<h2 class="hasAnchor">
<a href="#note" class="anchor"></a>Note</h2>
<p>If there is an error like <code>Error in fetch(key) : lazy-load database.rdb' is corrupt</code> when accessing the help pages of functions in this package after being loaded. It’s <a href="https://github.com/r-lib/devtools/issues/1660">a limitation of base R</a> and has not been solved yet. Restart R session and re-load this package will help.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#netzoor">netZooR</a><ul class="nav nav-pills nav-stacked">
<li><a href="#table-of-contents">Table of Contents</a></li>
      <li><a href="#getting-started">Getting Started</a></li>
      <li><a href="#data-resources">Data Resources</a></li>
      <li><a href="#running-the-sample-tb-datasets">Running the sample TB datasets</a></li>
      <li><a href="#more-tutorials">More tutorials</a></li>
      <li><a href="#note">Note</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Tian Wang, John Platig, Marieke Kuijjer, Magha Padi.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
