<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="netZooR">
<title>Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R • netZooR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R">
<meta property="og:description" content="netZooR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">netZooR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.17</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    <a class="dropdown-item" href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    <a class="dropdown-item" href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    <a class="dropdown-item" href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    <a class="dropdown-item" href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    <a class="dropdown-item" href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    <a class="dropdown-item" href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    <a class="dropdown-item" href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    <a class="dropdown-item" href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    <a class="dropdown-item" href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/netZoo/netZooR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</h1>
                        <h4 data-toc-skip class="author">Deborah
Weighill</h4>
            
            <h4 data-toc-skip class="date">2019-11-06</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/netZoo/netZooR/blob/HEAD/vignettes/ApplicationinGTExData.Rmd" class="external-link"><code>vignettes/ApplicationinGTExData.Rmd</code></a></small>
      <div class="d-none name"><code>ApplicationinGTExData.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In this vignette, we will demonstrate the functionalities of
netZooR.</p>
</div>
<div class="section level2">
<h2 id="panda">PANDA<a class="anchor" aria-label="anchor" href="#panda"></a>
</h2>
<div class="section level3">
<h3 id="panda-overview">PANDA Overview<a class="anchor" aria-label="anchor" href="#panda-overview"></a>
</h3>
<p>PANDA (Passing Attributes between Networks for Data Assimilation) is
a method for constructing gene regulatory networks. It uses message
passing to find congruence between 3 different data layers:
protein-protein interaction (PPI), gene expression, and transcription
factor (TF) motif data.</p>
<p>More details can be found in the published paper <a href="https://doi.org/10.1371/journal.pone.0064832" class="external-link uri">https://doi.org/10.1371/journal.pone.0064832</a>.</p>
</div>
<div class="section level3">
<h3 id="running-a-single-panda-analysis">Running a single PANDA analysis<a class="anchor" aria-label="anchor" href="#running-a-single-panda-analysis"></a>
</h3>
<p>Load some libraries. We use the data.table library for reading in
large datasets as it is more efficient.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/netZoo/netZooR" class="external-link">netZooR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"visNetwork"</span>,repos <span class="op">=</span> <span class="st">"http://cran.us.r-project.org"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; The downloaded binary packages are in</span></span>
<span><span class="co">#&gt;  /var/folders/dc/zj7mjqq93034_xd2wszf4jjw0000gn/T//RtmpHoGYyJ/downloaded_packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://datastorm-open.github.io/visNetwork/" class="external-link">visNetwork</a></span><span class="op">)</span> <span class="co"># to visualize the networks</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># point R to your python 3 installation. Make sure that this is the installation that has all the required python libraries (numpy, scipy, etc) installed. netZooR uses a python implementation of PANDA under the hood.</span></span>
<span><span class="co">#use_python("/usr/bin/python3") # for example, you can check the installation with py_config()</span></span></code></pre></div>
<p>The previous command is necessary to bind R to Python since we are
calling PANDA from Python because netZooPy has an optimized
implementation of PANDA. Check <a href="http://netbooks.networkmedicine.org/user/marouenbg/notebooks/netZooR/panda_gtex_tutorial_server.ipynb" class="external-link">this
tutorial</a> for an example using a pure R implementation of PANDA. Now
we locate our ppi and motif priors. The ppi represents physical
interactions between transcription factor proteins, and is an undirected
network. The motif prior represents putative regulation events where a
transcription factor binds in the promotor of a gene to regulate its
expression, as predicted by the presence of transcription factor binding
motifs in the promotor region of the gene. The motif prior is thus a
directed network linking transcription factors to their predicted gene
targets. These are small example priors for the purposes of
demonstrating this method.</p>
<p>The ppi and motif priors are available in our AWS public bucket, and
can be downloaded into current working directory.</p>
<p>Let’s download and take a look at the priors:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout<span class="op">=</span><span class="fl">600</span><span class="op">)</span> <span class="co"># set timeout for file download</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/motif_GTEx.txt"</span>,<span class="st">"motif_GTEx.txt"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/ppi_GTEx.txt"</span>,<span class="st">"ppi_GTEx.txt"</span><span class="op">)</span></span>
<span><span class="va">motif</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"./motif_GTEx.txt"</span><span class="op">)</span> </span>
<span><span class="va">ppi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"./ppi_GTEx.txt"</span><span class="op">)</span></span>
<span><span class="va">ppi</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span>
<span><span class="co">#&gt;    V1   V2    V3</span></span>
<span><span class="co">#&gt; 1 AHR  AHR 1.000</span></span>
<span><span class="co">#&gt; 2 AHR ALX1 0.179</span></span>
<span><span class="co">#&gt; 3 AHR ALX3 0.179</span></span>
<span><span class="co">#&gt; 4 AHR ALX4 0.179</span></span>
<span><span class="co">#&gt; 5 AHR   AR 0.811</span></span>
<span><span class="va">motif</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span>
<span><span class="co">#&gt;    V1              V2 V3</span></span>
<span><span class="co">#&gt; 1 AHR ENSG00000004897  1</span></span>
<span><span class="co">#&gt; 2 AHR ENSG00000005339  1</span></span>
<span><span class="co">#&gt; 3 AHR ENSG00000006071  1</span></span>
<span><span class="co">#&gt; 4 AHR ENSG00000006194  1</span></span>
<span><span class="co">#&gt; 5 AHR ENSG00000006283  1</span></span></code></pre></div>
<p>Now we locate out expression data.</p>
<p>As example, We will use a portion of the GTEx (Genotype-Tissue
Expression) version 7 RNA-Seq data, read in the expression data and the
list of LCL samples. Then parse the expression data.</p>
<p>We can either</p>
<ol style="list-style-type: decimal">
<li>downlaod the file
GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_tpm.gct from from <a href="https://gtexportal.org/home/datasets" class="external-link uri">https://gtexportal.org/home/datasets</a> or in our AWS
bucket and place it in the folder “expressionData”. We will initially
use the LCL RNA-seq data to create a regulatory network for this cell
line. Later, we will also generate a regulatory network for whole blood
for comaprison.</li>
</ol>
<p>Here, we use the expression data and sample ids file copy from our
AWS bucket.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_tpm.gct"</span>,<span class="st">"GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_tpm.gct"</span><span class="op">)</span></span>
<span><span class="co"># load the GTEx expression matrix</span></span>
<span><span class="va">expr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/data.table/man/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">"./GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_tpm.gct"</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, skip <span class="op">=</span> <span class="fl">2</span>, data.table <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># remove the transcript ids so that the genes match the gene ids in the tf-motif prior</span></span>
<span><span class="va">expr</span><span class="op">$</span><span class="va">Name</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"\\.[0-9]"</span>,<span class="st">""</span>, <span class="va">expr</span><span class="op">$</span><span class="va">Name</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#downlooad and load the sample ids of LCL samples</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/LCL_samples.txt"</span>,<span class="st">"LCL_samples.txt"</span><span class="op">)</span></span>
<span><span class="va">lcl_samples</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/data.table/man/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">"./LCL_samples.txt"</span>, header <span class="op">=</span> <span class="cn">FALSE</span>, data.table<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#select the columns of the expression matrix corresponding to the LCL samples.</span></span>
<span><span class="va">lcl_expr</span> <span class="op">&lt;-</span> <span class="va">expr</span><span class="op">[</span>,<span class="fu"><a href="https://r.igraph.org/reference/union.html" class="external-link">union</a></span><span class="op">(</span><span class="st">"Name"</span>,<span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">lcl_samples</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">149</span>,<span class="op">]</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">expr</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, with<span class="op">=</span><span class="cn">FALSE</span><span class="op">]</span></span>
<span></span>
<span><span class="co">#determine the number of non-NA/non-zero rows in the expression data. This is to be able to ensure that PANDA will have enough values in the vectors to calculate pearson correlations between gene expression profiles in the construction of the gene co-exression prior.</span></span>
<span><span class="va">zero_na_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">lcl_expr</span>, MARGIN <span class="op">=</span> <span class="fl">1</span>, FUN <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">|</span> <span class="va">x</span><span class="op">!=</span><span class="fl">0</span><span class="op">)</span> <span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#maintain only genes with at least 20 valid gene expression entries</span></span>
<span><span class="va">clean_data</span> <span class="op">&lt;-</span> <span class="va">lcl_expr</span><span class="op">[</span><span class="va">zero_na_counts</span> <span class="op">&gt;</span> <span class="fl">20</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co">#write the cleaned expression data to a file, ready to be passed as an argument to the PANDA algorithm.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">clean_data</span>, file <span class="op">=</span> <span class="st">"pandaExprLCL.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, col.names <span class="op">=</span> <span class="cn">FALSE</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>, quote <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>Or</p>
<p>Download our pre-processed pandaExprLCL.txt from our AWS S3
Bucket.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/pandaExprLCL.txt"</span>,<span class="st">"pandaExprLCL.txt"</span><span class="op">)</span></span></code></pre></div>
<p>Now we run PANDA, pointing it to the parsed expression data, motif
prior and ppi prior.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">panda_results_LCL</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pandaPy.html">pandaPy</a></span><span class="op">(</span>expr_file <span class="op">=</span> <span class="st">"./pandaExprLCL.txt"</span> , motif_file <span class="op">=</span> <span class="st">"./motif_GTEx.txt"</span>, ppi_file <span class="op">=</span> <span class="st">"./ppi_GTEx.txt"</span>, modeProcess<span class="op">=</span><span class="st">"legacy"</span>, remove_missing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading motif data ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.02 sec.</span></span>
<span><span class="co">#&gt; Loading expression data ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.26 sec.</span></span>
<span><span class="co">#&gt; Loading PPI data ...</span></span>
<span><span class="co">#&gt; Number of PPIs: 86609</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.02 sec.</span></span>
<span><span class="co">#&gt; Remove expression not in motif:</span></span>
<span><span class="co">#&gt;    55689 rows removed from the initial 56202</span></span>
<span><span class="co">#&gt; Remove motif not in expression data:</span></span>
<span><span class="co">#&gt;    58955 rows removed from the initial 103400</span></span>
<span><span class="co">#&gt; Remove ppi not in motif:</span></span>
<span><span class="co">#&gt;    0 rows removed from the initial 86609</span></span>
<span><span class="co">#&gt; new case</span></span>
<span><span class="co">#&gt; Calculating coexpression network ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.00 sec.</span></span>
<span><span class="co">#&gt; Creating motif network ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.01 sec.</span></span>
<span><span class="co">#&gt; Creating PPI network ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.03 sec.</span></span>
<span><span class="co">#&gt; legacy ./motif_GTEx.txt ./pandaExprLCL.txt ./ppi_GTEx.txt False True False</span></span>
<span><span class="co">#&gt; Normalizing networks ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.02 sec.</span></span>
<span><span class="co">#&gt; Saving expression matrix and normalized networks ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.00 sec.</span></span>
<span><span class="co">#&gt; Running PANDA algorithm ...</span></span>
<span><span class="co">#&gt; step: 0, hamming: 0.9678742295275564</span></span>
<span><span class="co">#&gt; step: 1, hamming: 0.398740485692637</span></span>
<span><span class="co">#&gt; step: 2, hamming: 0.28448582363703867</span></span>
<span><span class="co">#&gt; step: 3, hamming: 0.2345843939954457</span></span>
<span><span class="co">#&gt; step: 4, hamming: 0.2042075476717615</span></span>
<span><span class="co">#&gt; step: 5, hamming: 0.18118973861477952</span></span>
<span><span class="co">#&gt; step: 6, hamming: 0.1614704894054021</span></span>
<span><span class="co">#&gt; step: 7, hamming: 0.14360782392336247</span></span>
<span><span class="co">#&gt; step: 8, hamming: 0.1271104933911665</span></span>
<span><span class="co">#&gt; step: 9, hamming: 0.11185402126446312</span></span>
<span><span class="co">#&gt; step: 10, hamming: 0.09783344721350776</span></span>
<span><span class="co">#&gt; step: 11, hamming: 0.08507016957437542</span></span>
<span><span class="co">#&gt; step: 12, hamming: 0.07356780340154848</span></span>
<span><span class="co">#&gt; step: 13, hamming: 0.06330116711146468</span></span>
<span><span class="co">#&gt; step: 14, hamming: 0.05421852200951187</span></span>
<span><span class="co">#&gt; step: 15, hamming: 0.04624764701825898</span></span>
<span><span class="co">#&gt; step: 16, hamming: 0.039301895859622696</span></span>
<span><span class="co">#&gt; step: 17, hamming: 0.03328718090872966</span></span>
<span><span class="co">#&gt; step: 18, hamming: 0.028107462583496577</span></span>
<span><span class="co">#&gt; step: 19, hamming: 0.023668722559827687</span></span>
<span><span class="co">#&gt; step: 20, hamming: 0.01988152451165294</span></span>
<span><span class="co">#&gt; step: 21, hamming: 0.016662707998748628</span></span>
<span><span class="co">#&gt; step: 22, hamming: 0.013936416334063792</span></span>
<span><span class="co">#&gt; step: 23, hamming: 0.011634447928678703</span></span>
<span><span class="co">#&gt; step: 24, hamming: 0.009696184613626214</span></span>
<span><span class="co">#&gt; step: 25, hamming: 0.008068282125831978</span></span>
<span><span class="co">#&gt; step: 26, hamming: 0.006704156831568275</span></span>
<span><span class="co">#&gt; step: 27, hamming: 0.005563429715892798</span></span>
<span><span class="co">#&gt; step: 28, hamming: 0.0046113098212006735</span></span>
<span><span class="co">#&gt; step: 29, hamming: 0.003817972519694833</span></span>
<span><span class="co">#&gt; step: 30, hamming: 0.0031579704664075493</span></span>
<span><span class="co">#&gt; step: 31, hamming: 0.002609676049734357</span></span>
<span><span class="co">#&gt; step: 32, hamming: 0.0021547746826681815</span></span>
<span><span class="co">#&gt; step: 33, hamming: 0.0017778073470498855</span></span>
<span><span class="co">#&gt; step: 34, hamming: 0.0014657623367399968</span></span>
<span><span class="co">#&gt; step: 35, hamming: 0.00120771534642688</span></span>
<span><span class="co">#&gt; step: 36, hamming: 0.0009945161790090437</span></span>
<span><span class="co">#&gt; Running panda took: 1.70 seconds!</span></span></code></pre></div>
<p>Let’s take a look at the results. The output contains a list of three
data frames:</p>
<ul>
<li>data frame containing the regulatory network (bipartite graph) with
edge weights representing the “likelihood” that a transcription factor
binds the promotor of and regulates the expression of a gene.<br>
</li>
<li>data frame odf the in-degrees of genes (sum of the weights of
inbound edges around a gene)</li>
<li>data frame of the out-degrees of TFs (sum of the weights of outbound
edges around a TF)</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># the bipartite regulatory network</span></span>
<span><span class="va">regNetLCL</span> <span class="op">&lt;-</span> <span class="va">panda_results_LCL</span><span class="op">$</span><span class="va">panda</span></span>
<span><span class="va">regNetLCL</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span>
<span><span class="co">#&gt;     TF            Gene Motif       Score</span></span>
<span><span class="co">#&gt; 1  AHR ENSG00000049245     0 -0.66273059</span></span>
<span><span class="co">#&gt; 2 AIRE ENSG00000049245     1  3.91520334</span></span>
<span><span class="co">#&gt; 3 ALX1 ENSG00000049245     0  0.04430445</span></span>
<span><span class="co">#&gt; 4 ALX3 ENSG00000049245     0  0.09652412</span></span>
<span><span class="co">#&gt; 5 ALX4 ENSG00000049245     0 -0.06595998</span></span>
<span></span>
<span><span class="co"># gene in-degree</span></span>
<span><span class="va">inDegreeLCL</span> <span class="op">&lt;-</span> <span class="va">panda_results_LCL</span><span class="op">$</span><span class="va">indegree</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">inDegreeLCL</span><span class="op">)</span></span>
<span><span class="co">#&gt;            Target Target_Score</span></span>
<span><span class="co">#&gt; 1 ENSG00000000005    -47.74480</span></span>
<span><span class="co">#&gt; 2 ENSG00000000419    -11.80864</span></span>
<span><span class="co">#&gt; 3 ENSG00000000457     38.85765</span></span>
<span><span class="co">#&gt; 4 ENSG00000000938     30.19761</span></span>
<span><span class="co">#&gt; 5 ENSG00000001036    -56.41923</span></span>
<span><span class="co">#&gt; 6 ENSG00000001084    292.77857</span></span>
<span></span>
<span><span class="co"># TF out-degree</span></span>
<span><span class="va">outDegreeLCL</span> <span class="op">&lt;-</span> <span class="va">panda_results_LCL</span><span class="op">$</span><span class="va">outdegree</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">outDegreeLCL</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Regulator Regulator_Score</span></span>
<span><span class="co">#&gt; 1       AHR      -172.47026</span></span>
<span><span class="co">#&gt; 2      AIRE       115.74408</span></span>
<span><span class="co">#&gt; 3      ALX1        83.45379</span></span>
<span><span class="co">#&gt; 4      ALX3       139.81888</span></span>
<span><span class="co">#&gt; 5      ALX4       119.41043</span></span>
<span><span class="co">#&gt; 6        AR      -146.19404</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-another-panda-analysis-on-whole-blood-samples">Run another PANDA analysis on Whole Blood Samples<a class="anchor" aria-label="anchor" href="#run-another-panda-analysis-on-whole-blood-samples"></a>
</h3>
<p>Like the LCL expression data in previous section, we can either
download the raw data and process;</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#### skip this part if you already did same process in LCL expression data section</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_tpm.gct"</span>,<span class="st">"GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_tpm.gct"</span><span class="op">)</span></span>
<span><span class="co"># load the GTEx expression matrix</span></span>
<span><span class="va">expr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/data.table/man/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">"./GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_tpm.gct"</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, skip <span class="op">=</span> <span class="fl">2</span>, data.table <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># remove the transcript ids so that the genes match the gene ids in the tf-motif prior</span></span>
<span><span class="va">expr</span><span class="op">$</span><span class="va">Name</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"\\.[0-9]"</span>,<span class="st">""</span>, <span class="va">expr</span><span class="op">$</span><span class="va">Name</span><span class="op">)</span></span>
<span><span class="co">#####</span></span>
<span></span>
<span><span class="co">#load the sample ids of Whole Blood samples</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/WholeBlood_samples.txt"</span>,<span class="st">"WholeBlood_samples.txt"</span><span class="op">)</span></span>
<span><span class="va">wblood_samples</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/data.table/man/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">"./WholeBlood_samples.txt"</span>, header <span class="op">=</span> <span class="cn">FALSE</span>, data.table<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#select the columns of the expression matrix corresponding to the LCL samples.</span></span>
<span><span class="va">wblood_expr</span> <span class="op">&lt;-</span> <span class="va">expr</span><span class="op">[</span>,<span class="fu"><a href="https://r.igraph.org/reference/union.html" class="external-link">union</a></span><span class="op">(</span><span class="st">"Name"</span>,<span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">wblood_samples</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">149</span>,<span class="op">]</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">expr</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, with<span class="op">=</span><span class="cn">FALSE</span><span class="op">]</span></span>
<span></span>
<span><span class="co">#determine the number of non-NA/non-zero rows in the expression data. This is to be able to ensure that PANDA will have enough values in the vectors to calculate pearson correlations between gene expression profiles in the construction of the gene co-exression prior.</span></span>
<span><span class="va">zero_na_counts_wblood</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">wblood_expr</span>, MARGIN <span class="op">=</span> <span class="fl">1</span>, FUN <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">|</span> <span class="va">x</span><span class="op">!=</span><span class="fl">0</span><span class="op">)</span> <span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#maintain only genes with at least 20 valid gene expression entries</span></span>
<span><span class="va">clean_data_wb</span> <span class="op">&lt;-</span> <span class="va">wblood_expr</span><span class="op">[</span><span class="va">zero_na_counts_wblood</span> <span class="op">&gt;</span> <span class="fl">20</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co">#write the cleaned expression data to a file, ready to be passed as an argument to the PANDA algorithm.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">clean_data_wb</span>, file <span class="op">=</span> <span class="st">"pandaExprWholeBlood.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, col.names <span class="op">=</span> <span class="cn">FALSE</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>, quote <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>or download the whole blood expression data directly from AWS
Bucket.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/pandaExprWholeBlood.txt"</span>,<span class="st">"pandaExprWholeBlood.txt"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#run PANDA</span></span>
<span><span class="va">panda_results_wblood</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pandaPy.html">pandaPy</a></span><span class="op">(</span>expr_file <span class="op">=</span> <span class="st">"./pandaExprWholeBlood.txt"</span> , motif_file <span class="op">=</span> <span class="st">"./motif_GTEx.txt"</span>, ppi_file <span class="op">=</span> <span class="st">"./ppi_GTEx.txt"</span>, modeProcess<span class="op">=</span><span class="st">"legacy"</span>, remove_missing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Use the legacy mode to pre-process the input dataset and keep only the matched TFs or Genes</span></span>
<span><span class="co">#&gt; Loading motif data ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.02 sec.</span></span>
<span><span class="co">#&gt; Loading expression data ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.07 sec.</span></span>
<span><span class="co">#&gt; Loading PPI data ...</span></span>
<span><span class="co">#&gt; Number of PPIs: 86609</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.02 sec.</span></span>
<span><span class="co">#&gt; Remove expression not in motif:</span></span>
<span><span class="co">#&gt;    55689 rows removed from the initial 56202</span></span>
<span><span class="co">#&gt; Remove motif not in expression data:</span></span>
<span><span class="co">#&gt;    58955 rows removed from the initial 103400</span></span>
<span><span class="co">#&gt; Remove ppi not in motif:</span></span>
<span><span class="co">#&gt;    0 rows removed from the initial 86609</span></span>
<span><span class="co">#&gt; new case</span></span>
<span><span class="co">#&gt; Calculating coexpression network ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.00 sec.</span></span>
<span><span class="co">#&gt; Creating motif network ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.01 sec.</span></span>
<span><span class="co">#&gt; Creating PPI network ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.03 sec.</span></span>
<span><span class="co">#&gt; legacy ./motif_GTEx.txt ./pandaExprWholeBlood.txt ./ppi_GTEx.txt False True False</span></span>
<span><span class="co">#&gt; Normalizing networks ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.02 sec.</span></span>
<span><span class="co">#&gt; Saving expression matrix and normalized networks ...</span></span>
<span><span class="co">#&gt;   Elapsed time: 0.01 sec.</span></span>
<span><span class="co">#&gt; Running PANDA algorithm ...</span></span>
<span><span class="co">#&gt; step: 0, hamming: 1.0690849805252594</span></span>
<span><span class="co">#&gt; step: 1, hamming: 0.4691397610863155</span></span>
<span><span class="co">#&gt; step: 2, hamming: 0.33036461786634547</span></span>
<span><span class="co">#&gt; step: 3, hamming: 0.26806926123263264</span></span>
<span><span class="co">#&gt; step: 4, hamming: 0.23042570772644422</span></span>
<span><span class="co">#&gt; step: 5, hamming: 0.20256536152502197</span></span>
<span><span class="co">#&gt; step: 6, hamming: 0.17929462709712923</span></span>
<span><span class="co">#&gt; step: 7, hamming: 0.15865735441197737</span></span>
<span><span class="co">#&gt; step: 8, hamming: 0.13989696240746843</span></span>
<span><span class="co">#&gt; step: 9, hamming: 0.12274921075135972</span></span>
<span><span class="co">#&gt; step: 10, hamming: 0.10712627123983814</span></span>
<span><span class="co">#&gt; step: 11, hamming: 0.09299263928203307</span></span>
<span><span class="co">#&gt; step: 12, hamming: 0.08031383413612993</span></span>
<span><span class="co">#&gt; step: 13, hamming: 0.0690361089114833</span></span>
<span><span class="co">#&gt; step: 14, hamming: 0.05908483808770143</span></span>
<span><span class="co">#&gt; step: 15, hamming: 0.05036857084130915</span></span>
<span><span class="co">#&gt; step: 16, hamming: 0.04278429695791332</span></span>
<span><span class="co">#&gt; step: 17, hamming: 0.036223979927264414</span></span>
<span><span class="co">#&gt; step: 18, hamming: 0.030579252521705833</span></span>
<span><span class="co">#&gt; step: 19, hamming: 0.025745150525848724</span></span>
<span><span class="co">#&gt; step: 20, hamming: 0.021622615090325713</span></span>
<span><span class="co">#&gt; step: 21, hamming: 0.018120119978282237</span></span>
<span><span class="co">#&gt; step: 22, hamming: 0.015154398422204278</span></span>
<span><span class="co">#&gt; step: 23, hamming: 0.012650788145759574</span></span>
<span><span class="co">#&gt; step: 24, hamming: 0.010543041908389054</span></span>
<span><span class="co">#&gt; step: 25, hamming: 0.008772952126201682</span></span>
<span><span class="co">#&gt; step: 26, hamming: 0.0072897621090774025</span></span>
<span><span class="co">#&gt; step: 27, hamming: 0.006049506006422676</span></span>
<span><span class="co">#&gt; step: 28, hamming: 0.0050143227470869035</span></span>
<span><span class="co">#&gt; step: 29, hamming: 0.004151769096928692</span></span>
<span><span class="co">#&gt; step: 30, hamming: 0.0034341712803001153</span></span>
<span><span class="co">#&gt; step: 31, hamming: 0.002838014367091252</span></span>
<span><span class="co">#&gt; step: 32, hamming: 0.0023433884769824866</span></span>
<span><span class="co">#&gt; step: 33, hamming: 0.0019334886365366035</span></span>
<span><span class="co">#&gt; step: 34, hamming: 0.0015941705108742305</span></span>
<span><span class="co">#&gt; step: 35, hamming: 0.0013135594305309585</span></span>
<span><span class="co">#&gt; step: 36, hamming: 0.0010817088184296895</span></span>
<span><span class="co">#&gt; step: 37, hamming: 0.0008903048282875369</span></span>
<span><span class="co">#&gt; Running panda took: 1.64 seconds!</span></span>
<span><span class="co">#&gt; ...Finish PANDA...</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"visNetwork"</span>,repos <span class="op">=</span> <span class="st">"http://cran.us.r-project.org"</span>,dependencies<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://datastorm-open.github.io/visNetwork/" class="external-link">visNetwork</a></span><span class="op">)</span></span>
<span><span class="va">edges</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">panda_results_wblood</span><span class="op">$</span><span class="va">panda</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">panda_results_wblood</span><span class="op">$</span><span class="va">panda</span><span class="op">$</span><span class="va">Score</span>,decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span>, <span class="fl">500</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">edges</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"from"</span>,<span class="st">"to"</span>,<span class="st">"motif"</span>,<span class="st">"force"</span><span class="op">)</span></span>
<span><span class="va">nodes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">edges</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">nodes</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">nodes</span><span class="op">$</span><span class="va">id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">edges</span><span class="op">$</span><span class="va">from</span>, <span class="st">"TF"</span>, <span class="st">"gene"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/visNetwork/man/visNetwork.html" class="external-link">visNetwork</a></span><span class="op">(</span><span class="va">nodes</span>, <span class="va">edges</span>, width <span class="op">=</span> <span class="st">"100%"</span><span class="op">)</span></span>
<span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/visNetwork/man/visGroups.html" class="external-link">visGroups</a></span><span class="op">(</span><span class="va">net</span>, groupname <span class="op">=</span> <span class="st">"TF"</span>, shape <span class="op">=</span> <span class="st">"square"</span>,</span>
<span>                     color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>background <span class="op">=</span> <span class="st">"teal"</span>, border<span class="op">=</span><span class="st">"black"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/visNetwork/man/visGroups.html" class="external-link">visGroups</a></span><span class="op">(</span><span class="va">net</span>, groupname <span class="op">=</span> <span class="st">"gene"</span>, shape <span class="op">=</span> <span class="st">"dot"</span>,       </span>
<span>                     color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>background <span class="op">=</span> <span class="st">"gold"</span>, border<span class="op">=</span><span class="st">"black"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/visNetwork/man/visLegend.html" class="external-link">visLegend</a></span><span class="op">(</span><span class="va">net</span>, main<span class="op">=</span><span class="st">"Legend"</span>, position<span class="op">=</span><span class="st">"right"</span>, ncol<span class="op">=</span><span class="fl">1</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="lioness">LIONESS<a class="anchor" aria-label="anchor" href="#lioness"></a>
</h2>
<p>LIONESS (Linear Interpolation to Obtain Network Estimates for Single
Samples) is a method for creating sample-specific networks. When applied
to a PANDA regulatory network, the result is a set of gene regulatory
networks, one for each sample in the gene expression dataset. More
information on LIONESS can be found in the published paper: <a href="https://doi.org/10.1016/j.isci.2019.03.021" class="external-link uri">https://doi.org/10.1016/j.isci.2019.03.021</a></p>
<p>Running LIONESS with netZoo is simple, and very similar to running
PANDA:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lionessLCL</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lionessPy.html">lionessPy</a></span><span class="op">(</span>expr_file <span class="op">=</span> <span class="st">"pandaExprLCL.txt"</span> , motif_file <span class="op">=</span> <span class="st">"./motif_GTEx.txt"</span>, ppi_file <span class="op">=</span> <span class="st">"./ppi_GTEx.txt"</span>, modeProcess<span class="op">=</span><span class="st">"legacy"</span>, remove_missing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Use the legacy mode to pre-process the input dataset and keep only the matched TFs or Genes</span></span></code></pre></div>
<p>The result is a data frame in which the first colum contains TFs, the
second column contains genes and each subsequent column contains the
edge weight for that particular TF-gene pair in a particular sample.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lionessLCL</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span><span class="co">#&gt;     tf            gene            1            2            3            4</span></span>
<span><span class="co">#&gt; 1  AHR ENSG00000049245 0.068203.... -1.01318.... -0.11908.... -0.64714....</span></span>
<span><span class="co">#&gt; 2 AIRE ENSG00000049245 3.405967.... 3.606517.... 1.341021.... 3.962872....</span></span>
<span><span class="co">#&gt; 3 ALX1 ENSG00000049245 -1.56464.... -0.65310.... -2.58347.... 0.069443....</span></span>
<span><span class="co">#&gt; 4 ALX3 ENSG00000049245 -2.26576.... 0.437433.... -1.84254.... 0.169395....</span></span>
<span><span class="co">#&gt; 5 ALX4 ENSG00000049245 -2.30018.... 0.112392.... -1.83354.... 0.015880....</span></span>
<span><span class="co">#&gt;              5            6            7            8</span></span>
<span><span class="co">#&gt; 1 -4.72954.... -0.17908.... -0.54619.... -0.96282....</span></span>
<span><span class="co">#&gt; 2 10.14026.... 2.654841.... 4.577900.... 3.730394....</span></span>
<span><span class="co">#&gt; 3 1.548765.... -0.63456.... 0.829474.... 2.063450....</span></span>
<span><span class="co">#&gt; 4 7.957915.... -0.44872.... 0.339952.... 1.848870....</span></span>
<span><span class="co">#&gt; 5 5.195770.... -0.37209.... 0.187463.... 2.238460....</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill, Kate Shutta.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
