<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="netZooR">
<title>netZooR application with TB dataset • netZooR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="netZooR application with TB dataset">
<meta property="og:description" content="netZooR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">netZooR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.17</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    <a class="dropdown-item" href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    <a class="dropdown-item" href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    <a class="dropdown-item" href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    <a class="dropdown-item" href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    <a class="dropdown-item" href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    <a class="dropdown-item" href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    <a class="dropdown-item" href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    <a class="dropdown-item" href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    <a class="dropdown-item" href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/netZoo/netZooR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>netZooR application with TB dataset</h1>
                        <h4 data-toc-skip class="author">Tian Wang</h4>
            
            <h4 data-toc-skip class="date">2019-10-16</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/netZoo/netZooR/blob/HEAD/vignettes/ApplicationwithTBdataset.Rmd" class="external-link"><code>vignettes/ApplicationwithTBdataset.Rmd</code></a></small>
      <div class="d-none name"><code>ApplicationwithTBdataset.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>netZooR is an R package which consists of seven main algorithms and
is able to construct, analyse and plot gene regulatory networks.</p>
<ul>
<li><p><strong>PANDA</strong>(Passing Attributes between Networks for
Data Assimilation) is a message-passing model to gene regulatory network
reconstruction. It integrates multiple sources of biological data,
including protein-protein interaction, gene expression, and
transcription factor binding motifs information to reconstruct
genome-wide, condition-specific regulatory networks.<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0064832" class="external-link">[Glass
et al. 2013]</a></p></li>
<li><p><strong>LIONESS</strong>(Linear Interpolation to Obtain Network
Estimates for Single Samples) is a method to estimate sample-specific
regulatory networks by applying linear interpolation to the predictions
made by existing aggregate network inference approaches.[[Kuijjer et
al. 2019]]](<a href="https://www.sciencedirect.com/science/article/pii/S2589004219300872" class="external-link uri">https://www.sciencedirect.com/science/article/pii/S2589004219300872</a>)</p></li>
<li><p><strong>CONDOR</strong> (COmplex Network Description Of
Regulators) implements methods to cluster biapartite networks and
estimatiing the contribution of each node to its community’s
modularity.<a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005033" class="external-link">[Platig
et al. 2016]</a></p></li>
<li><p><strong>ALPACA</strong>(ALtered Partitions Across Community
Architectures) is a method to compare two genome-scale networks derived
from different phenotypic states to identify condition-specific modules.
<a href="https://www.nature.com/articles/s41540-018-0052-5" class="external-link">[Padi and
Quackenbush 2018]</a></p></li>
<li><p><strong>SAMBAR</strong>(Subtyping Agglomerated Mutations By
Annotation Relations) is a method to identify subtypes based on somatic
mutation data.<a href="https://www.nature.com/articles/s41416-018-0109-7" class="external-link">[Kuijjer et
al.]</a>.</p></li>
<li><p><strong>MONSTER</strong>(Modeling Network State Transitions from
Expression and Regulatory data)<a href="https://doi.org/10.1186/s12918-017-0517-y" class="external-link">[Schlauch et al.]</a>:
infers transcription factor which drivers of cell state conditions at
the gene regulatory network level.</p></li>
<li><p><strong>OTTER</strong>(Optimization to Estimate Regulation)
[publication in preparation]: models gene regulation estimation as a
graph matrching problem</p></li>
</ul>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<div class="section level4">
<h4 id="prerequisites">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h4>
<p>Using this pacakage requires <a href="https://www.python.org/downloads/" class="external-link"><strong>Python</strong></a>
(3.X) and some <a href="#required-python-libraries">Python
libraries</a>, <a href="https://cran.r-project.org/" class="external-link"><strong>R</strong></a> (&gt;= 3.3.3),
and stable <strong>Internet access</strong>.</p>
<p>Some plotting functions will require the <a href="https://cytoscape.org/" class="external-link"><strong>Cytoscape</strong></a>
installed.</p>
</div>
<div class="section level4">
<h4 id="required-python-libraries">Required Python libraries<a class="anchor" aria-label="anchor" href="#required-python-libraries"></a>
</h4>
<p>How to install Python libraries depends varies from different
platforms. More instructions could be find <a href="https://packaging.python.org/tutorials/installing-packages/" class="external-link">here</a>.</p>
<p>The following Python libraries (or packages) are required by running
PANDA and LIONESS algorithms:</p>
<p>The required Python packages are: <a href="https://pandas.pydata.org/" class="external-link">pandas</a>, <a href="http://www.numpy.org/" class="external-link">numpy</a>, <a href="https://networkx.github.io/" class="external-link">networkx</a>, <a href="https://matplotlib.org/api/pyplot_api.html" class="external-link">matplotlib.pyplot</a>.</p>
</div>
<div class="section level4">
<h4 id="installing">Installing<a class="anchor" aria-label="anchor" href="#installing"></a>
</h4>
<p>This package could be downloaded via <code><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github()</a></code>
function from <code>devtools</code> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools") </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="co"># install netZooR pkg with vignettes, otherwise remove the "build_vignettes = TRUE" argument.</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"netZoo/netZooR"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridisLite/" class="external-link">viridisLite</a></span><span class="op">)</span><span class="co">#To visualize communities</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="data-resources">Data Resources<a class="anchor" aria-label="anchor" href="#data-resources"></a>
</h3>
<div class="section level4">
<h4 id="motif-data">Motif data<a class="anchor" aria-label="anchor" href="#motif-data"></a>
</h4>
<p>Here is some pre-prepared specie-sepcific PANDA-ready transcription
factor binding motifs data stored in our AWS bucket <a href="https://s3.console.aws.amazon.com/s3/buckets/netzoo/netZooR/example_datasets/PANDA_ready_motif_prior/?region=us-east-2&amp;tab=overview" class="external-link uri">https://s3.console.aws.amazon.com/s3/buckets/netzoo/netZooR/example_datasets/PANDA_ready_motif_prior/?region=us-east-2&amp;tab=overview</a>,
which are derived from motif scan and motif info files located on <a href="https://sites.google.com/a/channing.harvard.edu/kimberlyglass/tools/resourcesby" class="external-link uri">https://sites.google.com/a/channing.harvard.edu/kimberlyglass/tools/resourcesby</a>
.</p>
</div>
<div class="section level4">
<h4 id="ppi">PPI<a class="anchor" aria-label="anchor" href="#ppi"></a>
</h4>
<p>This package includes a function <code>source.PPI</code> may source a
Protein-Protein Interactions (PPI) througt STRING database given a list
of proteins of interest. The <a href="http://www.bioconductor.org/packages/release/bioc/html/STRINGdb.html" class="external-link">STRINGdb</a>
is already loaded while loading netZooR.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># TF is a data frame with single column filled with TFs of Mycobacterium tuberculosis H37Rv.</span></span>
<span><span class="va">PPI</span> <span class="op">&lt;-</span> <span class="fu">source.PPI</span><span class="op">(</span><span class="va">TF</span>, STRING.version<span class="op">=</span><span class="st">"10"</span>, species.index<span class="op">=</span><span class="fl">83332</span>, score_threshold<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="running-the-sample-tb-datasets">Running the sample TB datasets<a class="anchor" aria-label="anchor" href="#running-the-sample-tb-datasets"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/netZoo/netZooR" class="external-link">netZooR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: igraph</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'igraph'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     decompose, spectrum</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     union</span></span>
<span><span class="co">#&gt; Loading required package: reticulate</span></span>
<span><span class="co">#&gt; Loading required package: pandaR</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:igraph':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     normalize, path, union</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; Loading required package: matrixcalc</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'matrixcalc'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:igraph':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     %s%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Setting options('download.file.method.GEOquery'='auto')</span></span>
<span><span class="co">#&gt; Setting options('GEOquery.inmemory.gpl'=FALSE)</span></span></code></pre></div>
<p>Accessing the help pages for the usage of core functions.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">?</span><span class="va">pandaPy</span></span>
<span><span class="op">?</span><span class="va">createCondorObject</span></span>
<span><span class="op">?</span><span class="va">pandaToCondorObject</span></span>
<span><span class="op">?</span><span class="va">lionessPy</span></span>
<span><span class="op">?</span><span class="va">alpaca</span></span>
<span><span class="op">?</span><span class="va">pandaToAlpaca</span></span>
<span><span class="op">?</span><span class="va">sambar</span></span></code></pre></div>
<p>This package will invoke the Python in R environment through
reticulate package. Configure which version of Python to use if
necessary, here in netZooR, Python 3.X is required. More details can be
found <a href="https://cran.r-project.org/web/packages/reticulate/vignettes/versions.html" class="external-link">here</a></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#check your Python configuration and the specific version of Python in use currently</span></span>
<span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/py_config.html" class="external-link">py_config</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># reset to Python 3.X if necessary, like below:</span></span>
<span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/use_python.html" class="external-link">use_python</a></span><span class="op">(</span><span class="st">"/usr/local/bin/python3"</span><span class="op">)</span></span></code></pre></div>
<p>The previous command is necessary to bind R to Python since we are
calling PANDA from Python because netZooPy has an optimized
implementation of PANDA. Check <a href="http://netbooks.networkmedicine.org/user/marouenbg/notebooks/netZooR/panda_gtex_tutorial_server.ipynb" class="external-link">this
tutorial</a> for an example using a pure R implementation of PANDA. Use
example data sets within package to test this package. Refer to four
input datasets files: one TB expression dataset control group , one TB
expression dataset treated, one transcription factor binding motifs
dataset, and one protein-protein interaction datasets from either
inst/extdat or AWS.</p>
<p>retrieve the file path of these files came with the netZooR
package.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># retrieve the file path of these files</span></span>
<span><span class="va">treated_expression_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"expr4_matched.txt"</span>, package <span class="op">=</span> <span class="st">"netZooR"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">control_expression_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"expr10_matched.txt"</span>, package <span class="op">=</span> <span class="st">"netZooR"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">motif_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"chip_matched.txt"</span>, package <span class="op">=</span> <span class="st">"netZooR"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">ppi_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ppi_matched.txt"</span>, package <span class="op">=</span> <span class="st">"netZooR"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="panda-algorithm">PANDA algorithm<a class="anchor" aria-label="anchor" href="#panda-algorithm"></a>
</h4>
<p>Assign the above file paths to flag <code>e</code>(refers to
“expression dataset”), <code>m</code>(refers to “motif dataset”), and
<code>ppi</code>(refers to “PPI” dataset), respectively. Then set option
<code>rm_missing</code> to <code>TRUE</code> to run
<strong>PANDA</strong> to generate an aggregate network without
unmatched TF and genes.</p>
<p>Repeat with control group.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treated_all_panda_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pandaPy.html">pandaPy</a></span><span class="op">(</span>expr_file <span class="op">=</span> <span class="va">treated_expression_file_path</span>, motif_file <span class="op">=</span> <span class="va">motif_file_path</span>, ppi_file<span class="op">=</span> <span class="va">ppi_file_path</span>,modeProcess<span class="op">=</span><span class="st">"legacy"</span>,  remove_missing <span class="op">=</span> <span class="cn">TRUE</span> <span class="op">)</span></span>
<span><span class="va">control_all_panda_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pandaPy.html">pandaPy</a></span><span class="op">(</span>expr_file <span class="op">=</span> <span class="va">control_expression_file_path</span>,motif_file <span class="op">=</span> <span class="va">motif_file_path</span>, ppi_file<span class="op">=</span> <span class="va">ppi_file_path</span>,modeProcess<span class="op">=</span><span class="st">"legacy"</span>,  remove_missing <span class="op">=</span> <span class="cn">TRUE</span> <span class="op">)</span></span></code></pre></div>
<p>Vector <code>treated_all_panda_result</code> and vector
<code>control_all_panda_result</code> below are large lists with three
elements: the entire PANDA network, indegree (“to” nodes) nodes and
score, outdegree (“from” nodes) nodes and score. Use
<code>$panda</code>,<code>$indegree</code> and <code>$outdegree</code>
to access each list item resepctively.</p>
<p>Use <code>$panda</code>to access the entire PANDA network.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treated_net</span> <span class="op">&lt;-</span> <span class="va">treated_all_panda_result</span><span class="op">$</span><span class="va">panda</span></span>
<span><span class="va">control_net</span> <span class="op">&lt;-</span> <span class="va">control_all_panda_result</span><span class="op">$</span><span class="va">panda</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="panda-cytoscape-plotting">PANDA Cytoscape Plotting<a class="anchor" aria-label="anchor" href="#panda-cytoscape-plotting"></a>
</h4>
<p>Cytoscape is an interactivity network visualization tool highly
recommanded to explore the PANDA network. Before using this function
<code>plot.panda.in.cytoscape</code>, please install and launch
Cytoscape (3.6.1 or greater) and keep it running whenever using.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select top 1000 edges in PANDA network by edge weight.</span></span>
<span><span class="va">panda.net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">treated_net</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">control_net</span><span class="op">$</span><span class="va">force</span>,decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span>, <span class="fl">1000</span><span class="op">)</span></span>
<span> </span>
<span><span class="co"># run this function to create a network in Cytoscape.</span></span>
<span><span class="fu">vis.panda.in.cytoscape</span><span class="op">(</span><span class="va">panda.net</span>, network.name<span class="op">=</span><span class="st">"PANDA"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="lioness-algorithm">LIONESS Algorithm<a class="anchor" aria-label="anchor" href="#lioness-algorithm"></a>
</h4>
<p>How to run LIONESS is mostly idential with method how to run PANDA in
this package, unless the return values of <code><a href="../reference/lionessPy.html">lionessPy()</a></code> is a
data frame where first two columns represent TFs (regulators) and Genes
(targets) while the rest columns represent each sample. each cell filled
with estimated score calculated by LIONESS.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run LIONESS algorithm for the first two samples</span></span>
<span><span class="co"># removing start_sample and end_sample arguments to generate whole LIONESS network with all samples.</span></span>
<span><span class="va">control_lioness_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lionessPy.html">lionessPy</a></span><span class="op">(</span>expr_file <span class="op">=</span> <span class="va">control_expression_file_path</span>,motif_file <span class="op">=</span> <span class="va">motif_file_path</span>, ppi_file<span class="op">=</span> <span class="va">ppi_file_path</span>,modeProcess<span class="op">=</span><span class="st">"legacy"</span>,  remove_missing <span class="op">=</span> <span class="cn">TRUE</span>, start_sample<span class="op">=</span><span class="fl">1</span>, end_sample<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="condor-algorithm-and-plotting">CONDOR Algorithm and plotting<a class="anchor" aria-label="anchor" href="#condor-algorithm-and-plotting"></a>
</h4>
<p>PANDA network can simply be converted into condor.object by
<code>pandaToCondorObject(panda.net, threshold)</code> Defaults option
<code>threshold</code> is the average of [median weight of non-prior
edges] and [median weight of prior edges], all weights mentioned
previous are transformationed with formula <code>w'=ln(e^w+1)</code>
before calculating the median and average. But all the edges selected
will remain the orginal weights calculated by PANDA.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treated_condor_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pandaToCondorObject.html">pandaToCondorObject</a></span><span class="op">(</span><span class="va">treated_net</span>, threshold <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p>The communities structure can be plotted by igraph.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridisLite/" class="external-link">viridisLite</a></span><span class="op">)</span></span>
<span><span class="va">treated_condor_object</span> <span class="op">&lt;-</span><span class="fu"><a href="../reference/condorCluster.html">condorCluster</a></span><span class="op">(</span><span class="va">treated_condor_object</span>,project <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "modularity of projected graph 0.305307827990727"</span></span>
<span><span class="co">#&gt; [1] "Q = 0.306247018503079"</span></span>
<span><span class="co">#&gt; [1] "Q = 0.306499579969213"</span></span>
<span><span class="co">#&gt; [1] "Q = 0.306499579969213"</span></span>
<span><span class="va">treated_color_num</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">treated_condor_object</span><span class="op">$</span><span class="va">red.memb</span><span class="op">$</span><span class="va">com</span><span class="op">)</span></span>
<span><span class="va">treated_color</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span><span class="va">treated_color_num</span>, alpha <span class="op">=</span> <span class="fl">1</span>, begin <span class="op">=</span> <span class="fl">0</span>, end <span class="op">=</span> <span class="fl">1</span>, direction <span class="op">=</span> <span class="fl">1</span>, option <span class="op">=</span> <span class="st">"D"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/condorPlotCommunities.html">condorPlotCommunities</a></span><span class="op">(</span><span class="va">treated_condor_object</span>, color_list<span class="op">=</span><span class="va">treated_color</span>, point.size<span class="op">=</span><span class="fl">0.04</span>, xlab<span class="op">=</span><span class="st">"Genes"</span>, ylab<span class="op">=</span><span class="st">"TFs"</span><span class="op">)</span></span></code></pre></div>
<p><img src="ApplicationwithTBdataset_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="alpaca-algorithm">ALPACA Algorithm<a class="anchor" aria-label="anchor" href="#alpaca-algorithm"></a>
</h4>
<p>ALPACA community structure can also be generated from two PANDA
network by <code>pandaToAlpaca</code></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alpaca</span><span class="op">&lt;-</span> <span class="fu"><a href="../reference/pandaToAlpaca.html">pandaToAlpaca</a></span><span class="op">(</span><span class="va">treated_net</span>, <span class="va">control_net</span>, <span class="cn">NULL</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Detecting communities in control network..."</span></span>
<span><span class="co">#&gt; Weights detected. Building condor object with weighted edges.</span></span>
<span><span class="co">#&gt; [1] "modularity of projected graph 0.479298199886884"</span></span>
<span><span class="co">#&gt; [1] "Q = 0.487741857619518"</span></span>
<span><span class="co">#&gt; [1] "Q = 0.487741857619518"</span></span>
<span><span class="co">#&gt; [1] "Computing differential modularity matrix..."</span></span>
<span><span class="co">#&gt; [1] "Computing differential modules..."</span></span>
<span><span class="co">#&gt; [1] "Merging 2644 communities"</span></span>
<span><span class="co">#&gt; [1] 1</span></span>
<span><span class="co">#&gt; [1] 2</span></span>
<span><span class="co">#&gt; [1] 3</span></span>
<span><span class="co">#&gt; [1] "Merging 138 communities"</span></span>
<span><span class="co">#&gt; [1] 1</span></span>
<span><span class="co">#&gt; [1] 2</span></span>
<span><span class="co">#&gt; [1] 3</span></span>
<span><span class="co">#&gt; [1] 4</span></span>
<span><span class="co">#&gt; [1] 5</span></span>
<span><span class="co">#&gt; [1] "Merging 19 communities"</span></span>
<span><span class="co">#&gt; [1] 1</span></span>
<span><span class="co">#&gt; [1] 2</span></span>
<span><span class="co">#&gt; [1] 3</span></span>
<span><span class="co">#&gt; [1] "Merging 13 communities"</span></span>
<span><span class="co">#&gt; [1] 1</span></span>
<span><span class="co">#&gt; [1] "Computing node scores..."</span></span>
<span><span class="co">#&gt; [1] 1</span></span>
<span><span class="co">#&gt; [1] 2</span></span>
<span><span class="co">#&gt; [1] 3</span></span>
<span><span class="co">#&gt; [1] 4</span></span>
<span><span class="co">#&gt; [1] 5</span></span>
<span><span class="co">#&gt; [1] 6</span></span>
<span><span class="co">#&gt; [1] 7</span></span>
<span><span class="co">#&gt; [1] 8</span></span>
<span><span class="co">#&gt; [1] 9</span></span>
<span><span class="co">#&gt; [1] 10</span></span>
<span><span class="co">#&gt; [1] 11</span></span>
<span><span class="co">#&gt; [1] 12</span></span>
<span><span class="co">#&gt; [1] 13</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="more-tutorials">More tutorials<a class="anchor" aria-label="anchor" href="#more-tutorials"></a>
</h3>
<p>Browse with <code>browseVignettes("netZooR")</code></p>
</div>
<div class="section level3">
<h3 id="information">Information<a class="anchor" aria-label="anchor" href="#information"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Sonoma 14.1.2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/New_York</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] viridisLite_0.4.2   netZooR_1.5.17      matrixcalc_1.0-6   </span></span>
<span><span class="co">#&gt; [4] pandaR_1.34.0       Biobase_2.62.0      BiocGenerics_0.48.1</span></span>
<span><span class="co">#&gt; [7] reticulate_1.35.0   igraph_2.0.3       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] STRINGdb_2.14.3             fs_1.6.3                   </span></span>
<span><span class="co">#&gt;   [3] matrixStats_1.2.0           bitops_1.0-7               </span></span>
<span><span class="co">#&gt;   [5] httr_1.4.7                  RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt;   [7] doParallel_1.0.17           Rgraphviz_2.46.0           </span></span>
<span><span class="co">#&gt;   [9] repr_1.1.7                  tools_4.3.2                </span></span>
<span><span class="co">#&gt;  [11] doRNG_1.8.6                 backports_1.4.1            </span></span>
<span><span class="co">#&gt;  [13] utf8_1.2.4                  R6_2.5.1                   </span></span>
<span><span class="co">#&gt;  [15] vegan_2.6-4                 HDF5Array_1.30.1           </span></span>
<span><span class="co">#&gt;  [17] mgcv_1.9-1                  rhdf5filters_1.14.1        </span></span>
<span><span class="co">#&gt;  [19] permute_0.9-7               withr_3.0.0                </span></span>
<span><span class="co">#&gt;  [21] prettyunits_1.2.0           base64_2.0.1               </span></span>
<span><span class="co">#&gt;  [23] preprocessCore_1.64.0       cli_3.6.2                  </span></span>
<span><span class="co">#&gt;  [25] textshaping_0.3.7           penalized_0.9-52           </span></span>
<span><span class="co">#&gt;  [27] sass_0.4.9                  readr_2.1.5                </span></span>
<span><span class="co">#&gt;  [29] genefilter_1.84.0           askpass_1.2.0              </span></span>
<span><span class="co">#&gt;  [31] pkgdown_2.0.7               Rsamtools_2.18.0           </span></span>
<span><span class="co">#&gt;  [33] systemfonts_1.0.6           pbdZMQ_0.3-11              </span></span>
<span><span class="co">#&gt;  [35] siggenes_1.76.0             illuminaio_0.44.0          </span></span>
<span><span class="co">#&gt;  [37] AnnotationForge_1.44.0      scrime_1.3.5               </span></span>
<span><span class="co">#&gt;  [39] plotrix_3.8-4               limma_3.58.1               </span></span>
<span><span class="co">#&gt;  [41] rstudioapi_0.16.0           RSQLite_2.3.5              </span></span>
<span><span class="co">#&gt;  [43] generics_0.1.3              GOstats_2.68.0             </span></span>
<span><span class="co">#&gt;  [45] quantro_1.36.0              BiocIO_1.12.0              </span></span>
<span><span class="co">#&gt;  [47] gtools_3.9.5                distributional_0.4.0       </span></span>
<span><span class="co">#&gt;  [49] dplyr_1.1.4                 GO.db_3.18.0               </span></span>
<span><span class="co">#&gt;  [51] Matrix_1.6-5                fansi_1.0.6                </span></span>
<span><span class="co">#&gt;  [53] S4Vectors_0.40.2            abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [55] lifecycle_1.0.4             yaml_2.3.8                 </span></span>
<span><span class="co">#&gt;  [57] edgeR_4.0.16                SummarizedExperiment_1.32.0</span></span>
<span><span class="co">#&gt;  [59] gplots_3.1.3.1              rhdf5_2.46.1               </span></span>
<span><span class="co">#&gt;  [61] SparseArray_1.2.4           BiocFileCache_2.10.1       </span></span>
<span><span class="co">#&gt;  [63] grid_4.3.2                  blob_1.2.4                 </span></span>
<span><span class="co">#&gt;  [65] crayon_1.5.2                lattice_0.22-6             </span></span>
<span><span class="co">#&gt;  [67] GenomicFeatures_1.54.4      annotate_1.80.0            </span></span>
<span><span class="co">#&gt;  [69] KEGGREST_1.42.0             pillar_1.9.0               </span></span>
<span><span class="co">#&gt;  [71] knitr_1.45                  beanplot_1.3.1             </span></span>
<span><span class="co">#&gt;  [73] GenomicRanges_1.54.1        rjson_0.2.21               </span></span>
<span><span class="co">#&gt;  [75] codetools_0.2-19            glue_1.7.0                 </span></span>
<span><span class="co">#&gt;  [77] downloader_0.4              data.table_1.15.2          </span></span>
<span><span class="co">#&gt;  [79] vctrs_0.6.5                 png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [81] gtable_0.3.4                assertthat_0.2.1           </span></span>
<span><span class="co">#&gt;  [83] gsubfn_0.7                  cachem_1.0.8               </span></span>
<span><span class="co">#&gt;  [85] xfun_0.42                   S4Arrays_1.2.1             </span></span>
<span><span class="co">#&gt;  [87] survival_3.5-8              iterators_1.0.14           </span></span>
<span><span class="co">#&gt;  [89] statmod_1.5.0               nlme_3.1-164               </span></span>
<span><span class="co">#&gt;  [91] Category_2.68.0             bit64_4.0.5                </span></span>
<span><span class="co">#&gt;  [93] progress_1.2.3              filelock_1.0.3             </span></span>
<span><span class="co">#&gt;  [95] rprojroot_2.0.4             GenomeInfoDb_1.38.8        </span></span>
<span><span class="co">#&gt;  [97] tensorA_0.36.2.1            bslib_0.6.2                </span></span>
<span><span class="co">#&gt;  [99] nor1mix_1.3-2               KernSmooth_2.23-22         </span></span>
<span><span class="co">#&gt; [101] colorspace_2.1-0            DBI_1.2.2                  </span></span>
<span><span class="co">#&gt; [103] nnet_7.3-19                 processx_3.8.4             </span></span>
<span><span class="co">#&gt; [105] tidyselect_1.2.1            bit_4.0.5                  </span></span>
<span><span class="co">#&gt; [107] compiler_4.3.2              curl_5.2.1                 </span></span>
<span><span class="co">#&gt; [109] chron_2.3-61                graph_1.80.0               </span></span>
<span><span class="co">#&gt; [111] xml2_1.3.6                  desc_1.4.3                 </span></span>
<span><span class="co">#&gt; [113] ggdendro_0.2.0              DelayedArray_0.28.0        </span></span>
<span><span class="co">#&gt; [115] posterior_1.5.0             rtracklayer_1.62.0         </span></span>
<span><span class="co">#&gt; [117] checkmate_2.3.1             scales_1.3.0               </span></span>
<span><span class="co">#&gt; [119] caTools_1.18.2              hexbin_1.28.3              </span></span>
<span><span class="co">#&gt; [121] quadprog_1.5-8              RBGL_1.78.0                </span></span>
<span><span class="co">#&gt; [123] rappdirs_0.3.3              stringr_1.5.1              </span></span>
<span><span class="co">#&gt; [125] digest_0.6.35               rmarkdown_2.26             </span></span>
<span><span class="co">#&gt; [127] GEOquery_2.70.0             XVector_0.42.0             </span></span>
<span><span class="co">#&gt; [129] htmltools_0.5.7             pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt; [131] base64enc_0.1-3             sparseMatrixStats_1.14.0   </span></span>
<span><span class="co">#&gt; [133] MatrixGenerics_1.14.0       highr_0.10                 </span></span>
<span><span class="co">#&gt; [135] dbplyr_2.5.0                fastmap_1.1.1              </span></span>
<span><span class="co">#&gt; [137] rlang_1.1.3                 DelayedMatrixStats_1.24.0  </span></span>
<span><span class="co">#&gt; [139] jquerylib_0.1.4             jsonlite_1.8.8             </span></span>
<span><span class="co">#&gt; [141] BiocParallel_1.36.0         mclust_6.1                 </span></span>
<span><span class="co">#&gt; [143] RCurl_1.98-1.14             magrittr_2.0.3             </span></span>
<span><span class="co">#&gt; [145] GenomeInfoDbData_1.2.11     Rhdf5lib_1.24.2            </span></span>
<span><span class="co">#&gt; [147] IRkernel_1.3.2              munsell_0.5.0              </span></span>
<span><span class="co">#&gt; [149] Rcpp_1.0.12                 proto_1.0.0                </span></span>
<span><span class="co">#&gt; [151] sqldf_0.4-11                stringi_1.8.3              </span></span>
<span><span class="co">#&gt; [153] RJSONIO_1.3-1.9             zlibbioc_1.48.2            </span></span>
<span><span class="co">#&gt; [155] MASS_7.3-60.0.1             plyr_1.8.9                 </span></span>
<span><span class="co">#&gt; [157] org.Hs.eg.db_3.18.0         bumphunter_1.44.0          </span></span>
<span><span class="co">#&gt; [159] minfi_1.48.0                parallel_4.3.2             </span></span>
<span><span class="co">#&gt; [161] Biostrings_2.70.3           IRdisplay_1.1              </span></span>
<span><span class="co">#&gt; [163] splines_4.3.2               multtest_2.58.0            </span></span>
<span><span class="co">#&gt; [165] hash_2.2.6.3                hms_1.1.3                  </span></span>
<span><span class="co">#&gt; [167] locfit_1.5-9.9              ps_1.7.6                   </span></span>
<span><span class="co">#&gt; [169] uuid_1.2-0                  RUnit_0.4.33               </span></span>
<span><span class="co">#&gt; [171] base64url_1.4               rngtools_1.5.2             </span></span>
<span><span class="co">#&gt; [173] reshape2_1.4.4              biomaRt_2.58.2             </span></span>
<span><span class="co">#&gt; [175] stats4_4.3.2                XML_3.99-0.16.1            </span></span>
<span><span class="co">#&gt; [177] evaluate_0.23               tzdb_0.4.0                 </span></span>
<span><span class="co">#&gt; [179] foreach_1.5.2               tidyr_1.3.1                </span></span>
<span><span class="co">#&gt; [181] openssl_2.1.1               purrr_1.0.2                </span></span>
<span><span class="co">#&gt; [183] reshape_0.8.9               ggplot2_3.5.0              </span></span>
<span><span class="co">#&gt; [185] xtable_1.8-4                restfulr_0.0.15            </span></span>
<span><span class="co">#&gt; [187] ragg_1.3.0                  RCy3_2.23.2                </span></span>
<span><span class="co">#&gt; [189] tibble_3.2.1                memoise_2.0.1              </span></span>
<span><span class="co">#&gt; [191] AnnotationDbi_1.64.1        GenomicAlignments_1.38.2   </span></span>
<span><span class="co">#&gt; [193] IRanges_2.36.0              cluster_2.1.6              </span></span>
<span><span class="co">#&gt; [195] cmdstanr_0.6.1.9000         here_1.0.1                 </span></span>
<span><span class="co">#&gt; [197] GSEABase_1.64.0</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a>
</h3>
<p>If there is an error like
<code>Error in fetch(key) : lazy-load database.rdb' is corrupt</code>
when accessing the help pages of functions in this package after being
loaded. It’s <a href="https://github.com/r-lib/devtools/issues/1660" class="external-link">a
limitation of base R</a> and has not been solved yet. Restart R session
and re-load this package will help.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill, Kate Shutta.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
