<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="netZooR">
<title>Using CONDOR for community detection in bipartite graphs • netZooR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Using CONDOR for community detection in bipartite graphs">
<meta property="og:description" content="netZooR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">netZooR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.17</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    <a class="dropdown-item" href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    <a class="dropdown-item" href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    <a class="dropdown-item" href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    <a class="dropdown-item" href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    <a class="dropdown-item" href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    <a class="dropdown-item" href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    <a class="dropdown-item" href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    <a class="dropdown-item" href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    <a class="dropdown-item" href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/netZoo/netZooR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Using CONDOR for community detection in bipartite graphs</h1>
                        <h4 data-toc-skip class="author">John
Platig</h4>
            
            <h4 data-toc-skip class="date">2015-09-28</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/netZoo/netZooR/blob/HEAD/vignettes/CONDOR.Rmd" class="external-link"><code>vignettes/CONDOR.Rmd</code></a></small>
      <div class="d-none name"><code>CONDOR.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>COmplex Network Description Of Regulators (CONDOR) implements methods
for clustering bipartite networks and estimating the contribution of
each node to its community’s modularity. For an application of this
method to identify diesease-associated single nucleotide polymorphisms,
see (<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005033" class="external-link uri">https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005033</a>).</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>CONDOR can be installed through netZooR as follows:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"netZooR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="implementing-the-bipartite-modularity-maximization">Implementing the Bipartite Modularity Maximization<a class="anchor" aria-label="anchor" href="#implementing-the-bipartite-modularity-maximization"></a>
</h2>
The code in <strong>condorModularityMax</strong> is an implementation of
the method described in Michael Barber’s paper <strong>Modularity and
community detection in bipartite networks</strong> (Phys. Rev. E 76,
066102 (2007)). A few general comments:
</div>
<div class="section level2">
<h2 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/netZoo/netZooR" class="external-link">netZooR</a></span><span class="op">)</span></span></code></pre></div>
<p><strong>condor</strong> works with an edgelist
(<strong>elist</strong> in the code below) as its input.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">r</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">3</span>,<span class="fl">3</span>,<span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span>;</span>
<span><span class="va">b</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">4</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">4</span>,<span class="fl">3</span>,<span class="fl">4</span><span class="op">)</span>;</span>
<span><span class="va">reds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Alice"</span>,<span class="st">"Sue"</span>,<span class="st">"Janine"</span>,<span class="st">"Mary"</span><span class="op">)</span></span>
<span><span class="va">blues</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Bob"</span>,<span class="st">"John"</span>,<span class="st">"Ed"</span>,<span class="st">"Hank"</span><span class="op">)</span></span>
<span><span class="va">elist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>red<span class="op">=</span><span class="va">reds</span><span class="op">[</span><span class="va">r</span><span class="op">]</span>, blue<span class="op">=</span><span class="va">blues</span><span class="op">[</span><span class="va">b</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>In <strong>elist</strong>, notice all nodes of the same type–women
and men in this case–appear in the same column together. This is a
requirement. <strong>createCondorObject</strong> will throw an error if
a node appears in both columns.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">condor.object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createCondorObject.html">createCondorObject</a></span><span class="op">(</span><span class="va">elist</span><span class="op">)</span></span></code></pre></div>
<p>A condor.object is just a list. You can look at the different items
using <strong>names</strong></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">condor.object</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "G"          "edges"      "Qcoms"      "modularity" "red.memb"  </span></span>
<span><span class="co">## [6] "blue.memb"  "qscores"</span></span></code></pre>
<p><strong>condorCluster</strong> will cluster the nodes and produce the
overall modularity along with two community membership
<strong>data.frames</strong>:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">condor.object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/condorCluster.html">condorCluster</a></span><span class="op">(</span><span class="va">condor.object</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "modularity of projected graph 0"</span></span>
<span><span class="co">## [1] "Q = 0"</span></span>
<span><span class="co">## [1] "Q = 0.198347107438017"</span></span>
<span><span class="co">## [1] "Q = 0.231404958677686"</span></span>
<span><span class="co">## [1] "Q = 0.231404958677686"</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">condor.object</span><span class="op">$</span><span class="va">red.memb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   red.names com</span></span>
<span><span class="co">## 1     Alice   1</span></span>
<span><span class="co">## 2    Janine   2</span></span>
<span><span class="co">## 3      Mary   2</span></span>
<span><span class="co">## 4       Sue   1</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">condor.object</span><span class="op">$</span><span class="va">blue.memb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   blue.names com</span></span>
<span><span class="co">## 1        Bob   1</span></span>
<span><span class="co">## 2         Ed   2</span></span>
<span><span class="co">## 3       Hank   2</span></span>
<span><span class="co">## 4       John   1</span></span></code></pre>
<p>Nodes in first community are {Alice, John, Bob, Sue}, nodes in second
community are {Ed, Janine, Hank, Mary} based on the modularity
maximization. Here’s a picture:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gtoy</span> <span class="op">=</span> <span class="fu"><a href="https://r.igraph.org/reference/graph.edgelist.html" class="external-link">graph.edgelist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">elist</span><span class="op">)</span>,directed<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://r.igraph.org/reference/set.graph.attribute.html" class="external-link">set.graph.attribute</a></span><span class="op">(</span><span class="va">gtoy</span>, <span class="st">"layout"</span>, <span class="fu"><a href="https://r.igraph.org/reference/layout.deprecated.html" class="external-link">layout.kamada.kawai</a></span><span class="op">(</span><span class="va">gtoy</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## IGRAPH 5a9c5ad UN-- 8 11 -- </span></span>
<span><span class="co">## + attr: layout (g/n), name (v/c)</span></span>
<span><span class="co">## + edges from 5a9c5ad (vertex names):</span></span>
<span><span class="co">##  [1] Alice--Bob    Alice--John   Alice--Ed     Bob  --Sue    John --Sue   </span></span>
<span><span class="co">##  [6] Sue  --Hank   John --Janine Ed   --Janine Hank --Janine Ed   --Mary  </span></span>
<span><span class="co">## [11] Hank --Mary</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">gtoy</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">reds</span>,<span class="va">blues</span><span class="op">)</span><span class="op">]</span><span class="op">$</span><span class="va">color</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"red"</span>,<span class="fl">4</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"blue"</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gtoy</span>,vertex.label.dist<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="CONDOR_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>To get each node’s modularity contribution (as a fraction of the
community’s modularity), run</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">condor.object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/condorQscore.html">condorQscore</a></span><span class="op">(</span><span class="va">condor.object</span><span class="op">)</span></span></code></pre></div>
<p>If you have a subset of nodes that you think are more likely to lie
at the cores of your communities, you can test this using
<strong>condorCoreEnrich</strong>:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">q_women</span> <span class="op">&lt;-</span> <span class="va">condor.object</span><span class="op">$</span><span class="va">qscores</span><span class="op">$</span><span class="va">red.qscore</span></span>
<span><span class="va">core_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/condorCoreEnrich.html">condorCoreEnrich</a></span><span class="op">(</span>test_nodes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Alice"</span>,<span class="st">"Mary"</span><span class="op">)</span>,</span>
<span>                                 q<span class="op">=</span><span class="va">q_women</span>,perm<span class="op">=</span><span class="cn">TRUE</span>,plot.hist<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="CONDOR_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p><strong>condor</strong> also works on weighted bipartite networks.
The package comes with a quantitative pollination network data set
(Small 1976) taken from the NCEAS interaction webs data base, containing
interactions between 13 plants and 34 pollinators.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">small1976</span><span class="op">)</span></span>
<span><span class="va">condor.object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createCondorObject.html">createCondorObject</a></span><span class="op">(</span><span class="va">small1976</span><span class="op">)</span></span>
<span><span class="va">condor.object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/condorCluster.html">condorCluster</a></span><span class="op">(</span><span class="va">condor.object</span>, project<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "modularity of projected graph 0.525346928655047"</span></span>
<span><span class="co">## [1] "Q = 0.52666696475026"</span></span>
<span><span class="co">## [1] "Q = 0.52666696475026"</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/condorPlotHeatmap.html">condorPlotHeatmap</a></span><span class="op">(</span><span class="va">condor.object</span><span class="op">)</span></span></code></pre></div>
<p><img src="CONDOR_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">## Platform: aarch64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Sonoma 14.1.2</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/New_York</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] netZooR_1.5.17      matrixcalc_1.0-6    pandaR_1.34.0      </span></span>
<span><span class="co">## [4] Biobase_2.62.0      BiocGenerics_0.48.1 reticulate_1.35.0  </span></span>
<span><span class="co">## [7] igraph_2.0.3       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] STRINGdb_2.14.3             fs_1.6.3                   </span></span>
<span><span class="co">##   [3] matrixStats_1.2.0           bitops_1.0-7               </span></span>
<span><span class="co">##   [5] httr_1.4.7                  RColorBrewer_1.1-3         </span></span>
<span><span class="co">##   [7] doParallel_1.0.17           Rgraphviz_2.46.0           </span></span>
<span><span class="co">##   [9] repr_1.1.7                  tools_4.3.2                </span></span>
<span><span class="co">##  [11] doRNG_1.8.6                 backports_1.4.1            </span></span>
<span><span class="co">##  [13] utf8_1.2.4                  R6_2.5.1                   </span></span>
<span><span class="co">##  [15] vegan_2.6-4                 HDF5Array_1.30.1           </span></span>
<span><span class="co">##  [17] mgcv_1.9-1                  rhdf5filters_1.14.1        </span></span>
<span><span class="co">##  [19] permute_0.9-7               prettyunits_1.2.0          </span></span>
<span><span class="co">##  [21] base64_2.0.1                preprocessCore_1.64.0      </span></span>
<span><span class="co">##  [23] cli_3.6.2                   textshaping_0.3.7          </span></span>
<span><span class="co">##  [25] penalized_0.9-52            sass_0.4.9                 </span></span>
<span><span class="co">##  [27] readr_2.1.5                 genefilter_1.84.0          </span></span>
<span><span class="co">##  [29] askpass_1.2.0               pkgdown_2.0.7              </span></span>
<span><span class="co">##  [31] Rsamtools_2.18.0            systemfonts_1.0.6          </span></span>
<span><span class="co">##  [33] pbdZMQ_0.3-11               siggenes_1.76.0            </span></span>
<span><span class="co">##  [35] illuminaio_0.44.0           AnnotationForge_1.44.0     </span></span>
<span><span class="co">##  [37] scrime_1.3.5                plotrix_3.8-4              </span></span>
<span><span class="co">##  [39] limma_3.58.1                rstudioapi_0.16.0          </span></span>
<span><span class="co">##  [41] RSQLite_2.3.5               generics_0.1.3             </span></span>
<span><span class="co">##  [43] GOstats_2.68.0              quantro_1.36.0             </span></span>
<span><span class="co">##  [45] BiocIO_1.12.0               gtools_3.9.5               </span></span>
<span><span class="co">##  [47] distributional_0.4.0        dplyr_1.1.4                </span></span>
<span><span class="co">##  [49] GO.db_3.18.0                Matrix_1.6-5               </span></span>
<span><span class="co">##  [51] fansi_1.0.6                 S4Vectors_0.40.2           </span></span>
<span><span class="co">##  [53] abind_1.4-5                 lifecycle_1.0.4            </span></span>
<span><span class="co">##  [55] yaml_2.3.8                  edgeR_4.0.16               </span></span>
<span><span class="co">##  [57] SummarizedExperiment_1.32.0 gplots_3.1.3.1             </span></span>
<span><span class="co">##  [59] rhdf5_2.46.1                SparseArray_1.2.4          </span></span>
<span><span class="co">##  [61] BiocFileCache_2.10.1        grid_4.3.2                 </span></span>
<span><span class="co">##  [63] blob_1.2.4                  crayon_1.5.2               </span></span>
<span><span class="co">##  [65] lattice_0.22-6              GenomicFeatures_1.54.4     </span></span>
<span><span class="co">##  [67] annotate_1.80.0             KEGGREST_1.42.0            </span></span>
<span><span class="co">##  [69] pillar_1.9.0                knitr_1.45                 </span></span>
<span><span class="co">##  [71] beanplot_1.3.1              GenomicRanges_1.54.1       </span></span>
<span><span class="co">##  [73] rjson_0.2.21                codetools_0.2-19           </span></span>
<span><span class="co">##  [75] glue_1.7.0                  downloader_0.4             </span></span>
<span><span class="co">##  [77] data.table_1.15.2           vctrs_0.6.5                </span></span>
<span><span class="co">##  [79] png_0.1-8                   gtable_0.3.4               </span></span>
<span><span class="co">##  [81] assertthat_0.2.1            gsubfn_0.7                 </span></span>
<span><span class="co">##  [83] cachem_1.0.8                xfun_0.42                  </span></span>
<span><span class="co">##  [85] S4Arrays_1.2.1              survival_3.5-8             </span></span>
<span><span class="co">##  [87] iterators_1.0.14            statmod_1.5.0              </span></span>
<span><span class="co">##  [89] nlme_3.1-164                Category_2.68.0            </span></span>
<span><span class="co">##  [91] bit64_4.0.5                 progress_1.2.3             </span></span>
<span><span class="co">##  [93] filelock_1.0.3              GenomeInfoDb_1.38.8        </span></span>
<span><span class="co">##  [95] tensorA_0.36.2.1            bslib_0.6.2                </span></span>
<span><span class="co">##  [97] nor1mix_1.3-2               KernSmooth_2.23-22         </span></span>
<span><span class="co">##  [99] colorspace_2.1-0            DBI_1.2.2                  </span></span>
<span><span class="co">## [101] nnet_7.3-19                 processx_3.8.4             </span></span>
<span><span class="co">## [103] tidyselect_1.2.1            bit_4.0.5                  </span></span>
<span><span class="co">## [105] compiler_4.3.2              curl_5.2.1                 </span></span>
<span><span class="co">## [107] chron_2.3-61                graph_1.80.0               </span></span>
<span><span class="co">## [109] xml2_1.3.6                  desc_1.4.3                 </span></span>
<span><span class="co">## [111] ggdendro_0.2.0              DelayedArray_0.28.0        </span></span>
<span><span class="co">## [113] posterior_1.5.0             rtracklayer_1.62.0         </span></span>
<span><span class="co">## [115] checkmate_2.3.1             scales_1.3.0               </span></span>
<span><span class="co">## [117] caTools_1.18.2              hexbin_1.28.3              </span></span>
<span><span class="co">## [119] quadprog_1.5-8              RBGL_1.78.0                </span></span>
<span><span class="co">## [121] rappdirs_0.3.3              stringr_1.5.1              </span></span>
<span><span class="co">## [123] digest_0.6.35               rmarkdown_2.26             </span></span>
<span><span class="co">## [125] GEOquery_2.70.0             XVector_0.42.0             </span></span>
<span><span class="co">## [127] htmltools_0.5.7             pkgconfig_2.0.3            </span></span>
<span><span class="co">## [129] base64enc_0.1-3             sparseMatrixStats_1.14.0   </span></span>
<span><span class="co">## [131] MatrixGenerics_1.14.0       highr_0.10                 </span></span>
<span><span class="co">## [133] dbplyr_2.5.0                fastmap_1.1.1              </span></span>
<span><span class="co">## [135] rlang_1.1.3                 DelayedMatrixStats_1.24.0  </span></span>
<span><span class="co">## [137] jquerylib_0.1.4             jsonlite_1.8.8             </span></span>
<span><span class="co">## [139] BiocParallel_1.36.0         mclust_6.1                 </span></span>
<span><span class="co">## [141] RCurl_1.98-1.14             magrittr_2.0.3             </span></span>
<span><span class="co">## [143] GenomeInfoDbData_1.2.11     Rhdf5lib_1.24.2            </span></span>
<span><span class="co">## [145] IRkernel_1.3.2              munsell_0.5.0              </span></span>
<span><span class="co">## [147] Rcpp_1.0.12                 proto_1.0.0                </span></span>
<span><span class="co">## [149] sqldf_0.4-11                stringi_1.8.3              </span></span>
<span><span class="co">## [151] RJSONIO_1.3-1.9             zlibbioc_1.48.2            </span></span>
<span><span class="co">## [153] MASS_7.3-60.0.1             plyr_1.8.9                 </span></span>
<span><span class="co">## [155] org.Hs.eg.db_3.18.0         bumphunter_1.44.0          </span></span>
<span><span class="co">## [157] minfi_1.48.0                parallel_4.3.2             </span></span>
<span><span class="co">## [159] Biostrings_2.70.3           IRdisplay_1.1              </span></span>
<span><span class="co">## [161] splines_4.3.2               multtest_2.58.0            </span></span>
<span><span class="co">## [163] hash_2.2.6.3                hms_1.1.3                  </span></span>
<span><span class="co">## [165] locfit_1.5-9.9              ps_1.7.6                   </span></span>
<span><span class="co">## [167] uuid_1.2-0                  RUnit_0.4.33               </span></span>
<span><span class="co">## [169] base64url_1.4               rngtools_1.5.2             </span></span>
<span><span class="co">## [171] reshape2_1.4.4              biomaRt_2.58.2             </span></span>
<span><span class="co">## [173] stats4_4.3.2                XML_3.99-0.16.1            </span></span>
<span><span class="co">## [175] evaluate_0.23               tzdb_0.4.0                 </span></span>
<span><span class="co">## [177] foreach_1.5.2               tidyr_1.3.1                </span></span>
<span><span class="co">## [179] openssl_2.1.1               purrr_1.0.2                </span></span>
<span><span class="co">## [181] reshape_0.8.9               ggplot2_3.5.0              </span></span>
<span><span class="co">## [183] xtable_1.8-4                restfulr_0.0.15            </span></span>
<span><span class="co">## [185] viridisLite_0.4.2           ragg_1.3.0                 </span></span>
<span><span class="co">## [187] RCy3_2.23.2                 tibble_3.2.1               </span></span>
<span><span class="co">## [189] memoise_2.0.1               AnnotationDbi_1.64.1       </span></span>
<span><span class="co">## [191] GenomicAlignments_1.38.2    IRanges_2.36.0             </span></span>
<span><span class="co">## [193] cluster_2.1.6               cmdstanr_0.6.1.9000        </span></span>
<span><span class="co">## [195] GSEABase_1.64.0</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill, Kate Shutta.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
