<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="netZooR">
<title>Comparing LIONESS Regulatory Networks using limma • netZooR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Comparing LIONESS Regulatory Networks using limma">
<meta property="og:description" content="netZooR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">netZooR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.17</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    <a class="dropdown-item" href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    <a class="dropdown-item" href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    <a class="dropdown-item" href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    <a class="dropdown-item" href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    <a class="dropdown-item" href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    <a class="dropdown-item" href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    <a class="dropdown-item" href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    <a class="dropdown-item" href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    <a class="dropdown-item" href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/netZoo/netZooR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Comparing LIONESS Regulatory Networks using limma</h1>
                        <h4 data-toc-skip class="author">Camila
Lopes-Ramos</h4>
            
            <h4 data-toc-skip class="date">2020-10-19</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/netZoo/netZooR/blob/HEAD/vignettes/LionessApplicationinGTExData.Rmd" class="external-link"><code>vignettes/LionessApplicationinGTExData.Rmd</code></a></small>
      <div class="d-none name"><code>LionessApplicationinGTExData.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>LIONESS (Linear Interpolation to Obtain Network Estimates for Single
Samples) is a method for creating sample-specific networks. When applied
to a PANDA regulatory network, the result is a set of gene regulatory
networks, one for each sample in the gene expression dataset. More
information on LIONESS can be found in the published paper: <a href="https://doi.org/10.1016/j.isci.2019.03.021" class="external-link uri">https://doi.org/10.1016/j.isci.2019.03.021</a></p>
<p>In this vignette, we will compare LIONESS regulatory networks from
207 females and 238 males with colon cancer using RNA-Seq data from
TCGA. We will compare the edge weights between females and males using a
linear regression model and correcting for the covariates age, race, and
disease stage, as available in the limma package. We will also compare
the gene’s in-degree (defined as the sum of the gene’s incoming edge
weights from all TFs in the network). Finally, we will perform gene set
enrichment analysis to find the pathways enriched for genes
differentially targeted by sex in colon cancer.</p>
<div class="section level3">
<h3 id="install-packages">Install packages<a class="anchor" aria-label="anchor" href="#install-packages"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>   </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span>,repos <span class="op">=</span> <span class="st">"http://cran.us.r-project.org"</span><span class="op">)</span>  </span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"limma"</span><span class="op">)</span>   </span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"fgsea"</span><span class="op">)</span>   </span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"Biobase"</span><span class="op">)</span> </span>
<span><span class="co"># install.packages("ggplot2",repos = "http://cran.us.r-project.org")    </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"igraph"</span>,repos <span class="op">=</span> <span class="st">"http://cran.us.r-project.org"</span><span class="op">)</span>   </span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/limma/" class="external-link">limma</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ctlab/fgsea/" class="external-link">fgsea</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biobase" class="external-link">Biobase</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="load-the-data">Load the data<a class="anchor" aria-label="anchor" href="#load-the-data"></a>
</h2>
<p>For the purposes of demonstrating the workflow, we will load only a
subet of LIONESS networks. Our subset shows the edge weights for 50,000
edges (rows) by 445 samples (columns). Let’s take a look at the
networks:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download a subset of LIONESS networks from netZooR AWS Bucket to working directory.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.html" class="external-link">system</a></span><span class="op">(</span><span class="st">"curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/lioness_coloncancer_subset.txt"</span><span class="op">)</span></span>
<span><span class="va">lioness</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="st">"./lioness_coloncancer_subset.txt"</span>,stringsAsFactors <span class="op">=</span> <span class="cn">F</span>, check.names <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lioness</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;       TF  gene TCGA-AY-A71X-01A-12R-A37K-07 TCGA-A6-4105-01A-02R-1774-07</span></span>
<span><span class="co">#&gt; 1  FOXD2 MGST3                       2.3076                       3.2365</span></span>
<span><span class="co">#&gt; 2  MEF2A MGST3                       1.8015                       3.2350</span></span>
<span><span class="co">#&gt; 3    TBP MGST3                       2.2111                       3.1582</span></span>
<span><span class="co">#&gt; 4  TBPL2 GSTA4                       2.2128                       4.5202</span></span>
<span><span class="co">#&gt; 5 POU3F3 MGST2                       4.2174                       5.3111</span></span>
<span><span class="co">#&gt; 6 HMG20B GSTA4                       2.5219                       4.4398</span></span>
<span><span class="co">#&gt;   TCGA-AA-3526-01A-02R-A32Z-07</span></span>
<span><span class="co">#&gt; 1                       3.2611</span></span>
<span><span class="co">#&gt; 2                       3.2992</span></span>
<span><span class="co">#&gt; 3                       3.3390</span></span>
<span><span class="co">#&gt; 4                       4.4191</span></span>
<span><span class="co">#&gt; 5                       5.4973</span></span>
<span><span class="co">#&gt; 6                       4.7402</span></span>
<span><span class="co"># Add row names as "TF_gene" and remove columns 1 and 2 with TF and gene name</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">lioness</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">lioness</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="va">x</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="co"># Remove TF and gene columns</span></span>
<span><span class="va">lioness</span> <span class="op">&lt;-</span> <span class="va">lioness</span><span class="op">[</span>,<span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lioness</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;              TCGA-AY-A71X-01A-12R-A37K-07 TCGA-A6-4105-01A-02R-1774-07</span></span>
<span><span class="co">#&gt; FOXD2_MGST3                        2.3076                       3.2365</span></span>
<span><span class="co">#&gt; MEF2A_MGST3                        1.8015                       3.2350</span></span>
<span><span class="co">#&gt; TBP_MGST3                          2.2111                       3.1582</span></span>
<span><span class="co">#&gt; TBPL2_GSTA4                        2.2128                       4.5202</span></span>
<span><span class="co">#&gt; POU3F3_MGST2                       4.2174                       5.3111</span></span>
<span><span class="co">#&gt; HMG20B_GSTA4                       2.5219                       4.4398</span></span>
<span><span class="co">#&gt;              TCGA-AA-3526-01A-02R-A32Z-07 TCGA-F4-6463-01A-11R-1723-07</span></span>
<span><span class="co">#&gt; FOXD2_MGST3                        3.2611                       2.7414</span></span>
<span><span class="co">#&gt; MEF2A_MGST3                        3.2992                       2.7100</span></span>
<span><span class="co">#&gt; TBP_MGST3                          3.3390                       2.6938</span></span>
<span><span class="co">#&gt; TBPL2_GSTA4                        4.4191                       4.6907</span></span>
<span><span class="co">#&gt; POU3F3_MGST2                       5.4973                       5.0715</span></span>
<span><span class="co">#&gt; HMG20B_GSTA4                       4.7402                       4.6494</span></span>
<span><span class="co">#&gt;              TCGA-A6-5665-01A-01R-1653-07</span></span>
<span><span class="co">#&gt; FOXD2_MGST3                       -2.5282</span></span>
<span><span class="co">#&gt; MEF2A_MGST3                       -1.7591</span></span>
<span><span class="co">#&gt; TBP_MGST3                         -1.3487</span></span>
<span><span class="co">#&gt; TBPL2_GSTA4                        4.1184</span></span>
<span><span class="co">#&gt; POU3F3_MGST2                       1.5492</span></span>
<span><span class="co">#&gt; HMG20B_GSTA4                       4.1758</span></span>
<span><span class="co"># Load the complete gene in-degree (sum of all edge weights for each gene) and clinical data stored as an expression set</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.html" class="external-link">system</a></span><span class="op">(</span><span class="st">"curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/inDegree_allEdges_coloncancer.rdata"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"./inDegree_allEdges_coloncancer.rdata"</span><span class="op">)</span></span>
<span><span class="co"># Clinical information</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">obj1</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">30</span><span class="op">:</span><span class="fl">35</span><span class="op">]</span></span>
<span><span class="co">#&gt;                              gender birth_days_to      race</span></span>
<span><span class="co">#&gt; TCGA-AY-A71X-01A-12R-A37K-07 FEMALE        -19904     black</span></span>
<span><span class="co">#&gt; TCGA-A6-4105-01A-02R-1774-07   MALE        -29097 caucasian</span></span>
<span><span class="co">#&gt; TCGA-AA-3526-01A-02R-A32Z-07   MALE        -21123      &lt;NA&gt;</span></span>
<span><span class="co">#&gt; TCGA-F4-6463-01A-11R-1723-07   MALE        -18804 caucasian</span></span>
<span><span class="co">#&gt; TCGA-A6-5665-01A-01R-1653-07 FEMALE        -30915 caucasian</span></span>
<span><span class="co">#&gt;                                           ethnicity history_other_malignancy</span></span>
<span><span class="co">#&gt; TCGA-AY-A71X-01A-12R-A37K-07 NOT HISPANIC OR LATINO                       No</span></span>
<span><span class="co">#&gt; TCGA-A6-4105-01A-02R-1774-07 NOT HISPANIC OR LATINO                      Yes</span></span>
<span><span class="co">#&gt; TCGA-AA-3526-01A-02R-A32Z-07        [Not Available]                       No</span></span>
<span><span class="co">#&gt; TCGA-F4-6463-01A-11R-1723-07 NOT HISPANIC OR LATINO                       No</span></span>
<span><span class="co">#&gt; TCGA-A6-5665-01A-01R-1653-07 NOT HISPANIC OR LATINO                       No</span></span>
<span><span class="co">#&gt;                              history_neoadjuvant_treatment</span></span>
<span><span class="co">#&gt; TCGA-AY-A71X-01A-12R-A37K-07                            No</span></span>
<span><span class="co">#&gt; TCGA-A6-4105-01A-02R-1774-07                            No</span></span>
<span><span class="co">#&gt; TCGA-AA-3526-01A-02R-A32Z-07                            No</span></span>
<span><span class="co">#&gt; TCGA-F4-6463-01A-11R-1723-07                            No</span></span>
<span><span class="co">#&gt; TCGA-A6-5665-01A-01R-1653-07                            No</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="compare-the-edge-weights">Compare the edge weights<a class="anchor" aria-label="anchor" href="#compare-the-edge-weights"></a>
</h2>
<p>Compare the edge weights between males and females using linear
regression model (limma package) and adjusting for covariates: stage,
age, race.</p>
<div class="section level3">
<h3 id="run-limma">Run limma<a class="anchor" aria-label="anchor" href="#run-limma"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define the covariates</span></span>
<span><span class="va">gender</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">obj1</span><span class="op">)</span><span class="op">$</span><span class="va">gender</span><span class="op">)</span>,levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MALE"</span>,<span class="st">"FEMALE"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">stage</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">obj1</span><span class="op">)</span><span class="op">$</span><span class="va">uicc_stage</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">stage</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">stage</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"NA"</span>    </span>
<span><span class="va">stage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">stage</span><span class="op">)</span></span>
<span><span class="va">race</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">obj1</span><span class="op">)</span><span class="op">$</span><span class="va">race</span><span class="op">)</span></span>
<span><span class="va">race</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">race</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"NA"</span></span>
<span><span class="va">race</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">race</span><span class="op">)</span></span>
<span><span class="va">age</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">obj1</span><span class="op">)</span><span class="op">$</span><span class="va">age_at_initial_pathologic_diagnosis</span><span class="op">)</span></span>
<span><span class="va">age</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">age</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">age</span>,na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">design</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">stage</span> <span class="op">+</span> <span class="va">race</span> <span class="op">+</span> <span class="va">age</span> <span class="op">+</span> <span class="va">gender</span><span class="op">)</span></span>
<span><span class="co"># Run limma</span></span>
<span><span class="va">fitGood</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">lioness</span><span class="op">)</span>,<span class="va">design</span><span class="op">)</span></span>
<span><span class="va">fitGood</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fitGood</span><span class="op">)</span></span>
<span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">fitGood</span>,coef<span class="op">=</span><span class="st">"genderFEMALE"</span>,number<span class="op">=</span><span class="cn">Inf</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span>
<span><span class="co">#&gt;                     logFC   AveExpr         t      P.Value  adj.P.Val        B</span></span>
<span><span class="co">#&gt; NKX2-5_SLCO3A1 -1.0747338 2.7091520 -4.871484 1.552517e-06 0.04199842 4.751052</span></span>
<span><span class="co">#&gt; SOX12_SLCO3A1  -2.0624528 0.9997061 -4.801388 2.169196e-06 0.04199842 4.456929</span></span>
<span><span class="co">#&gt; TBPL2_SLCO3A1  -2.6081364 0.2806817 -4.708294 3.361432e-06 0.04199842 4.072181</span></span>
<span><span class="co">#&gt; ZNF232_KBTBD2   0.9843244 4.3091270  4.646466 4.478727e-06 0.04199842 3.820378</span></span>
<span><span class="co">#&gt; NFATC3_PORCN   -0.6981933 1.8005115 -4.637017 4.678219e-06 0.04199842 3.782159</span></span>
<span><span class="co">#&gt; FOXO3_PDS5B     0.5946457 4.6099430  4.572347 6.291516e-06 0.04199842 3.522461</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualize-the-top-edges-with-differential-weights-by-sex">Visualize the top edges with differential weights by sex<a class="anchor" aria-label="anchor" href="#visualize-the-top-edges-with-differential-weights-by-sex"></a>
</h3>
<p>We select the top 50 edges with differential edge weights by sex and
convert them into an igraph graph.data.frame object for visualization.
We color edges red if they have higher coefficients in the female group,
and blue if they have higher coefficients in the male group.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">toptable_edges</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span>, <span class="st">"_"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">z</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">toptable_edges</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span>,<span class="op">]</span>, <span class="va">tb</span><span class="op">$</span><span class="va">logFC</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/graph.data.frame.html" class="external-link">graph.data.frame</a></span><span class="op">(</span><span class="va">z</span>, directed<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: `graph.data.frame()` was deprecated in igraph 2.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `graph_from_data_frame()` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="fu"><a href="https://r.igraph.org/reference/E.html" class="external-link">E</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">weight</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">z</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://r.igraph.org/reference/E.html" class="external-link">E</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">color</span><span class="op">[</span><span class="fu"><a href="https://r.igraph.org/reference/E.html" class="external-link">E</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">weight</span><span class="op">&lt;</span><span class="fl">0</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"blue"</span></span>
<span><span class="fu"><a href="https://r.igraph.org/reference/E.html" class="external-link">E</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">color</span><span class="op">[</span><span class="fu"><a href="https://r.igraph.org/reference/E.html" class="external-link">E</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">weight</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"red"</span></span>
<span><span class="fu"><a href="https://r.igraph.org/reference/E.html" class="external-link">E</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">weight</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span>,<span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">g</span>, vertex.label.cex<span class="op">=</span><span class="fl">0.7</span>, vertex.size<span class="op">=</span><span class="fl">10</span>,  vertex.label.font<span class="op">=</span><span class="fl">3</span>, edge.width<span class="op">=</span><span class="fl">5</span><span class="op">*</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">z</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="LionessApplicationinGTExData_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="compare-the-gene-in-degree">Compare the gene in-degree<a class="anchor" aria-label="anchor" href="#compare-the-gene-in-degree"></a>
</h2>
<p>Compare the gene in-degree between males and females using linear
regression model (limma package) and adjusting for covariates: stage,
age, race.</p>
<div class="section level3">
<h3 id="run-limma-1">Run limma<a class="anchor" aria-label="anchor" href="#run-limma-1"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">indegree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/assayData.html" class="external-link">assayData</a></span><span class="op">(</span><span class="va">obj1</span><span class="op">)</span><span class="op">[[</span><span class="st">"quantile"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">indegree</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;        TCGA-AY-A71X-01A-12R-A37K-07 TCGA-A6-4105-01A-02R-1774-07</span></span>
<span><span class="co">#&gt; A1BG                      -300.7761                   -309.57401</span></span>
<span><span class="co">#&gt; A1CF                       129.2204                     37.11690</span></span>
<span><span class="co">#&gt; A2M                        447.1353                    418.45826</span></span>
<span><span class="co">#&gt; A4GALT                    -215.1396                   -155.53291</span></span>
<span><span class="co">#&gt; AAAS                      -342.6264                   -344.35561</span></span>
<span><span class="co">#&gt; AACS                       369.7163                     22.04239</span></span>
<span><span class="co">#&gt;        TCGA-AA-3526-01A-02R-A32Z-07</span></span>
<span><span class="co">#&gt; A1BG                     -321.71607</span></span>
<span><span class="co">#&gt; A1CF                       14.00825</span></span>
<span><span class="co">#&gt; A2M                       397.25374</span></span>
<span><span class="co">#&gt; A4GALT                   -191.21809</span></span>
<span><span class="co">#&gt; AAAS                     -332.18513</span></span>
<span><span class="co">#&gt; AACS                      485.25668</span></span>
<span><span class="co"># Use the same design matrix as before</span></span>
<span><span class="va">fitGood</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">indegree</span>,<span class="va">design</span><span class="op">)</span></span>
<span><span class="va">fitGood</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fitGood</span><span class="op">)</span></span>
<span><span class="va">tb_degree</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">fitGood</span>,coef<span class="op">=</span><span class="st">"genderFEMALE"</span>,number<span class="op">=</span><span class="cn">Inf</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tb_degree</span><span class="op">)</span></span>
<span><span class="co">#&gt;             logFC   AveExpr         t      P.Value   adj.P.Val         B</span></span>
<span><span class="co">#&gt; OTUD4   -77.70043  306.3134 -5.223489 2.716224e-07 0.003481384 -4.225218</span></span>
<span><span class="co">#&gt; USP32   -86.32333  468.3938 -4.191101 3.357367e-05 0.089161433 -4.357195</span></span>
<span><span class="co">#&gt; MARCH8 -107.59057  183.7655 -4.172733 3.628637e-05 0.089161433 -4.359326</span></span>
<span><span class="co">#&gt; POLR2G   68.22211 -339.4559  4.089615 5.139259e-05 0.089161433 -4.368867</span></span>
<span><span class="co">#&gt; TBL1X  -171.12318  206.8917 -4.040868 6.285937e-05 0.089161433 -4.374386</span></span>
<span><span class="co">#&gt; GUK1     58.76388 -228.4243  4.030137 6.569152e-05 0.089161433 -4.375593</span></span>
<span><span class="co"># Save gene ranks</span></span>
<span><span class="va">indegree_rank</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span>object<span class="op">=</span><span class="va">tb_degree</span><span class="op">[</span>,<span class="st">"t"</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tb_degree</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="gene-set-enrichment-analysis">Gene Set Enrichment Analysis<a class="anchor" aria-label="anchor" href="#gene-set-enrichment-analysis"></a>
</h2>
<p>Well will use the fgsea package to perform gene set enrichment
analysis. We need to point to a ranked gene list (for example the gene
in-degree statistical difference (t value) between males and females),
and a list of gene sets (or signatures) in gmt format to test for
enrichment. The gene sets can be downloaded from MSigDB: <a href="http://software.broadinstitute.org/gsea/msigdb" class="external-link uri">http://software.broadinstitute.org/gsea/msigdb</a> Same gene
annotation should be used in the ranked gene list and gene sets.</p>
<div class="section level3">
<h3 id="run-fgsea">Run fgsea<a class="anchor" aria-label="anchor" href="#run-fgsea"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/system.html" class="external-link">system</a></span><span class="op">(</span><span class="st">"curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/c2.cp.kegg.v7.0.symbols.gmt"</span><span class="op">)</span></span>
<span><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/fgsea/man/gmtPathways.html" class="external-link">gmtPathways</a></span><span class="op">(</span><span class="st">"./c2.cp.kegg.v7.0.symbols.gmt"</span><span class="op">)</span></span>
<span><span class="va">fgseaRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/fgsea/man/fgsea.html" class="external-link">fgsea</a></span><span class="op">(</span><span class="va">pathways</span>, <span class="va">indegree_rank</span>, minSize<span class="op">=</span><span class="fl">15</span>, maxSize<span class="op">=</span><span class="fl">500</span>, nperm<span class="op">=</span><span class="fl">1000</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">fgseaRes</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                 pathway       pval      padj         ES</span></span>
<span><span class="co">#&gt;                                  &lt;char&gt;      &lt;num&gt;     &lt;num&gt;      &lt;num&gt;</span></span>
<span><span class="co">#&gt; 1:      KEGG_GLYCOLYSIS_GLUCONEOGENESIS 0.14011516 0.2466531  0.3240518</span></span>
<span><span class="co">#&gt; 2:         KEGG_CITRATE_CYCLE_TCA_CYCLE 0.52361396 0.6611802 -0.2702159</span></span>
<span><span class="co">#&gt; 3:       KEGG_PENTOSE_PHOSPHATE_PATHWAY 0.19771863 0.3207884  0.3795079</span></span>
<span><span class="co">#&gt; 4: KEGG_FRUCTOSE_AND_MANNOSE_METABOLISM 0.04313725 0.1004670  0.4200039</span></span>
<span><span class="co">#&gt; 5:            KEGG_GALACTOSE_METABOLISM 0.87578947 0.9519166 -0.2250237</span></span>
<span><span class="co">#&gt; 6:           KEGG_FATTY_ACID_METABOLISM 0.14426877 0.2466531  0.3533304</span></span>
<span><span class="co">#&gt;           NES nMoreExtreme  size  leadingEdge</span></span>
<span><span class="co">#&gt;         &lt;num&gt;        &lt;num&gt; &lt;int&gt;       &lt;list&gt;</span></span>
<span><span class="co">#&gt; 1:  1.2575584           72    47 ALDOB, T....</span></span>
<span><span class="co">#&gt; 2: -0.9618572          254    29 SDHA, CS....</span></span>
<span><span class="co">#&gt; 3:  1.2086535          103    21 ALDOB, D....</span></span>
<span><span class="co">#&gt; 4:  1.4564413           21    30 ALDOB, T....</span></span>
<span><span class="co">#&gt; 5: -0.7174393          415    20 HK1, GLA....</span></span>
<span><span class="co">#&gt; 6:  1.2553780           72    33 GCDH, AC....</span></span>
<span><span class="co"># Subset to pathways with FDR &lt; 0.05</span></span>
<span><span class="va">sig</span> <span class="op">&lt;-</span> <span class="va">fgseaRes</span><span class="op">[</span><span class="va">fgseaRes</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span>,<span class="op">]</span></span>
<span><span class="co"># Top 10 pathways enriched in females</span></span>
<span><span class="va">sig</span><span class="op">$</span><span class="va">pathway</span><span class="op">[</span><span class="va">sig</span><span class="op">$</span><span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span><span class="co">#&gt;  [1] "KEGG_STEROID_HORMONE_BIOSYNTHESIS"                </span></span>
<span><span class="co">#&gt;  [2] "KEGG_OXIDATIVE_PHOSPHORYLATION"                   </span></span>
<span><span class="co">#&gt;  [3] "KEGG_PYRIMIDINE_METABOLISM"                       </span></span>
<span><span class="co">#&gt;  [4] "KEGG_TERPENOID_BACKBONE_BIOSYNTHESIS"             </span></span>
<span><span class="co">#&gt;  [5] "KEGG_METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450"</span></span>
<span><span class="co">#&gt;  [6] "KEGG_DRUG_METABOLISM_CYTOCHROME_P450"             </span></span>
<span><span class="co">#&gt;  [7] "KEGG_RIBOSOME"                                    </span></span>
<span><span class="co">#&gt;  [8] "KEGG_PROTEASOME"                                  </span></span>
<span><span class="co">#&gt;  [9] "KEGG_PROTEIN_EXPORT"                              </span></span>
<span><span class="co">#&gt; [10] "KEGG_PEROXISOME"</span></span>
<span><span class="co"># Top 10 pathways enriched in males</span></span>
<span><span class="va">sig</span><span class="op">$</span><span class="va">pathway</span><span class="op">[</span><span class="va">sig</span><span class="op">$</span><span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span><span class="co">#&gt;  [1] "KEGG_INOSITOL_PHOSPHATE_METABOLISM"        </span></span>
<span><span class="co">#&gt;  [2] "KEGG_ABC_TRANSPORTERS"                     </span></span>
<span><span class="co">#&gt;  [3] "KEGG_MAPK_SIGNALING_PATHWAY"               </span></span>
<span><span class="co">#&gt;  [4] "KEGG_ERBB_SIGNALING_PATHWAY"               </span></span>
<span><span class="co">#&gt;  [5] "KEGG_CHEMOKINE_SIGNALING_PATHWAY"          </span></span>
<span><span class="co">#&gt;  [6] "KEGG_PHOSPHATIDYLINOSITOL_SIGNALING_SYSTEM"</span></span>
<span><span class="co">#&gt;  [7] "KEGG_MTOR_SIGNALING_PATHWAY"               </span></span>
<span><span class="co">#&gt;  [8] "KEGG_WNT_SIGNALING_PATHWAY"                </span></span>
<span><span class="co">#&gt;  [9] "KEGG_NOTCH_SIGNALING_PATHWAY"              </span></span>
<span><span class="co">#&gt; [10] "KEGG_TGF_BETA_SIGNALING_PATHWAY"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="bubble-plot-of-differentially-targeted-pathways-by-sex">Bubble plot of differentially targeted pathways by sex<a class="anchor" aria-label="anchor" href="#bubble-plot-of-differentially-targeted-pathways-by-sex"></a>
</h3>
<p>Bubble plot of gene sets on y-axis and adjusted p-value (padj) on
x-axis. Bubble size indicates the number of genes in each gene set, and
bubble color indicates the normalized enrichment score (NES). Blue is
for negative NES (enrichment of higher targeted genes in males), and red
is for positive NES (enrichment of higher targeted genes in
females).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">fgseaRes</span><span class="op">)</span></span>
<span><span class="co"># Settings</span></span>
<span><span class="va">fdrcut</span> <span class="op">&lt;-</span> <span class="fl">0.05</span> <span class="co"># FDR cut-off to use as output for significant signatures</span></span>
<span><span class="va">dencol_neg</span> <span class="op">&lt;-</span> <span class="st">"blue"</span> <span class="co"># bubble plot color for negative ES</span></span>
<span><span class="va">dencol_pos</span> <span class="op">&lt;-</span> <span class="st">"red"</span> <span class="co"># bubble plot color for positive ES</span></span>
<span><span class="va">signnamelength</span> <span class="op">&lt;-</span> <span class="fl">4</span> <span class="co"># set to remove prefix from signature names (2 for "GO", 4 for "KEGG", 8 for "REACTOME")</span></span>
<span><span class="va">asp</span> <span class="op">&lt;-</span> <span class="fl">3</span> <span class="co"># aspect ratio of bubble plot</span></span>
<span><span class="va">charcut</span> <span class="op">&lt;-</span> <span class="fl">100</span> <span class="co"># cut signature name in heatmap to this nr of characters</span></span>
<span><span class="co"># Make signature names more readable</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">dat</span><span class="op">$</span><span class="va">pathway</span><span class="op">)</span> <span class="co"># 'a' is a great variable name to substitute row names with something more readable</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">a</span><span class="op">[</span><span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="va">a</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>, <span class="va">signnamelength</span><span class="op">+</span><span class="fl">2</span>, <span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nchar</a></span><span class="op">(</span><span class="va">a</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">tolower</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span> <span class="co"># convert to lower case (you may want to comment this out, it really depends on what signatures you are looking at, c6 signatures contain gene names, and converting those to lower case may be confusing)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nchar</a></span><span class="op">(</span><span class="va">a</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">)</span><span class="op">&gt;</span><span class="va">charcut</span><span class="op">)</span> <span class="op">{</span> <span class="va">a</span><span class="op">[</span><span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="va">a</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>, <span class="fl">1</span>, <span class="va">charcut</span><span class="op">)</span>, <span class="st">"..."</span>, sep<span class="op">=</span><span class="st">" "</span><span class="op">)</span><span class="op">}</span></span>
<span><span class="op">}</span> <span class="co"># cut signature names that have more characters than charcut, and add "..."</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"_"</span>, <span class="st">" "</span>, <span class="va">a</span><span class="op">)</span></span>
<span><span class="va">dat</span><span class="op">$</span><span class="va">NAME</span> <span class="op">&lt;-</span> <span class="va">a</span></span>
<span><span class="co"># Determine what signatures to plot (based on FDR cut)</span></span>
<span><span class="va">dat2</span> <span class="op">&lt;-</span> <span class="va">dat</span><span class="op">[</span><span class="va">dat</span><span class="op">[</span>,<span class="st">"padj"</span><span class="op">]</span><span class="op">&lt;</span><span class="va">fdrcut</span>,<span class="op">]</span></span>
<span><span class="va">dat2</span> <span class="op">&lt;-</span> <span class="va">dat2</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">[</span>,<span class="st">"padj"</span><span class="op">]</span><span class="op">)</span>,<span class="op">]</span> </span>
<span><span class="va">dat2</span><span class="op">$</span><span class="va">signature</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">$</span><span class="va">NAME</span>, <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">$</span><span class="va">NAME</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Determine what labels to color</span></span>
<span><span class="va">sign_neg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">[</span>,<span class="st">"NES"</span><span class="op">]</span><span class="op">&lt;</span><span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">sign_pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">[</span>,<span class="st">"NES"</span><span class="op">]</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span></span>
<span><span class="co"># Color labels</span></span>
<span><span class="va">signcol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">$</span><span class="va">signature</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">signcol</span><span class="op">[</span><span class="va">sign_neg</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">dencol_neg</span> <span class="co"># text color of negative signatures</span></span>
<span><span class="va">signcol</span><span class="op">[</span><span class="va">sign_pos</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">dencol_pos</span> <span class="co"># text color of positive signatures</span></span>
<span><span class="va">signcol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="va">signcol</span><span class="op">)</span> <span class="co"># need to revert vector of colors, because ggplot starts plotting these from below</span></span>
<span><span class="co"># Plot bubble plot</span></span>
<span><span class="va">g</span><span class="op">&lt;-</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">dat2</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">padj</span>,y<span class="op">=</span><span class="va">signature</span>,size<span class="op">=</span><span class="va">size</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">NES</span><span class="op">)</span>, shape<span class="op">=</span><span class="fl">21</span>, colour<span class="op">=</span><span class="st">"white"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span> <span class="co"># white background, needs to be placed before the "signcol" line</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">xlim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="va">fdrcut</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html" class="external-link">scale_size_area</a></span><span class="op">(</span>max_size<span class="op">=</span><span class="fl">10</span>,guide<span class="op">=</span><span class="st">"none"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradient2</a></span><span class="op">(</span>low<span class="op">=</span><span class="va">dencol_neg</span>, high<span class="op">=</span><span class="va">dencol_pos</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>colour<span class="op">=</span><span class="va">signcol</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="va">asp</span>, axis.title.y<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="co"># test aspect.ratio</span></span></code></pre></div>
<p><img src="LionessApplicationinGTExData_files/figure-html/unnamed-chunk-9-1.png" width="672"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill, Kate Shutta.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
