<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An Introduction to the MONSTER Package • netZooR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/all.min.css" integrity="sha256-PbSmjxuVAzJ6FPvNYsrXygfGhNJYyZ2GktDbkMBqQZg=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/v4-shims.min.css" integrity="sha256-A6jcAdwFD48VMjlI3GDxUd+eCQa7/KWy6G9oe/ovaPA=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An Introduction to the MONSTER Package">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.6.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/ApplicationwithTBdataset.html">Application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>An Introduction to the MONSTER Package</h1>
                        <h4 class="author">Daniel Schlauch</h4>
            
            <h4 class="date">2017-09-22</h4>
      
      
      <div class="hidden name"><code>MONSTER.Rmd</code></div>

    </div>

    
    
<div id="using-monster" class="section level1">
<h1 class="hasAnchor">
<a href="#using-monster" class="anchor"></a>Using MONSTER</h1>
<p>MONSTER (MOdeling Network State Transitions from Expression and Regulatory data) is a tool for identifying the transcription factor drivers of state change.</p>
<p>Specific cellular states are often associated with distinct gene expression patterns. These states are plastic, changing during development, or in the transition from health to disease. One relatively simple extension of this concept is to recognize that we can classify different cell-types by their active gene regulatory networks and that, consequently, transitions between cellular states can be modeled by changes in these underlying regulatory networks. This package is an implementation of MONSTER, MOdeling Network State Transitions from Expression and Regulatory data, a regression-based method for inferring transcription factor drivers of cell state conditions at the gene regulatory network level.</p>
<p>MONSTER takes in sequence motif data linking transcription factors (TFs) to genes and gene expression from two conditions. The goal is generate bipartite networks from the gene expression data which quantify evidence of the regulatory roles of each of the TFs to each of the genes. Next, critical TFs are identified by computing a transition matrix, which maps the gene regulatory network in the first state to the gene regulatory network in the second state.</p>
<div id="installing-monster" class="section level2">
<h2 class="hasAnchor">
<a href="#installing-monster" class="anchor"></a>Installing MONSTER</h2>
<p>This package is available from Bioconductor.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"http://www.bioconductor.org/biocLite.R"</span>)
<span class="kw">biocLite</span>(<span class="st">"MONSTER"</span>)</code></pre></div>
<p>The development version of MONSTER can be installed from Github using the <code>devtools</code> package</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">'devtools'</span>)
devtools<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/devtools/man/reexports.html">install_github</a></span>(<span class="st">'QuackenbushLab/MONSTER'</span>)</code></pre></div>
<p>The package can then be loaded via</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#library(MONSTER)</span></code></pre></div>
</div>
<div id="input-files" class="section level2">
<h2 class="hasAnchor">
<a href="#input-files" class="anchor"></a>Input files</h2>
<p>In this demo, we use the included Yeast dataset, containing three separate Yeast datasets, and a sequence motif object. The <code>yeast</code> dataset includes three separate experimental designs each involving microarray assays of 2555 genes of yeast (Saccharomyces cerevisiae). <code>yeast$exp.ko</code> is a set of 106 gene expression samples of following a number of gene knockouts. <code>yeast$exp.cc</code> is a set of 50 gene expression samples taken sets of two across 25 timepoints spanning the cellular cycle. <code>yeast$exp.sr</code> is a set of 173 gene expression samples collected under conditions of stress such as heat shock. Each expression dataset has been normalized and scaled.</p>
<p>In this tutorial, we will run <code>monster</code> to identify suspected TF drivers of the change from the early cell cycle to late cell cycle.</p>
<p>First, we load the data included in the <code>MONSTER</code> package</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(yeast)</code></pre></div>
<p>Next, we create our design vector indicating to which group each sample belongs. This vector must contain only 0’s and 1’s (NAs allowed).</p>
<p>In this example we are running the analysis on the first 10 timepoints compared to the last 10 timepoints, ignoring the middle 5 for the purposes of simplicity in this tutorial. Each timepoint contains two samples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">design &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="dv">0</span>,<span class="dv">20</span>),<span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="ot">NA</span>,<span class="dv">10</span>),<span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="dv">1</span>,<span class="dv">20</span>))</code></pre></div>
<p>The main method in MONSTER is the <code>monster</code> function. This function has three required arguments,</p>
<ul>
<li>A gene expression matrix, <code>yeast$exp.cc</code>
</li>
<li>A motif mapping data.frame, <code>yeast$motif</code>
</li>
<li>A design integer vector, <code>design</code>
</li>
</ul>
<p>The gene expression argument may be a <code>matrix</code>, a <code>data.frame</code> or an <code>ExpressionSet</code>. In this example, <code>yeast$exp.cc</code> is a <code>data.frame</code> consisting of 2555 rows and 50 columns.</p>
<p>The first five rows and columns can be seen</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">yeast<span class="op">$</span>exp.cc[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</code></pre></div>
<p>The motif mapping data.frame tells the MONSTER algorithm which genes contain likely transcription factor binding sites in the vicinity of their promoter. This serves as the regulatory prior and informs the initial network inference method by supplying a partial list of TF targeting.</p>
<p>This data.frame contains 3 columns, where each row is a specific edge in the prior network.</p>
<ul>
<li>Column 1 specifies the transcription factor for the edge.</li>
<li>Column 2 specifies the targeted gene for the edge</li>
<li>Column 3 defines the strength of the edge. By default, in unweighted graphs, this column may be populated entirely with 1’s.</li>
</ul>
<p>The set of unique TFs in column 1 and unique genes in column 2 serve to determine the set of TFs and genes that are used in the downstream analysis.</p>
<p>The first five rows and columns of the example motif <code>data.frame</code> can be seen</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">yeast<span class="op">$</span>motif[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,]</code></pre></div>
<p>MONSTER tests the statistical significance of its results by permuting the samples <code>n</code> times and rerunning the analysis for each permutation. By default, the number of permutations is set to be 100 and can be manually set via the argument <code>nullPerms</code>.</p>
<p>Monster is optimized to run on multiple cores, if available. We can specify the maximum number of cores which are available to run the algorithm. If <code>numMaxCores</code> unspecified, MONSTER will check available resources and run on all but four of the available cores.</p>
</div>
<div id="running-monster" class="section level2">
<h2 class="hasAnchor">
<a href="#running-monster" class="anchor"></a>Running MONSTER</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">yeast<span class="op">$</span>exp.cc[<span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(yeast<span class="op">$</span>exp.cc)] &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(yeast<span class="op">$</span>exp.cc),<span class="dt">na.rm=</span>T)
monsterRes &lt;-<span class="st"> </span><span class="kw"><a href="../reference/monster.html">monster</a></span>(yeast<span class="op">$</span>exp.cc, design, yeast<span class="op">$</span>motif, <span class="dt">nullPerms=</span><span class="dv">100</span>, <span class="dt">numMaxCores=</span><span class="dv">4</span>)</code></pre></div>
<p>We can print the details of the analysis result</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">monsterRes</code></pre></div>
<p>In addition to the three required arguments, we have specified that we will compute 100 randomized runs of the analysis to estimate our result’s statistical signficance. MONSTER makes use of parallelization, and we have specified that 4 cores will be used in this analysis to reduce computation time.</p>
<p>Our result comes in the form of a monster object, <code>monsterRes</code>, which contains the estimated transition matrix as well as the transition matrices from the 100 null transitions.</p>
</div>
<div id="visualizing-monster-results" class="section level2">
<h2 class="hasAnchor">
<a href="#visualizing-monster-results" class="anchor"></a>Visualizing MONSTER results</h2>
<p>Many different plotting options are available in the MONSTER package which make use of additional libraries such as ggplot and igraph. Typically, we are interested in features of the transition matrix, particularly with respect to the distribution of those features under the null.</p>
<hr>
<p>The main plot function is the dTFI plot, utilizing the <code>ggplot</code> library. Of interest is the degree to which the observed transition matrix differs from those obtained via random premutations of the samples. We quantify this difference via differential TF Involvement, <span class="math inline">\(dTFI\)</span>, defined as the sum of squared off-diagonal elements in each column of the transition matrix, <span class="math display">\[\hat{dTFI_{j}}=\frac{\sum_{i=1}^{m}I\left(i\ne j\right)\hat{\tau}_{i,j}^{2}}{\sum_{i=1}^{m}\hat{\tau}_{i,j}^{2}}\]</span></p>
<p>We can view the <span class="math inline">\(dTFI\)</span> with the generic <code>plot</code> function</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(monsterRes)</code></pre></div>
<p>In this plot, we have the <span class="math inline">\(dTFI\)</span> for each transcription factor along the x-axis. The observed values are shown in red and the null values are shown in blue.</p>
<p>Due to the complex nature of the structure of gene regulatory networks, it is not uncommon to see transcription factors which exhibit a high degree of transitional change, but which is not statistically significant due to the high variability of that particular TF (e.g. YDR463W). Conversely, some TFs show weak changes, but those changes are large compared to the changes observed in null transitions (e.g. YKL062W). Ideally, we are most interested in TFs which demonstrate large changes in targetting pattern which is found to be strongly significant (e.g. YJL056C).</p>
<p>Adding the argument <code>sort.by.sig=TRUE</code>, sorts the x-axis so that the most significant transcription factors are on the left.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(monsterRes, <span class="dt">rescale=</span><span class="ot">TRUE</span>)</code></pre></div>
<p>Our top hit here is YDL056W, which reassuringly is established in the literature as being involved in regulation of cell cycle progression from G1 to S phase [Koch C, et al. (1993)]</p>
<hr>
<p>The dTFI plot focuses primarily on individual transcription factors which have systematically changed their targetting patterns between groups. To dive further into the mechanisms, we may be specifically interested in which TFs are acquiring the targetting signatures of which other TFs. We can visualize the transition matrix in a number of ways using MONSTER.</p>
<p>First, using the package <code>gplots</code>, we can simply plot the transition matrix. The <code>heatmap.2</code> function will show the <span class="math inline">\(m\times m\)</span> transition matrix in the form of a square heatmap with <span class="math inline">\(m\)</span> being the number of transcription factors. Intuitively, this is the operator, <span class="math inline">\(\textbf{T}\)</span>, on which we transform gene regulatory network <span class="math inline">\(\textbf{A}\)</span> (The first 10 timepoints in the Yeast Cell Cycle) to network <span class="math inline">\(\textbf{B}\)</span> (The last 10 timepoints in the Yeast Cell Cycle) via the equation <span class="math display">\[\textbf{B}=\textbf{AT} + \textbf{E}\]</span> where <span class="math inline">\(\textbf{E}\)</span> is the <span class="math inline">\(p\times m\)</span> error matrix which we are minimizing.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(gplots)
<span class="kw"><a href="https://rdrr.io/pkg/gplots/man/heatmap.2.html">heatmap.2</a></span>(<span class="kw">slot</span>(monsterRes, <span class="st">'tm'</span>), <span class="dt">col =</span> <span class="st">"bluered"</span>,
<span class="dt">density.info=</span><span class="st">"none"</span>,  
<span class="dt">trace=</span><span class="st">"none"</span>, 
<span class="dt">dendrogram=</span><span class="st">'none'</span>,     
<span class="dt">Rowv=</span><span class="ot">FALSE</span>,
<span class="dt">Colv=</span><span class="ot">FALSE</span>)</code></pre></div>
<p>In examining this heatmap, we are interested in strong deviations from the identity matrix. The diagonal is removed for visualization purposes. We can see that the cell cycle change is strong driven by a handful of transcription factors. Specifically, YBL005W, YLR228C, YLR451W and YML0051W.</p>
<p>This transition may also be depicted as a graph, displaying the gain or loss of features between transcription factors. Recall, that a large deviation from zero off of the diagonal indicates that the targetting pattern of one transcription factor is being “transferred” to another transcription factor as we move from the initial state to the final state.</p>
<p><code>MONSTER</code> contains the function <code>transitionNetworkPlot</code> to makes use of the <code>igraph</code> package to display the transition in network states. Since this graph is complete with negative edgeweights allowed, the argument <code>numEdges=20</code> (default is 100 edges) is used to specify the number of top transitions to display.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">transitionNetworkPlot</span>(monsterRes, <span class="dt">numEdges=</span><span class="dv">20</span>)</code></pre></div>
<p>A network visualization of the strongest 20 transitions identified based on the transition matrix above. Arrows indicate a change in edges from a transcription factor in the network of the first 10 timepoints in the Yeast Cell Cycle to resemble those of a transcription factor in the last 10 timepoints in the Yeast Cell Cycle. Edges are sized according to the magnitude of the transition and nodes (TFs) are sized by the dTFI for that TF. The gain of targeting features is indicated by the color blue while the loss of features is indicated by red.</p>
<p>Furthermore, we are often interested in correlated targetting pattern sharing. To find clusters of transcription factor transitions, we can plot the set of TFs onto the first two principal components taken from the transition matrix.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(ggplot2)
    tm.pca &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/princomp.html">princomp</a></span>(<span class="kw">slot</span>(monsterRes, <span class="st">'tm'</span>))
    odsm &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="kw">slot</span>(monsterRes, <span class="st">'tm'</span>),<span class="dv">2</span>,<span class="cf">function</span>(x){<span class="kw"><a href="https://rdrr.io/r/base/t.html">t</a></span>(x)<span class="op">%*%</span>x})
    odsm.scaled &lt;-<span class="st"> </span>(odsm<span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(odsm))<span class="op">/</span><span class="kw"><a href="https://rdrr.io/r/stats/sd.html">sd</a></span>(odsm)<span class="op">+</span><span class="dv">4</span>
    scores.pca &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(tm.pca<span class="op">$</span>scores)
    scores.pca &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(scores.pca,<span class="st">'node.names'</span>=<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(scores.pca))
    <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="dt">data =</span> scores.pca, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x =</span> Comp.<span class="dv">1</span>, <span class="dt">y =</span> Comp.<span class="dv">2</span>, <span class="dt">label =</span> node.names)) <span class="op">+</span>
<span class="st">        </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_hline</a></span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">colour =</span> <span class="st">"gray65"</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span>(<span class="dt">xintercept =</span> <span class="dv">0</span>, <span class="dt">colour =</span> <span class="st">"gray65"</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span>(<span class="dt">size =</span> odsm.scaled) <span class="op">+</span>
<span class="st">        </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/expand_limits.html">expand_limits</a></span>(<span class="dt">x=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="op">-</span>.<span class="dv">6</span>,.<span class="dv">7</span>))<span class="op">+</span>
<span class="st">        </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"PCA of transitions of Cell Cycle Transcription Factors in Yeast"</span>)</code></pre></div>
<p>The above plot can also be achieved using the included <code>MONSTER</code> function, <code>transitionPCAPlot(monsterRes)</code>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#using-monster">Using MONSTER</a><ul class="nav nav-pills nav-stacked">
<li><a href="#installing-monster">Installing MONSTER</a></li>
      <li><a href="#input-files">Input files</a></li>
      <li><a href="#running-monster">Running MONSTER</a></li>
      <li><a href="#visualizing-monster-results">Visualizing MONSTER results</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Tian Wang, John Platig, Marieke Kuijjer, Magha Padi.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
