<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Building PANDA Regulatory Networks from GTEx Gene Expression Data in R • netZooR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/all.min.css" integrity="sha256-PbSmjxuVAzJ6FPvNYsrXygfGhNJYyZ2GktDbkMBqQZg=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/v4-shims.min.css" integrity="sha256-A6jcAdwFD48VMjlI3GDxUd+eCQa7/KWy6G9oe/ovaPA=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Building PANDA Regulatory Networks from GTEx Gene Expression Data in R">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    </li>
    <li>
      <a href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="../articles/netZooRDevelopmentHandbook.html">netZooR Development Handbook</a>
    </li>
    <li>
      <a href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    </li>
    <li>
      <a href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</h1>
                        <h4 class="author">Camila Lopes-Ramos</h4>
            
            <h4 class="date">2019-12-20</h4>
      
      
      <div class="hidden name"><code>pandaRApplicationinGTExData.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>In this vignette, we will build one regulatory network for LCL cell line samples and one for whole blood samples from the GTEx gene expression data using the pandaR package. Next, we will compare the two networks, and find the pathways enriched for genes differentially targeted between the LCL cell line and whole blood.</p>
<p>Cell lines are an essential tool in biomedical research and often used as surrogates for tissues. LCLs (obtained from the transformation of B cells present in whole blood) are among the most widely used continuous cell lines with the ability to proliferate indefinitely. By comparing the regulatory networks of LCL cell lines with its tissue of origin (whole blood), we find that LCLs exhibit large changes in their patterns of transcription factor regulation, specifically a loss of repressive transcription factor targeting of cell cycle genes.</p>
<div id="install-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#install-packages" class="anchor"></a>Install packages</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))
    <span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>,<span class="dt">repos =</span> <span class="st">"http://cran.us.r-project.org"</span>)
BiocManager<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"fgsea"</span>)
<span class="co"># install.packages("reshape2",repos = "http://cran.us.r-project.org")</span>
<span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"ggplot2"</span>,<span class="dt">repos =</span> <span class="st">"http://cran.us.r-project.org"</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"devtools"</span>)
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(devtools)
devtools<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/devtools/man/reexports.html">install_github</a></span>(<span class="st">"netZoo/netZooR"</span>, <span class="dt">build_vignettes =</span> <span class="ot">TRUE</span>)</code></pre></div>
</div>
<div id="load-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(netZooR)
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(fgsea)
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ggplot2)
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(reshape2)
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(data.table)</code></pre></div>
</div>
</div>
<div id="panda" class="section level1">
<h1 class="hasAnchor">
<a href="#panda" class="anchor"></a>PANDA</h1>
<div id="panda-overview" class="section level2">
<h2 class="hasAnchor">
<a href="#panda-overview" class="anchor"></a>PANDA Overview</h2>
<p>PANDA (Passing Attributes between Networks for Data Assimilation) is a method for constructing gene regulatory networks. It uses message passing to find congruence between 3 different data layers: protein-protein interaction (PPI), gene expression, and transcription factor (TF) motif data.</p>
<p>More details can be found in the published paper <a href="https://doi.org/10.1371/journal.pone.0064832" class="uri">https://doi.org/10.1371/journal.pone.0064832</a>.</p>
</div>
<div id="building-a-panda-regulatory-network" class="section level2">
<h2 class="hasAnchor">
<a href="#building-a-panda-regulatory-network" class="anchor"></a>Building a PANDA regulatory network</h2>
<p>Now we locate our ppi and motif priors. The ppi represents physical interactions between transcription factor proteins, and is an undirected network. The transcription factor motif prior represents putative regulation events where a transcription factor binds in the promotor of a gene to regulate its expression, as predicted by the presence of transcription factor binding motifs in the promotor region of the gene. The motif prior is thus a directed network linking transcription factors to their predicted gene targets. These are small example priors for the purposes of demonstrating this method. A complete set of priors by species can be downloaded from: <a href="https://sites.google.com/a/channing.harvard.edu/kimberlyglass/tools/resources" class="uri">https://sites.google.com/a/channing.harvard.edu/kimberlyglass/tools/resources</a></p>
<p>Let’s take a look at the priors:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># download motif and ppi file from AWS Bucket</span>
<span class="kw"><a href="https://rdrr.io/r/base/system.html">system</a></span>(<span class="st">"curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/motif_subset.txt"</span>)
<span class="kw"><a href="https://rdrr.io/r/base/system.html">system</a></span>(<span class="st">"curl -O https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/ppi_subset.txt"</span>)
motif &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"./motif_subset.txt"</span>, <span class="dt">stringsAsFactors=</span>F, <span class="dt">header=</span>F)
ppi &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"./ppi_subset.txt"</span>, <span class="dt">stringsAsFactors=</span>F, <span class="dt">header=</span>F)
ppi[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,]
<span class="co">#&gt;      V1      V2    V3</span>
<span class="co">#&gt; 1 ASCL1   ASCL1 1.000</span>
<span class="co">#&gt; 2 ASCL1 BHLHE41 0.314</span>
<span class="co">#&gt; 3 ASCL1    DLX3 0.188</span>
<span class="co">#&gt; 4 ASCL1     EN2 0.790</span>
<span class="co">#&gt; 5 ASCL1    HES4 0.451</span>
motif[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,]
<span class="co">#&gt;      V1     V2 V3</span>
<span class="co">#&gt; 1 ASCL1  RPAP3  0</span>
<span class="co">#&gt; 2 ASCL1  ACSM3  0</span>
<span class="co">#&gt; 3 ASCL1 CREBBP  0</span>
<span class="co">#&gt; 4 ASCL1 ZNF263  0</span>
<span class="co">#&gt; 5 ASCL1   RALA  0</span></code></pre></div>
<p>Now we locate our expression data. As an example, we will use a subset of the GTEx version 7 RNA-Seq data, downloaded from <a href="https://gtexportal.org/home/datasets" class="uri">https://gtexportal.org/home/datasets</a>. We start with a subset of RNA-Seq data (tpm normalized) for 1,000 genes from 130 LCL cell line samples and 407 whole blood samples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># dowload and load the GTEx expression matrix (tpm normalized expression)</span>
<span class="kw"><a href="https://rdrr.io/r/base/system.html">system</a></span>(<span class="st">"curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/expression_tpm_lcl_blood_subset.txt"</span>)
exp &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"./expression_tpm_lcl_blood_subset.txt"</span>, <span class="dt">stringsAsFactors =</span> F, <span class="dt">check.names =</span> F)
<span class="co"># Log transform the tpm normalized expression</span>
exp &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log2</a></span>(exp<span class="op">+</span><span class="dv">1</span>)

<span class="co"># Determine the number of non-NA/non-zero rows in the expression data. This is to be able to ensure that PANDA will have enough values in the vectors to calculate pearson correlations between gene expression profiles in the construction of the gene co-expression prior.</span>
zero_na_counts &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(exp, <span class="dt">MARGIN =</span> <span class="dv">1</span>, <span class="dt">FUN =</span> <span class="cf">function</span>(x) <span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(x[(<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(x) <span class="op">&amp;</span><span class="st"> </span>x<span class="op">!=</span><span class="dv">0</span>) ]))
<span class="co"># Maintain only genes with at least 20 valid gene expression entries</span>
exp &lt;-<span class="st"> </span>exp[zero_na_counts <span class="op">&gt;</span><span class="st"> </span><span class="dv">20</span>,]

<span class="co"># The set of genes in the expression, motif and ppi matrices must be the same</span>
exp &lt;-<span class="st"> </span>exp[<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(exp) <span class="op">%in%</span><span class="st"> </span>motif<span class="op">$</span>V2,]
motif_subset &lt;-<span class="st"> </span>motif[(motif<span class="op">$</span>V1 <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(exp)) <span class="op">&amp;</span><span class="st"> </span>(motif<span class="op">$</span>V2 <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(exp)),]
ppi_subset &lt;-<span class="st"> </span>ppi[(ppi<span class="op">$</span>V1 <span class="op">%in%</span><span class="st"> </span>motif_subset<span class="op">$</span>V1) <span class="op">&amp;</span><span class="st"> </span>(ppi<span class="op">$</span>V2 <span class="op">%in%</span><span class="st"> </span>motif_subset<span class="op">$</span>V1),]

<span class="co"># Load the sample ids of LCL samples</span>
<span class="kw"><a href="https://rdrr.io/r/base/system.html">system</a></span>(<span class="st">"curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/LCL_samples.txt"</span>)
lcl_samples &lt;-<span class="kw"><a href="https://rdrr.io/pkg/data.table/man/fread.html">fread</a></span>(<span class="st">"./LCL_samples.txt"</span>, <span class="dt">header =</span> <span class="ot">FALSE</span>, <span class="dt">data.table=</span><span class="ot">FALSE</span>)

<span class="co"># Select the columns of the expression matrix corresponding to the LCL samples</span>
lcl_exp &lt;-<span class="st"> </span>exp[,<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(exp) <span class="op">%in%</span><span class="st"> </span>lcl_samples[,<span class="dv">1</span>]]

<span class="co"># Load the sample ids of whole blood samples</span>
<span class="kw"><a href="https://rdrr.io/r/base/system.html">system</a></span>(<span class="st">"curl -O https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/WholeBlood_samples.txt"</span>)
wblood_samples &lt;-<span class="kw"><a href="https://rdrr.io/pkg/data.table/man/fread.html">fread</a></span>(<span class="st">"./WholeBlood_samples.txt"</span>, <span class="dt">header =</span> <span class="ot">FALSE</span>, <span class="dt">data.table=</span><span class="ot">FALSE</span>)

<span class="co"># Select the columns of the expression matrix corresponding to the whole blood samples</span>
wb_exp &lt;-<span class="st"> </span>exp[,<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(exp) <span class="op">%in%</span><span class="st"> </span>wblood_samples[,<span class="dv">1</span>]]</code></pre></div>
<p>Now we run PANDA, pointing it to the parsed expression data, motif prior and ppi prior. We will point to the same motif and ppi priors for each PANDA run, which represents the initial putative regulatory information. We then point to the expression matrix correspoding to the LCL samples to generate the LCL regulatory network, and to the expression matrix corresponding to the whole blood samples to generate the whole blood regulatory network.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pandaLCL &lt;-<span class="st"> </span><span class="kw">panda</span>(motif_subset, lcl_exp, ppi_subset)
pandaWB &lt;-<span class="st"> </span><span class="kw">panda</span>(motif_subset, wb_exp, ppi_subset)

pandaLCL
<span class="co">#&gt; PANDA network for968genes and49transcription factors.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Slots:</span>
<span class="co">#&gt; regNet   : Regulatory network of 49 transcription factors to 968 genes.</span>
<span class="co">#&gt; coregNet: Co-regulation network of 968 genes.</span>
<span class="co">#&gt; coopNet  : Cooperative network of 49 transcription factors.</span>
<span class="co">#&gt; Regulatory graph contains 47432 edges.</span>
<span class="co">#&gt; Regulatory graph is complete.</span>
pandaWB
<span class="co">#&gt; PANDA network for968genes and49transcription factors.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Slots:</span>
<span class="co">#&gt; regNet   : Regulatory network of 49 transcription factors to 968 genes.</span>
<span class="co">#&gt; coregNet: Co-regulation network of 968 genes.</span>
<span class="co">#&gt; coopNet  : Cooperative network of 49 transcription factors.</span>
<span class="co">#&gt; Regulatory graph contains 47432 edges.</span>
<span class="co">#&gt; Regulatory graph is complete.</span></code></pre></div>
<p>The regulatory network (bipartite graph) with edge weights representing the “likelihood” that a transcription factor binds the promotor of and regulates the expression of a gene.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># The bipartite regulatory network (transcription factors as rows and target genes as columns)</span>
regNetLCL &lt;-<span class="st"> </span>pandaLCL<span class="op">@</span>regNet
regNetLCL[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]
<span class="co">#&gt;         RP4-669L17.8    MTND2P28        HES4     ACTRT2 RP3-467L1.6</span>
<span class="co">#&gt; ASCL1     -0.4208359 -0.04542644  0.44233153 -0.3490675 -0.10754463</span>
<span class="co">#&gt; BARX1     -0.5501750 -0.84394635 -0.67999717 -0.2696099  0.07233836</span>
<span class="co">#&gt; BHLHE41   -0.4587494 -0.22638411 -0.04797392 -0.2289900 -0.44595612</span>
<span class="co">#&gt; CEBPD     -0.1076370  0.06141078 -0.04441761 -0.2097561 -0.51995579</span>
<span class="co">#&gt; CEBPZ     -0.2217286 -0.18001207 -0.13000309 -0.3191476 -0.51898843</span>
regNetWB &lt;-<span class="st"> </span>pandaWB<span class="op">@</span>regNet</code></pre></div>
</div>
</div>
<div id="visualizing-networks-in-cytoscape" class="section level1">
<h1 class="hasAnchor">
<a href="#visualizing-networks-in-cytoscape" class="anchor"></a>Visualizing networks in Cytoscape</h1>
<p>In this section we will visualize parts of the network using the Cytoscape software. Download Cytoscape from: <a href="https://cytoscape.org" class="uri">https://cytoscape.org</a></p>
<div id="preparing-data-to-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#preparing-data-to-plot" class="anchor"></a>Preparing data to plot</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># We will use the function vis.panda.in.cytoscape to plot a set of nodes and edges on Cytoscape. The input for this function is a data.frame of edges to plot with 4 columns: "tf", "gene", "motif" (TF motif present or not on gene promoter), "force" (edge weight calculated by PANDA).</span>
lcl_vis &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span>(pandaLCL<span class="op">@</span>regNet)
wb_vis &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span>(pandaWB<span class="op">@</span>regNet)
lcl_vis &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="st">"TF"</span>=<span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(lcl_vis[,<span class="dv">1</span>]),<span class="st">"Gene"</span>=<span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(lcl_vis[,<span class="dv">2</span>]),<span class="st">"Motif"</span>=<span class="ot">NA</span>,<span class="st">"Score"</span>=<span class="kw"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(lcl_vis[,<span class="dv">3</span>]),<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
wb_vis &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="st">"TF"</span>=<span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(wb_vis[,<span class="dv">1</span>]),<span class="st">"Gene"</span>=<span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(wb_vis[,<span class="dv">2</span>]),<span class="st">"Motif"</span>=<span class="ot">NA</span>,<span class="st">"Score"</span>=<span class="kw"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(wb_vis[,<span class="dv">3</span>]),<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(lcl_vis)
<span class="co">#&gt;        TF         Gene Motif      Score</span>
<span class="co">#&gt; 1   ASCL1 RP4-669L17.8    NA -0.4208359</span>
<span class="co">#&gt; 2   BARX1 RP4-669L17.8    NA -0.5501750</span>
<span class="co">#&gt; 3 BHLHE41 RP4-669L17.8    NA -0.4587494</span>
<span class="co">#&gt; 4   CEBPD RP4-669L17.8    NA -0.1076370</span>
<span class="co">#&gt; 5   CEBPZ RP4-669L17.8    NA -0.2217286</span>
<span class="co">#&gt; 6   CENPB RP4-669L17.8    NA -0.1081259</span></code></pre></div>
</div>
<div id="plot-the-200-highest-edge-weights" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-200-highest-edge-weights" class="anchor"></a>Plot the 200 highest edge weights</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">n=<span class="dv">200</span> <span class="co"># number of edges to plot </span>
top &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/data.table/man/setorder.html">order</a></span>(lcl_vis<span class="op">$</span>Score,<span class="dt">decreasing=</span>T)[<span class="dv">1</span><span class="op">:</span>n]
lcl_vis_top &lt;-<span class="st"> </span>lcl_vis[top,]
<span class="co"># Plot in cytoscape (open the cytoscape software before running this command)</span>
<span class="kw"><a href="../reference/vis.panda.in.cytoscape.html">vis.panda.in.cytoscape</a></span>(lcl_vis_top, <span class="dt">network_name=</span><span class="st">"LCL"</span>)
<span class="co"># Here we will load a customized visual style for our network, in which TF nodes are orange circles, target gene nodes are blue squares, and edges shade and width are the edge weight (likelyhood of regulatory interaction between the TF and gene). You can further customize the network style directly from Cytoscape.</span>
<span class="kw"><a href="../reference/create.panda.style.html">create.panda.style</a></span>(<span class="dt">style_name=</span><span class="st">"PandaStyle"</span>)</code></pre></div>
</div>
<div id="plot-the-top-differential-edges-betwen-lcl-and-wb" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-top-differential-edges-betwen-lcl-and-wb" class="anchor"></a>Plot the top differential edges betwen LCL and WB</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Select the top differential edge weights betweeen LCL and whole blood</span>
diffRes &lt;-<span class="st"> </span><span class="kw"><a href="../reference/panda.diff.edges.html">panda.diff.edges</a></span>(lcl_vis, wb_vis, <span class="dt">condition_name=</span><span class="st">"LCL"</span>)
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(diffRes)
<span class="co"># Number of differential edges is:</span>
<span class="kw"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(diffRes)

<span class="co"># Select the top differential edges higher in LCL to plot in Cytoscape</span>
n=<span class="dv">200</span> <span class="co"># number of edges to select from each condition</span>
diffResLCL &lt;-<span class="st"> </span>diffRes[diffRes<span class="op">$</span>LCL<span class="op">==</span><span class="st">"T"</span>,]
diffResLCL &lt;-<span class="st"> </span>diffResLCL[<span class="kw"><a href="https://rdrr.io/pkg/data.table/man/setorder.html">order</a></span>(diffResLCL<span class="op">$</span>Score,<span class="dt">decreasing=</span><span class="ot">TRUE</span>),][<span class="dv">1</span><span class="op">:</span>n,]

<span class="co"># Select the top differential edges higher in whole blood to plot in Cytoscape</span>
diffResWB &lt;-<span class="st"> </span>diffRes[diffRes<span class="op">$</span>LCL<span class="op">==</span><span class="st">"F"</span>,]
diffResWB &lt;-<span class="st"> </span>diffResWB[<span class="kw"><a href="https://rdrr.io/pkg/data.table/man/setorder.html">order</a></span>(diffResWB<span class="op">$</span>Score,<span class="dt">decreasing=</span><span class="ot">TRUE</span>),][<span class="dv">1</span><span class="op">:</span>n,]

<span class="co"># Combine top differential edges in LCL and WB to plot in Cytoscape</span>
diffRes_vis &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/data.table/man/rbindlist.html">rbind</a></span>(diffResLCL, diffResWB)

<span class="co"># Plot the network</span>
<span class="co"># Purple edges indicate higher edge weight in the defined "condition_name" parameter (LCL in our example), and green edges indicate higher edge weight in the other condition (whole blood in our example).</span>
<span class="kw"><a href="../reference/vis.diff.panda.in.cytoscape.html">vis.diff.panda.in.cytoscape</a></span>(diffRes_vis, <span class="dt">condition_name =</span> <span class="st">"LCL"</span>, <span class="dt">network_name=</span><span class="st">"diff.PANDA"</span>)
<span class="co"># Apply the style to the network</span>
<span class="kw"><a href="../reference/create.diff.panda.style.html">create.diff.panda.style</a></span>(<span class="dt">style_name=</span><span class="st">"Diff.PandaStyle"</span>, <span class="dt">condition_name=</span><span class="st">"LCL"</span>)</code></pre></div>
</div>
</div>
<div id="calculating-degree" class="section level1">
<h1 class="hasAnchor">
<a href="#calculating-degree" class="anchor"></a>Calculating degree</h1>
<ul>
<li>out-degrees of TFs: sum of the weights of outbound edges around a TF</li>
<li>in-degrees of genes: sum of the weights of inbound edges around a gene</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lcl_outdegree &lt;-<span class="st"> </span><span class="kw">calcDegree</span>(pandaLCL, <span class="dt">type=</span><span class="st">"tf"</span>)
wb_outdegree &lt;-<span class="st"> </span><span class="kw">calcDegree</span>(pandaWB, <span class="dt">type=</span><span class="st">"tf"</span>)

lcl_indegree &lt;-<span class="st"> </span><span class="kw">calcDegree</span>(pandaLCL, <span class="dt">type=</span><span class="st">"gene"</span>)
wb_indegree &lt;-<span class="st"> </span><span class="kw">calcDegree</span>(pandaWB, <span class="dt">type=</span><span class="st">"gene"</span>)

<span class="co"># Calculate the gene in-degree difference for two different panda regulatory networks (LCL minus whole blood)</span>
degreeDiff &lt;-<span class="st"> </span><span class="kw">calcDegreeDifference</span>(pandaLCL, pandaWB, <span class="dt">type=</span><span class="st">"gene"</span>)
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(degreeDiff)
<span class="co">#&gt; RP4-669L17.8     MTND2P28         HES4       ACTRT2  RP3-467L1.6 </span>
<span class="co">#&gt;   -0.5824133   -4.3127789   -0.6148939    0.5242457    0.3215045 </span>
<span class="co">#&gt;        FBXO6 </span>
<span class="co">#&gt;   -0.2831833</span></code></pre></div>
</div>
<div id="gene-set-enrichment-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#gene-set-enrichment-analysis" class="anchor"></a>Gene Set Enrichment Analysis</h1>
<p>Well will use the fgsea package to perform gene set enrichment analysis. We need to point to a ranked gene list (for example the gene in-degree difference between LCL and whole blood), and a list of gene sets (or signatures) in gmt format to test for enrichment. The gene sets can be downloaded from MSigDB: <a href="http://software.broadinstitute.org/gsea/msigdb" class="uri">http://software.broadinstitute.org/gsea/msigdb</a> Same gene annotation should be used in the ranked gene list and gene sets. In our example we will use the KEGG pathways downloaded from MSigDB.</p>
<div id="run-fgsea" class="section level2">
<h2 class="hasAnchor">
<a href="#run-fgsea" class="anchor"></a>Run fgsea</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/system.html">system</a></span>(<span class="st">"curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/c2.cp.kegg.v7.0.symbols.gmt"</span>)
pathways &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/fgsea/man/gmtPathways.html">gmtPathways</a></span>(<span class="st">"./c2.cp.kegg.v7.0.symbols.gmt"</span>)
<span class="co"># To retrieve biological-relevant processes, we will load and use the complete ranked gene list (27,175 genes) calculated from the complete network instead of the 1,000 subset genes we used in this tutorial example to build PANDA networks within a very short run time.</span>
<span class="kw"><a href="https://rdrr.io/r/base/system.html">system</a></span>(<span class="st">"curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/lclWB_indegreeDifference.rnk"</span>)
degreeDiff_all &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"./lclWB_indegreeDifference.rnk"</span>,<span class="dt">stringsAsFactors =</span> F,<span class="dt">header=</span>F)
degreeDiff_all &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span>(degreeDiff_all[,<span class="dv">2</span>], degreeDiff_all[,<span class="dv">1</span>])
fgseaRes &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/fgsea/man/fgsea.html">fgsea</a></span>(pathways, degreeDiff_all, <span class="dt">minSize=</span><span class="dv">15</span>, <span class="dt">maxSize=</span><span class="dv">500</span>, <span class="dt">nperm=</span><span class="dv">1000</span>)
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(fgseaRes)
<span class="co">#&gt;                                          pathway        pval       padj</span>
<span class="co">#&gt; 1:               KEGG_GLYCOLYSIS_GLUCONEOGENESIS 0.058524173 0.14931350</span>
<span class="co">#&gt; 2:                  KEGG_CITRATE_CYCLE_TCA_CYCLE 0.001831502 0.01912298</span>
<span class="co">#&gt; 3:                KEGG_PENTOSE_PHOSPHATE_PATHWAY 0.101123596 0.21722846</span>
<span class="co">#&gt; 4: KEGG_PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS 0.647855530 0.72921784</span>
<span class="co">#&gt; 5:          KEGG_FRUCTOSE_AND_MANNOSE_METABOLISM 0.053215078 0.14029430</span>
<span class="co">#&gt; 6:                     KEGG_GALACTOSE_METABOLISM 0.033707865 0.10663943</span>
<span class="co">#&gt;            ES        NES nMoreExtreme size</span>
<span class="co">#&gt; 1:  0.3355458  1.3826323           22   58</span>
<span class="co">#&gt; 2: -0.5923935 -2.0262498            0   29</span>
<span class="co">#&gt; 3:  0.4235984  1.3765703           44   21</span>
<span class="co">#&gt; 4:  0.2675875  0.8808769          286   22</span>
<span class="co">#&gt; 5:  0.4168076  1.5036493           23   30</span>
<span class="co">#&gt; 6:  0.4491757  1.5383352           14   25</span>
<span class="co">#&gt;                                        leadingEdge</span>
<span class="co">#&gt; 1:         ALDH3B1,LDHAL6A,PGM1,PKM,ALDOA,ENO1,...</span>
<span class="co">#&gt; 2:              MDH2,CS,DLST,SDHD,IDH3B,SUCLG2,...</span>
<span class="co">#&gt; 3:               H6PD,PGM1,ALDOA,PFKM,PFKL,RPE,...</span>
<span class="co">#&gt; 4: UGT1A10,UGT1A1,UGT2B15,UGT1A6,UGT2B17,CRYL1,...</span>
<span class="co">#&gt; 5:              PFKFB4,ALDOA,HK3,PFKM,HK1,PFKL,...</span>
<span class="co">#&gt; 6:                  GAA,PGM1,HK3,GLB1,PFKM,HK1,...</span>
<span class="co"># Subset to pathways with FDR &lt; 0.05</span>
sig &lt;-<span class="st"> </span>fgseaRes[fgseaRes<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>,]
<span class="co"># Get the top 10 significant pathways enriched for genes having lower targeting in LCLs  </span>
sig[<span class="kw"><a href="https://rdrr.io/pkg/data.table/man/setorder.html">order</a></span>(sig<span class="op">$</span>NES)[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>],]
<span class="co">#&gt;                              pathway        pval       padj         ES</span>
<span class="co">#&gt;  1:                 KEGG_SPLICEOSOME 0.001577287 0.01912298 -0.6155980</span>
<span class="co">#&gt;  2:                    KEGG_RIBOSOME 0.001600000 0.01912298 -0.6438056</span>
<span class="co">#&gt;  3:             KEGG_DNA_REPLICATION 0.001763668 0.01912298 -0.7301502</span>
<span class="co">#&gt;  4:                  KEGG_CELL_CYCLE 0.001555210 0.01912298 -0.5352336</span>
<span class="co">#&gt;  5:  KEGG_NUCLEOTIDE_EXCISION_REPAIR 0.001700680 0.01912298 -0.5896497</span>
<span class="co">#&gt;  6:              KEGG_PROTEIN_EXPORT 0.001788909 0.01912298 -0.6735362</span>
<span class="co">#&gt;  7: KEGG_AMINOACYL_TRNA_BIOSYNTHESIS 0.001736111 0.01912298 -0.5489421</span>
<span class="co">#&gt;  8:     KEGG_CITRATE_CYCLE_TCA_CYCLE 0.001831502 0.01912298 -0.5923935</span>
<span class="co">#&gt;  9: KEGG_BASAL_TRANSCRIPTION_FACTORS 0.001824818 0.01912298 -0.5641948</span>
<span class="co">#&gt; 10:             KEGG_MISMATCH_REPAIR 0.001788909 0.01912298 -0.5977217</span>
<span class="co">#&gt;           NES nMoreExtreme size</span>
<span class="co">#&gt;  1: -2.772922            0  116</span>
<span class="co">#&gt;  2: -2.711604            0   80</span>
<span class="co">#&gt;  3: -2.612124            0   34</span>
<span class="co">#&gt;  4: -2.427198            0  121</span>
<span class="co">#&gt;  5: -2.200200            0   41</span>
<span class="co">#&gt;  6: -2.161853            0   22</span>
<span class="co">#&gt;  7: -2.030366            0   40</span>
<span class="co">#&gt;  8: -2.026250            0   29</span>
<span class="co">#&gt;  9: -1.962229            0   31</span>
<span class="co">#&gt; 10: -1.918511            0   22</span>
<span class="co">#&gt;                                       leadingEdge</span>
<span class="co">#&gt;  1: SRSF6,SRSF3,SRSF7,HNRNPA1L2,PRPF38A,PPIL1,...</span>
<span class="co">#&gt;  2:      RPS2,RPL23A,RPL19,RPS12,RPS17,RPL10A,...</span>
<span class="co">#&gt;  3:           RFC3,RFC5,RPA2,POLE3,PRIM1,PCNA,...</span>
<span class="co">#&gt;  4:         YWHAQ,ORC6,DBF4,ANAPC10,BUB3,WEE1,...</span>
<span class="co">#&gt;  5:            XPA,RFC3,RFC5,RPA2,POLE3,ERCC8,...</span>
<span class="co">#&gt;  6:      SRP9,IMMP1L,HSPA5,SEC11C,OXA1L,SPCS2,...</span>
<span class="co">#&gt;  7:         MARS2,NARS,YARS,FARS2,YARS2,NARS2,...</span>
<span class="co">#&gt;  8:            MDH2,CS,DLST,SDHD,IDH3B,SUCLG2,...</span>
<span class="co">#&gt;  9:          TAF4,TAF7,TAF9,GTF2E1,GTF2F2,TBP,...</span>
<span class="co">#&gt; 10:             RFC3,RFC5,RPA2,PCNA,MSH2,RPA1,...</span></code></pre></div>
</div>
<div id="bubble-plot-of-top-differentially-targeted-pathways" class="section level2">
<h2 class="hasAnchor">
<a href="#bubble-plot-of-top-differentially-targeted-pathways" class="anchor"></a>Bubble plot of top differentially targeted pathways</h2>
<p>Bubble plot of gene sets (KEGG pathways) on y-axis and adjusted p-value (padj) on x-axis. Bubble size indicates the number of genes in each gene set, and bubble color indicates the normalized enrichment score (NES). Blue is for negative NES (enrichment of higher targeted genes in whole blood), and red is for positive NES (enrichment of higher targeted genes in LCL).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dat &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(fgseaRes)

<span class="co"># Settings</span>
fdrcut &lt;-<span class="st"> </span><span class="fl">0.05</span> <span class="co"># FDR cut-off to use as output for significant signatures</span>
dencol_neg &lt;-<span class="st"> "blue"</span> <span class="co"># bubble plot color for negative ES</span>
dencol_pos &lt;-<span class="st"> "red"</span> <span class="co"># bubble plot color for positive ES</span>
signnamelength &lt;-<span class="st"> </span><span class="dv">4</span> <span class="co"># set to remove prefix from signature names (2 for "GO", 4 for "KEGG", 8 for "REACTOME")</span>
asp &lt;-<span class="st"> </span><span class="dv">3</span> <span class="co"># aspect ratio of bubble plot</span>
charcut &lt;-<span class="st"> </span><span class="dv">100</span> <span class="co"># cut signature name in heatmap to this nr of characters</span>

<span class="co"># Make signature names more readable</span>
a &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(dat<span class="op">$</span>pathway) <span class="co"># 'a' is a great variable name to substitute row names with something more readable</span>
<span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(a)){
  a[j] &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/substr.html">substr</a></span>(a[j], signnamelength<span class="op">+</span><span class="dv">2</span>, <span class="kw"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span>(a[j]))
}
a &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/chartr.html">tolower</a></span>(a) <span class="co"># convert to lower case (you may want to comment this out, it really depends on what signatures you are looking at, c6 signatures contain gene names, and converting those to lower case may be confusing)</span>
<span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(a)){
  <span class="cf">if</span>(<span class="kw"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span>(a[j])<span class="op">&gt;</span>charcut) { a[j] &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/substr.html">substr</a></span>(a[j], <span class="dv">1</span>, charcut), <span class="st">"..."</span>, <span class="dt">sep=</span><span class="st">" "</span>)}
} <span class="co"># cut signature names that have more characters than charcut, and add "..."</span>
a &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">"_"</span>, <span class="st">" "</span>, a)
dat<span class="op">$</span>NAME &lt;-<span class="st"> </span>a

<span class="co"># Determine what signatures to plot (based on FDR cut)</span>
dat2 &lt;-<span class="st"> </span>dat[dat[,<span class="st">"padj"</span>]<span class="op">&lt;</span>fdrcut,]
dat2 &lt;-<span class="st"> </span>dat2[<span class="kw"><a href="https://rdrr.io/pkg/data.table/man/setorder.html">order</a></span>(dat2[,<span class="st">"padj"</span>]),] 
dat2<span class="op">$</span>signature &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(dat2<span class="op">$</span>NAME, <span class="kw"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(dat2<span class="op">$</span>NAME)))
<span class="co"># Determine what labels to color</span>
sign_neg &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/which.html">which</a></span>(dat2[,<span class="st">"NES"</span>]<span class="op">&lt;</span><span class="dv">0</span>)
sign_pos &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/which.html">which</a></span>(dat2[,<span class="st">"NES"</span>]<span class="op">&gt;</span><span class="dv">0</span>)
<span class="co"># Color labels</span>
signcol &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="ot">NA</span>, <span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(dat2<span class="op">$</span>signature))
signcol[sign_neg] &lt;-<span class="st"> </span>dencol_neg <span class="co"># text color of negative signatures</span>
signcol[sign_pos] &lt;-<span class="st"> </span>dencol_pos <span class="co"># text color of positive signatures</span>
signcol &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(signcol) <span class="co"># need to revert vector of colors, because ggplot starts plotting these from below</span>

<span class="co"># Plot bubble plot</span>
g&lt;-<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(dat2, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x=</span>padj,<span class="dt">y=</span>signature,<span class="dt">size=</span>size))
g<span class="op">+</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">fill=</span>NES), <span class="dt">shape=</span><span class="dv">21</span>, <span class="dt">colour=</span><span class="st">"white"</span>)<span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()<span class="op">+</span><span class="st"> </span><span class="co"># white background, needs to be placed before the "signcol" line</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span>(<span class="dv">0</span>,fdrcut)<span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html">scale_size_area</a></span>(<span class="dt">max_size=</span><span class="dv">10</span>,<span class="dt">guide=</span><span class="st">"none"</span>)<span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html">scale_fill_gradient2</a></span>(<span class="dt">low=</span>dencol_neg, <span class="dt">high=</span>dencol_pos)<span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">axis.text.y =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="dt">colour=</span>signcol))<span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">aspect.ratio=</span>asp, <span class="dt">axis.title.y=</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span>()) <span class="co"># test aspect.ratio</span>
<span class="co">#&gt; Warning: Vectorized input to `element_text()` is not officially supported.</span>
<span class="co">#&gt; Results may be unexpected or may change in future versions of ggplot2.</span></code></pre></div>
<p><img src="pandaRApplicationinGTExData_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#install-packages">Install packages</a></li>
      <li><a href="#load-packages">Load packages</a></li>
      </ul>
</li>
      <li>
<a href="#panda">PANDA</a><ul class="nav nav-pills nav-stacked">
<li><a href="#panda-overview">PANDA Overview</a></li>
      <li><a href="#building-a-panda-regulatory-network">Building a PANDA regulatory network</a></li>
      </ul>
</li>
      <li>
<a href="#visualizing-networks-in-cytoscape">Visualizing networks in Cytoscape</a><ul class="nav nav-pills nav-stacked">
<li><a href="#preparing-data-to-plot">Preparing data to plot</a></li>
      <li><a href="#plot-the-200-highest-edge-weights">Plot the 200 highest edge weights</a></li>
      <li><a href="#plot-the-top-differential-edges-betwen-lcl-and-wb">Plot the top differential edges betwen LCL and WB</a></li>
      </ul>
</li>
      <li><a href="#calculating-degree">Calculating degree</a></li>
      <li>
<a href="#gene-set-enrichment-analysis">Gene Set Enrichment Analysis</a><ul class="nav nav-pills nav-stacked">
<li><a href="#run-fgsea">Run fgsea</a></li>
      <li><a href="#bubble-plot-of-top-differentially-targeted-pathways">Bubble plot of top differentially targeted pathways</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Tian Wang, Marouen Ben Guebila, John Platig, Marieke Kuijjer, Magha Padi, Rebekka Burkholz.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
