<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="netZooR">
<title>An Introduction to the pandaR Package • netZooR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="An Introduction to the pandaR Package">
<meta property="og:description" content="netZooR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">netZooR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    <a class="dropdown-item" href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    <a class="dropdown-item" href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    <a class="dropdown-item" href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    <a class="dropdown-item" href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    <a class="dropdown-item" href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    <a class="dropdown-item" href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    <a class="dropdown-item" href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    <a class="dropdown-item" href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    <a class="dropdown-item" href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/netZoo/netZooR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>An Introduction to the pandaR Package</h1>
                        <h4 data-toc-skip class="author">Daniel
Schlauch, Albert Young, Joseph N. Paulson</h4>
            
            <h4 data-toc-skip class="date">2016-05-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/netZoo/netZooR/blob/HEAD/vignettes/pandaR.Rmd" class="external-link"><code>vignettes/pandaR.Rmd</code></a></small>
      <div class="d-none name"><code>pandaR.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The fundamental concepts behind the PANDA approach is to model the
regulatory network as a bipartite network and estimate edge weights
based on the evidence that information from a particular transcription
factor <em>i</em> is successfully being passed to a particular gene
<em>j</em>. This evidence comes from the agreement between two measured
quantities. First, the correlation in expression between gene <em>j</em>
and other genes. And second, the strength of evidence of the existence
of an edge between TF <em>i</em> and those same genes. This concordance
is measured using Tanimoto similarity. A gene is said to be available if
there is strong evidence of this type of agreement. Analogous to this is
the concept of responsibility which similarly focuses on a TF-gene
network edge but instead measures the concordance between suspected
protein-complex partners of TF <em>i</em> and the respective strength of
evidence of a regulatory pathway between those TFs and gene
<em>j</em>.</p>
<p>PANDA utilizes an iterative approach to updating the bipartite edge
weights incrementally as evidence for new edges emerges and evidence for
existing edges diminishes. This process continues until the algorithm
reaches a point of convergence settling on a final score for the
strength of information supporting a regulatory mechanism for every
pairwise combination of TFs and genes. This package provides a
straightforward tool for applying this established method. Beginning
with data.frames or matrices representing a set of gene expression
samples, motif priors and optional protein-protein interaction users can
generate an <em>m</em> by <em>n</em> matrix representing the bipartite
network from <em>m</em> TFs regulating <em>n</em> genes. Additionally,
pandaR reports the co-regulation and cooperative networks at
convergence. These are reported as complete graphs representing the
evidence for gene co-regulation and transcription factor
cooperation.</p>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>An example dataset derived from a subset of stress-induced Yeast is
available by running</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/netZoo/netZooR" class="external-link">netZooR</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: igraph</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'igraph'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     decompose, spectrum</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     union</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: reticulate</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: pandaR</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:igraph':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     normalize, path, union</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">##     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">##     table, tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: yarn</span></span></code></pre>
<pre><code><span><span class="co">## Setting options('download.file.method.GEOquery'='auto')</span></span></code></pre>
<pre><code><span><span class="co">## Setting options('GEOquery.inmemory.gpl'=FALSE)</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: matrixcalc</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'matrixcalc'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:igraph':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     %s%</span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pandaR</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">pandaToyData</span><span class="op">)</span></span></code></pre></div>
<p><strong>pandaToyData</strong> is a list containing a regulatory
structure derived from sequence motif analysis, protein-protein
interaction data and a gene expression. The primary function in pandaR
is called with</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pandaResult</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pandaR/man/panda.html" class="external-link">panda</a></span><span class="op">(</span><span class="va">pandaToyData</span><span class="op">$</span><span class="va">motif</span>, <span class="va">pandaToyData</span><span class="op">$</span><span class="va">expression</span>, <span class="va">pandaToyData</span><span class="op">$</span><span class="va">ppi</span><span class="op">)</span></span>
<span><span class="va">pandaResult</span></span></code></pre></div>
<pre><code><span><span class="co">## PANDA network for1000genes and87transcription factors.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Slots:</span></span></code></pre>
<pre><code><span><span class="co">## regNet   : Regulatory network of 87 transcription factors to 1000 genes.</span></span></code></pre>
<pre><code><span><span class="co">## coregNet: Co-regulation network of 1000 genes.</span></span></code></pre>
<pre><code><span><span class="co">## coopNet  : Cooperative network of 87 transcription factors.</span></span></code></pre>
<pre><code><span><span class="co">## Regulatory graph contains 87000 edges.</span></span></code></pre>
<pre><code><span><span class="co">## Regulatory graph is complete.</span></span></code></pre>
<p>Where <strong>pandaResult</strong> is a ‘panda’ object which contains
<strong>data.frame</strong>s describing the complete bipartite gene
regulatory network as well as complete networks for gene coregulation
and transcription factor cooperation. Due to completeness, edgeweights
for the regulatory network are reported for all <em>m</em>x<em>n</em>
possible TF-gene edges. The distribution of these edge weights for these
networks has approximate mean 0 and standard deviation 1. The edges are
therefore best interpreted in a relative sense. Strongly positive values
indicative of relatively larger amounts of evidence in favor a
regulatory mechanism and conversely, smaller or negative values can be
interpreted as lacking evidence of a shared biological role. It is
naturally of interest to specify a high edge weight subset of the
complete network to investigate as a set of present/absent edges. This
is easily performed by using the <strong>topedges</strong> function. A
network containing the top 1000 edge scores as binary edges can be
obtained by:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">topNet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pandaR/man/topedges.html" class="external-link">topedges</a></span><span class="op">(</span><span class="va">pandaResult</span>, <span class="fl">1000</span><span class="op">)</span></span></code></pre></div>
<p>Users may then examine the genes targeted by a transcription factor
of interest.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/pandaR/man/targetedGenes.html" class="external-link">targetedGenes</a></span><span class="op">(</span><span class="va">topNet</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AR"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "AKAP10"       "CNDP2"        "CRHR1"        "HNRNPD"       "KIAA0652"    </span></span>
<span><span class="co">##  [6] "LOC100093631" "LOC100128811" "PRR15"        "TCF4"         "TCP11L2"     </span></span>
<span><span class="co">## [11] "TMPRSS11B"    "VCX3B"        "WDR4"</span></span></code></pre>
<p>The network can be further simplified by focusing only on
transcription factors on interest and the genes that they are found to
regulate. The <strong>subnetwork</strong> method serves this
function</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">topSubnet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pandaR/man/subnetwork.html" class="external-link">subnetwork</a></span><span class="op">(</span><span class="va">topNet</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AR"</span>,<span class="st">"ARID3A"</span>,<span class="st">"ELK1"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Existing R packages, such as igraph, can be used to visualize the
results</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/pandaR/man/plotGraph.html" class="external-link">plotGraph</a></span><span class="op">(</span><span class="va">topSubnet</span><span class="op">)</span></span></code></pre></div>
<p><img src="pandaR_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="comparing-state-specific-panda-networks">Comparing state-specific PANDA networks<a class="anchor" aria-label="anchor" href="#comparing-state-specific-panda-networks"></a>
</h2>
<p>We provide a number of useful plotting functions for the analysis of
the networks. The main functions used to plot and analyze the PANDA
networks are:</p>
<ul>
<li>
<em>plotZ</em> -Comparison of Z scores between two PANDA runs</li>
<li>
<em>plotZbyTF</em> - Plot Z by TF outdegree quartiles</li>
</ul>
<p>We can compare how parameter choices effect the Z-score estimation
between two PANDA runs. Additionally, we can compare two phenotypes:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">panda.res1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">pandaToyData</span>, <span class="fu"><a href="https://rdrr.io/pkg/pandaR/man/panda.html" class="external-link">panda</a></span><span class="op">(</span><span class="va">motif</span>, <span class="va">expression</span>, <span class="va">ppi</span>, hamming<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">panda.res2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">pandaToyData</span>, <span class="fu"><a href="https://rdrr.io/pkg/pandaR/man/panda.html" class="external-link">panda</a></span><span class="op">(</span><span class="va">motif</span>, <span class="va">expression</span> <span class="op">+</span> </span>
<span>                   <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/prod.html" class="external-link">prod</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">expression</span><span class="op">)</span><span class="op">)</span>,sd<span class="op">=</span><span class="fl">5</span><span class="op">)</span>, <span class="va">ppi</span>, hamming<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pandaR/man/plotZ.html" class="external-link">plotZ</a></span><span class="op">(</span><span class="va">panda.res1</span>, <span class="va">panda.res2</span>,addLine<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by</span></span>
<span><span class="co">## the caller; using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by</span></span>
<span><span class="co">## the caller; using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by</span></span>
<span><span class="co">## the caller; using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(X[[i]], ...): 'as.is' should be specified by</span></span>
<span><span class="co">## the caller; using TRUE</span></span></code></pre>
<p><img src="pandaR_files/figure-html/plotZ-1.png" width="700"></p>
<p>Other comparisons are underway…</p>
</div>
<div class="section level2">
<h2 id="other-helpful-functions">Other helpful functions<a class="anchor" aria-label="anchor" href="#other-helpful-functions"></a>
</h2>
<p>There are a host of other helpful functions, including
<em>testMotif</em>, <em>plotCommunityDetection</em>, and
<em>multiplot</em>. See the full help pages.</p>
<p>##References</p>
<p>Glass K, Huttenhower C, Quackenbush J, Yuan GC. Passing Messages
Between Biological Networks to Refine Predicted Interactions, <em>PLoS
One</em>, 2013 May 31;8(5):e64832</p>
<p>Glass K, Quackenbush J, Silverman EK, Celli B, Rennard S, Yuan GC and
DeMeo DL. Sexually-dimorphic targeting of functionally-related genes in
COPD, <em>BMC Systems Biology</em>, 2014 Nov 28;
<strong>8</strong>:118</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill, Kate Shutta.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
