<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="netZooR">
<title>SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations • netZooR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations">
<meta property="og:description" content="netZooR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">netZooR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.17</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    <a class="dropdown-item" href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    <a class="dropdown-item" href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    <a class="dropdown-item" href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    <a class="dropdown-item" href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    <a class="dropdown-item" href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    <a class="dropdown-item" href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    <a class="dropdown-item" href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    <a class="dropdown-item" href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    <a class="dropdown-item" href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/netZoo/netZooR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</h1>
                        <h4 data-toc-skip class="author">Marieke L.
Kuijjer</h4>
            
            <h4 data-toc-skip class="date">2018-06-05</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/netZoo/netZooR/blob/HEAD/vignettes/SAMBAR.Rmd" class="external-link"><code>vignettes/SAMBAR.Rmd</code></a></small>
      <div class="d-none name"><code>SAMBAR.Rmd</code></div>
    </div>

    
    
<p>SAMBAR, or Subtyping Agglomerated Mutations By Annotation Relations,
is a method to identify subtypes based on somatic mutation data. SAMBAR
was used to identify mutational subtypes in 23 cancer types from The
Cancer Genome Atlas (Kuijjer ML, Paulson JN, Salzman P, Ding W,
Quackenbush J, British Journal of Cancer (May 16, 2018), doi:
10.1038/s41416-018-0109-7, <a href="https://www.nature.com/articles/s41416-018-0109-7" class="external-link uri">https://www.nature.com/articles/s41416-018-0109-7</a>,
BioRxiv, doi: <a href="https://doi.org/10.1101/228031" class="external-link uri">https://doi.org/10.1101/228031</a>).</p>
<p>SAMBAR’s input is a matrix that includes the number of non-synonymous
mutations in a sample <span class="math inline">\(i\)</span> and gene
<span class="math inline">\(j\)</span> . SAMBAR first subsets these data
to a set of 2,219 cancer-associated genes (optional) from the Catalogue
Of Somatic Mutations In Cancer (COSMIC) and Östlund et
al. (Network-based identification of novel cancer genes, 2010, Mol Cell
Prot), or from a user-defined list. It then divides the number of
non-synonymous mutations by the gene’s length <span class="math inline">\(L_j\)</span>, defined as the number of
non-overlapping exonic base pairs of a gene. For each sample, SAMBAR
then calculates the overall cancer-associated mutation rate by summing
mutation scores in all cancer-associated genes <span class="math inline">\(j'\)</span> . It removes samples for which the
mutation rate is zero and divides the mutation scores the remaining
samples by the sample’s mutation rate, resulting in a matrix of mutation
rate-adjusted scores <span class="math inline">\(G\)</span>:</p>
<p><span class="math inline">\(G_{ij}=\frac{N_{ij}/L_{j}}{\displaystyle\sum_{j'}({N_{ij'}/L_{j'}})}\)</span></p>
<p>The next step in SAMBAR is de-sparsification of these gene mutation
scores (agglomerated mutations) into pathway mutation (annotation
relation) scores. SAMBAR converts a (user-defined) gene signature (.gmt
format) into a binary matrix <span class="math inline">\(M\)</span> ,
with information of whether a gene <span class="math inline">\(j\)</span> belongs to a pathway <span class="math inline">\(q\)</span> . It then calculates pathway mutation
scores <span class="math inline">\(P\)</span> by correcting the sum of
mutation scores of all genes in a pathway for the number of pathways
<span class="math inline">\(q'\)</span> a gene belongs to, and for
the number of cancer-associated genes present in that pathway:</p>
<p><span class="math inline">\(P_{iq}=\frac{\displaystyle\sum_{j \in q}
G_{ij}/{\displaystyle\sum_{q'} M_{jq'}}}{\displaystyle\sum_{j}
M_{jq}}\)</span></p>
<p>Finally, SAMBAR uses binomial distance to cluster the pathway
mutation scores. The cluster dendrogram is then divided into <span class="math inline">\(k\)</span> groups (or a range of <span class="math inline">\(k\)</span> groups), and the cluster assignments
are returned in a list.</p>
<div class="section level2">
<h2 id="example-subtyping-uterine-corpus-endometrial-carcinoma">Example: Subtyping Uterine Corpus Endometrial Carcinoma<a class="anchor" aria-label="anchor" href="#example-subtyping-uterine-corpus-endometrial-carcinoma"></a>
</h2>
<p>As an example, we’ve added mutation data of Uterine Corpus
Endometrial Carcinoma (UCEC) primary tumor samples, obtained from The
Cancer Genome Atlas, to this package in <code>data(mut.ucec)</code>.
We’ve also added gene lengths (the number of non-overlapping exonic base
pairs) for 23,459 genes (hg19) in <code>data(exon.size)</code>. We’ve
added a list of cancer-associated genes in <code>data(gene)</code>.
Finally, the package includes a .gmt file that contains Hallmarks
pathways from MSigDb.</p>
<p>To run SAMBAR with default settings on these data:
<code>subtypes &lt;- sambar()</code></p>
<p>Or: <br>Download the signatureset from our public AWS S3 bucket by
using<code>system("curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/h.all.v6.1.symbols.gmt")</code>
and call SAMBAR by using the path where the signatureset is stored,
<code>subtypes &lt;- sambar(mutdata=mut.ucec, esize=exon.size, signatureset="./h.all.v6.1.symbols.gmt"), cangenes=genes, kmin=2, kmax=4)</code></p>
<p>This will run SAMBAR on the UCEC mutation data, using
desparsification on cancer-associated genes based on the MSigDb
“Hallmark” gene sets. It will return a list of samples belonging to
<span class="math inline">\(k=2-4\)</span> subtypes.</p>
<p>Instead of using the default signatureset, you can download any .gmt
file from MSigDb and add the path to this file in
<code>signatureset=PATH_TO_FILE</code>.</p>
</div>
<div class="section level2">
<h2 id="notes">Notes<a class="anchor" aria-label="anchor" href="#notes"></a>
</h2>
<p>Please note that, in order to be able to cluster the data using the
binomial distance, SAMBAR removes samples that have mutation scores of 0
across all pathways, and pathways that have mutation scores of 0 across
all samples. Using a small number of genes of interest
(<code>cangenes</code>) or a pathways that include small numbers of
genes could result in errors. We have not yet added any user-friendly
error checks to the package but plan to do this in the future.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill, Kate Shutta.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
