<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER • netZooR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER">
<meta property="og:description" content="netZooR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    </li>
    <li>
      <a href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    </li>
    <li>
      <a href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    </li>
    <li>
      <a href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    </li>
    <li>
      <a href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    </li>
    <li>
      <a href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="TutorialOTTER_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</h1>
                        <h4 class="author">Rebekka Burkholz</h4>
            
            <h4 class="date">2020-9-22</h4>
      
      
      <div class="hidden name"><code>TutorialOTTER.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>In this tutorial, we will learn how to infer OTTER gene regulatory networks from gene expression data with netZooR. It is an adaptation of the PANDA tutorial written by Camila Lopes-Ramos. We will focus on two examples: (1) the LCL cell line and (2) whole blood samples from the GTEx gene expression data. OTTER has been developed from a machine learning perspective in close analogy to PANDA. For this reason, we will follow the same steps as in the respective tutorial for PANDA.</p>
<p>First, we will build one regulatory network for LCL cell line samples and one for whole blood samples. Next, we will compare the two networks by a pathway enrichment analysis for differentially targeted genes.</p>
<p>Cell lines are an essential tool in biomedical research and are often used as surrogates for tissues. LCLs (obtained from the transformation of B cells present in whole blood) are among the most widely used continuous cell lines with the ability to proliferate indefinitely. By comparing the regulatory networks of LCL cell lines with its tissue of origin (whole blood), we find that LCLs exhibit large changes in their patterns of transcription factor regulation, specifically a loss of repressive transcription factor targeting of cell cycle genes.</p>
<div id="install-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#install-packages" class="anchor"></a>Install packages</h2>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span>,repos <span class="op">=</span> <span class="st">"http://cran.us.r-project.org"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"fgsea"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"reshape2"</span>,repos <span class="op">=</span> <span class="st">"http://cran.us.r-project.org"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"ggplot2"</span>,repos <span class="op">=</span> <span class="st">"http://cran.us.r-project.org"</span><span class="op">)</span></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"netZoo/netZooR"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></pre></div>
</div>
<div id="load-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h2>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">netZooR</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ctlab/fgsea/">fgsea</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hadley/reshape">reshape2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://r-datatable.com">data.table</a></span><span class="op">)</span></pre></div>
</div>
</div>
<div id="otter" class="section level1">
<h1 class="hasAnchor">
<a href="#otter" class="anchor"></a>OTTER</h1>
<div id="background-of-otter" class="section level2">
<h2 class="hasAnchor">
<a href="#background-of-otter" class="anchor"></a>Background of OTTER</h2>
<p>OTTER (Optimize to Estimate Regulation) is a general method to infer a bipartite network <span class="math inline">\(W\)</span> from noisy observations of its projections <span class="math inline">\(WW^T\)</span> and <span class="math inline">\(W^TW\)</span>. It is explained in detail in the accompanying publication <a href="https://www.biorxiv.org/content/10.1101/2020.06.23.167999v2" class="uri">https://www.biorxiv.org/content/10.1101/2020.06.23.167999v2</a>. In this tutorial, we are particularly interested in constructing a gene regulatory network <span class="math inline">\(W\)</span> between transcription factors (TFs) and genes. Higher link weights are associated with a higher probability of TFs binding to the promoter region of a gene. <img src="https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/genereg.png" alt="Inference of a bipartite gene regulatory network between transcription factors (TFs) and genes."></p>
<p>OTTER requires the following inputs: (1) a correlation matrix <span class="math inline">\(C\)</span>, which is based on gene expression data, (2) protein-protein interactions represented by the matrix <span class="math inline">\(P\)</span>, and (3) an initial guess <span class="math inline">\(W_0\)</span> of <span class="math inline">\(W\)</span>, which we base on TF binding motifs. Feel free to play with other choices of <span class="math inline">\(C\)</span>, <span class="math inline">\(P\)</span>, and <span class="math inline">\(W_0\)</span> as well.</p>
<p>OTTER solves the following optimization problem with ADAM gradient descent: <span class="math display">\[\min_W \frac{(1-\lambda)}{4}\Vert WW^T - \tilde{P} \Vert^2 + \frac{\lambda}{4} \Vert W^TW - C \Vert^2 + \frac{\gamma}{2} \Vert W \Vert^2,\]</span> which links transformed protein-protein interactions <span class="math inline">\(\tilde{P} = P+2.2\)</span> and the gene expression correlation matrix <span class="math inline">\(C\)</span> with the projections of the unknown gene regulatory network <span class="math inline">\(W\)</span>. <img src="https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/BipartiteProjections.png" alt="OTTER infers the gene regulatory network W assuming that P and C are its (noise corrupted) projections."></p>
<p>Gradient descent is an iterative optimization method, which needs to be initialized. OTTER starts from <span class="math inline">\(\tilde{P}W_0\)</span>, as this choice defines the current state-of-the-art in gene regulatory network inference based on gene expression data (see the original paper). The transformation <span class="math inline">\(\tilde{P}W_0\)</span> assumes that binding events to genes with a high number of TF bindings sites are more likely. More details can be found in the original paper.</p>
</div>
<div id="otter-parameters" class="section level2">
<h2 class="hasAnchor">
<a href="#otter-parameters" class="anchor"></a>OTTER parameters</h2>
<p>The success of OTTER depends greatly on the right choice of parameters. Some parameters are related to the OTTER objective, while others refer to the ADAM gradient descent approach. If we want to call OTTER with the original parameters that have been tuned to infer gene regulatory networks for breast cancer and cervix cancer tissues (see original paper), we can simply call the otter function as <span class="math inline">\(W &lt;- otter(W_0,P,C)\)</span>. This sets the original parameters to <span class="math inline">\(\lambda = 0.0035, \gamma = 0.335, Iter = 32, \eta = 0.00001, bexp = 1\)</span>, where <span class="math inline">\(\lambda \in [0,1]\)</span> is a tuning parameter in the OTTER objective that decides how much importance we give to matching <span class="math inline">\(C\)</span> or <span class="math inline">\(\tilde{P}\)</span>. <span class="math inline">\(\gamma \geq 0\)</span> is a regularization parameter in the OTTER objective, which corrects for high noise in <span class="math inline">\(\tilde{P}\)</span> and <span class="math inline">\(C\)</span>. The parameters <span class="math inline">\(Iter = 32, \eta = 0.00001, bexp = 1\)</span> refer to the gradient descent procedure, where <span class="math inline">\(Iter\)</span> controls the number of gradient steps, <span class="math inline">\(\eta\)</span> the stepsize, and <span class="math inline">\(bexp\)</span> the exponential decay of the stepsize. In the examples that we study next, we will only use the default parameters for simplicity.</p>
</div>
<div id="otter-network-inference" class="section level2">
<h2 class="hasAnchor">
<a href="#otter-network-inference" class="anchor"></a>OTTER Network Inference</h2>
<p>First, we have to define the input matrices <span class="math inline">\(P\)</span>, <span class="math inline">\(C\)</span>, and <span class="math inline">\(W_0\)</span>. Let’s start with <span class="math inline">\(P\)</span> and <span class="math inline">\(W_0\)</span> and locate our ppi data (to construct P) and motif priors (for W_0). The ppi represents physical interactions between transcription factor proteins, and is an undirected network. The transcription factor motif prior represents putative regulation events where a transcription factor binds in the promotor of a gene to regulate its expression, as predicted by the presence of transcription factor binding motifs in the promotor region of the gene. The motif prior is thus a directed bipartite network linking transcription factors to their predicted gene targets. These are small example priors for the purposes of demonstrating this method. A complete set of priors by species can be downloaded from: <a href="https://sites.google.com/a/channing.harvard.edu/kimberlyglass/tools/resources" class="uri">https://sites.google.com/a/channing.harvard.edu/kimberlyglass/tools/resources</a></p>
<p>Let’s take a look at the priors:</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="co"># download motif and ppi file from AWS Bucket</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.html">system</a></span><span class="op">(</span><span class="st">"curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/motif_subset.txt"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.html">system</a></span><span class="op">(</span><span class="st">"curl -O https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/ppi_subset.txt"</span><span class="op">)</span>
<span class="va">motif</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="st">"./motif_subset.txt"</span>, stringsAsFactors<span class="op">=</span><span class="cn">F</span>, header<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">ppi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="st">"./ppi_subset.txt"</span>, stringsAsFactors<span class="op">=</span><span class="cn">F</span>, header<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">ppi</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span>
<span class="co">#&gt;      V1      V2    V3</span>
<span class="co">#&gt; 1 ASCL1   ASCL1 1.000</span>
<span class="co">#&gt; 2 ASCL1 BHLHE41 0.314</span>
<span class="co">#&gt; 3 ASCL1    DLX3 0.188</span>
<span class="co">#&gt; 4 ASCL1     EN2 0.790</span>
<span class="co">#&gt; 5 ASCL1    HES4 0.451</span>
<span class="va">motif</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span>
<span class="co">#&gt;      V1     V2 V3</span>
<span class="co">#&gt; 1 ASCL1  RPAP3  0</span>
<span class="co">#&gt; 2 ASCL1  ACSM3  0</span>
<span class="co">#&gt; 3 ASCL1 CREBBP  0</span>
<span class="co">#&gt; 4 ASCL1 ZNF263  0</span>
<span class="co">#&gt; 5 ASCL1   RALA  0</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">ppi</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 430   3</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">motif</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 49950     3</span></pre></div>
<p>Next we locate our expression data and filter out genes, which are not expressed in too many samples. As example, we will use a subset of the GTEx version 7 RNA-Seq data, downloaded from <a href="https://gtexportal.org/home/datasets" class="uri">https://gtexportal.org/home/datasets</a>. We start with a subset of RNA-Seq data (tpm normalized) for 1,000 genes from 130 LCL cell line samples and 407 whole blood samples.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="co"># dowload and load the GTEx expression matrix (tpm normalized expression)</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.html">system</a></span><span class="op">(</span><span class="st">"curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/expression_tpm_lcl_blood_subset.txt"</span><span class="op">)</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="st">"./expression_tpm_lcl_blood_subset.txt"</span>, stringsAsFactors <span class="op">=</span> <span class="cn">F</span>, check.names <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="co"># Log transform the tpm normalized expression</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span><span class="op">(</span><span class="va">exp</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span>
<span class="co"># Determine the number of non-NA/non-zero rows in the expression data. This ensures that we have enough values in the vectors to calculate pearson correlations between gene expression profiles in the construction of the gene co-expression prior.</span>
<span class="va">zero_na_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">exp</span>, MARGIN <span class="op">=</span> <span class="fl">1</span>, FUN <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&amp;</span> <span class="va">x</span><span class="op">!=</span><span class="fl">0</span><span class="op">)</span> <span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="co"># Maintain only genes with at least 20 valid gene expression entries</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="va">exp</span><span class="op">[</span><span class="va">zero_na_counts</span> <span class="op">&gt;</span> <span class="fl">20</span>,<span class="op">]</span>
<span class="co"># The set of genes in the expression, motif and ppi matrices must be the same</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="va">exp</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">motif</span><span class="op">$</span><span class="va">V2</span>,<span class="op">]</span>
<span class="va">motif_subset</span> <span class="op">&lt;-</span> <span class="va">motif</span><span class="op">[</span><span class="op">(</span><span class="va">motif</span><span class="op">$</span><span class="va">V1</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span><span class="va">motif</span><span class="op">$</span><span class="va">V2</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">ppi_subset</span> <span class="op">&lt;-</span> <span class="va">ppi</span><span class="op">[</span><span class="op">(</span><span class="va">ppi</span><span class="op">$</span><span class="va">V1</span> <span class="op">%in%</span> <span class="va">motif_subset</span><span class="op">$</span><span class="va">V1</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span><span class="va">ppi</span><span class="op">$</span><span class="va">V2</span> <span class="op">%in%</span> <span class="va">motif_subset</span><span class="op">$</span><span class="va">V1</span><span class="op">)</span>,<span class="op">]</span>
<span class="co"># Load the sample ids of LCL samples</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.html">system</a></span><span class="op">(</span><span class="st">"curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/LCL_samples.txt"</span><span class="op">)</span>
<span class="va">lcl_samples</span> <span class="op">&lt;-</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span><span class="st">"./LCL_samples.txt"</span>, header <span class="op">=</span> <span class="cn">FALSE</span>, data.table<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="co"># Select the columns of the expression matrix corresponding to the LCL samples</span>
<span class="va">lcl_exp</span> <span class="op">&lt;-</span> <span class="va">exp</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">lcl_samples</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="co"># Load the sample ids of whole blood samples</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.html">system</a></span><span class="op">(</span><span class="st">"curl -O https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/WholeBlood_samples.txt"</span><span class="op">)</span>
<span class="va">wblood_samples</span> <span class="op">&lt;-</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span><span class="st">"./WholeBlood_samples.txt"</span>, header <span class="op">=</span> <span class="cn">FALSE</span>, data.table<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="co"># Select the columns of the expression matrix corresponding to the whole blood samples</span>
<span class="va">wb_exp</span> <span class="op">&lt;-</span> <span class="va">exp</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">wblood_samples</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">]</span></pre></div>
<p>For use in OTTER, we have to transform the edgelists into matrices and compute correlation matrices</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="co">#names of genes and transcription factors (tf)</span>
<span class="va">geneNames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html">unique</a></span><span class="op">(</span><span class="va">motif_subset</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="va">tfNames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html">unique</a></span><span class="op">(</span><span class="va">motif_subset</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="co">#number of genes and tf</span>
<span class="va">ng</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">geneNames</span><span class="op">)</span>
<span class="va">ntf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">tfNames</span><span class="op">)</span>
<span class="co">#initial motif prior:</span>
<span class="va">W0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span>data<span class="op">=</span><span class="fl">0</span>, nrow <span class="op">=</span> <span class="va">ntf</span>, ncol<span class="op">=</span><span class="va">ng</span>, dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">tfNames</span>, <span class="va">geneNames</span><span class="op">)</span><span class="op">)</span>
<span class="va">W0</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/cbind.html">cbind</a></span><span class="op">(</span><span class="va">motif_subset</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>, <span class="va">motif_subset</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">motif_subset</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span>
<span class="co">#protein-protein interaction matrix</span>
<span class="va">P</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span>data<span class="op">=</span><span class="fl">0</span>, nrow <span class="op">=</span> <span class="va">ntf</span>, ncol<span class="op">=</span><span class="va">ntf</span>, dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">tfNames</span>, <span class="va">tfNames</span><span class="op">)</span><span class="op">)</span>
<span class="va">P</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/cbind.html">cbind</a></span><span class="op">(</span><span class="va">ppi_subset</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,<span class="va">ppi_subset</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">ppi_subset</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span>
<span class="co">#correlation matrices</span>
<span class="va">C_lcl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">lcl_exp</span><span class="op">)</span><span class="op">)</span>
<span class="va">C_wb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">wb_exp</span><span class="op">)</span><span class="op">)</span></pre></div>
<p>Now we can run OTTER. We want to generate two gene regulatory networks for comparison, one based on the LCL and one based on the whole blood data. Hence, we also have to run OTTER twice. Note that the matrices <span class="math inline">\(P\)</span> and <span class="math inline">\(W_0\)</span> are identical in each run. The input to OTTER only differ in the correlation matrix <span class="math inline">\(C_{lcl}\)</span> or <span class="math inline">\(C_{wb}\)</span>, respectively.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="va">otterLCL</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/otter.html">otter</a></span><span class="op">(</span><span class="va">W0</span>, <span class="va">P</span>, <span class="va">C_lcl</span><span class="op">)</span>
<span class="va">otterWB</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/otter.html">otter</a></span><span class="op">(</span><span class="va">W0</span>, <span class="va">P</span>, <span class="va">C_wb</span><span class="op">)</span></pre></div>
<p>OTTER networks have relatively small weights because of the internal normalization of edges. But the scaling does not matter for the prediction of regulatory links. The higher the edge weight, the higher is the probability that a transcription factor binds to the promoter region of a gene and regulates its expression. For convenience, we multiply all weights with a factor that sets the maximum weight to one.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="va">otterLCL</span> <span class="op">&lt;-</span> <span class="va">otterLCL</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">otterLCL</span><span class="op">)</span>
<span class="va">otterWB</span> <span class="op">&lt;-</span> <span class="va">otterWB</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">otterWB</span><span class="op">)</span></pre></div>
</div>
</div>
<div id="visualizing-networks-in-cytoscape" class="section level1">
<h1 class="hasAnchor">
<a href="#visualizing-networks-in-cytoscape" class="anchor"></a>Visualizing networks in Cytoscape</h1>
<p>In this section we will visualize parts of the network using the Cytoscape software. Download Cytoscape from: <a href="https://cytoscape.org" class="uri">https://cytoscape.org</a></p>
<div id="preparing-data-to-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#preparing-data-to-plot" class="anchor"></a>Preparing data to plot</h2>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="co"># We will use the function vis.panda.in.cytoscape to plot a set of nodes and edges on Cytoscape. The input for this function is a data.frame of edges to plot with 4 columns: "tf", "gene", "motif" (TF motif present or not on gene promoter), "force" (edge weight calculated by OTTER).</span>
<span class="va">lcl_vis</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span><span class="op">(</span><span class="va">otterLCL</span><span class="op">)</span>
<span class="co">#&gt; Warning in melt(otterLCL): The melt generic in data.table has been passed a</span>
<span class="co">#&gt; matrix and will attempt to redirect to the relevant reshape2 method; please note</span>
<span class="co">#&gt; that reshape2 is deprecated, and this redirection is now deprecated as well.</span>
<span class="co">#&gt; To continue using melt methods from reshape2 while both libraries are attached,</span>
<span class="co">#&gt; e.g. melt.list, you can prepend the namespace like reshape2::melt(otterLCL). In</span>
<span class="co">#&gt; the next version, this warning will become an error.</span>
<span class="va">wb_vis</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span><span class="op">(</span><span class="va">otterWB</span><span class="op">)</span>
<span class="co">#&gt; Warning in melt(otterWB): The melt generic in data.table has been passed a</span>
<span class="co">#&gt; matrix and will attempt to redirect to the relevant reshape2 method; please note</span>
<span class="co">#&gt; that reshape2 is deprecated, and this redirection is now deprecated as well.</span>
<span class="co">#&gt; To continue using melt methods from reshape2 while both libraries are attached,</span>
<span class="co">#&gt; e.g. melt.list, you can prepend the namespace like reshape2::melt(otterWB). In</span>
<span class="co">#&gt; the next version, this warning will become an error.</span>
<span class="va">lcl_vis</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="st">"TF"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">lcl_vis</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span>,<span class="st">"Gene"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">lcl_vis</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span>,<span class="st">"Motif"</span><span class="op">=</span><span class="cn">NA</span>,<span class="st">"Score"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">lcl_vis</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span><span class="op">)</span>,stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">wb_vis</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="st">"TF"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">wb_vis</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span>,<span class="st">"Gene"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">wb_vis</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span>,<span class="st">"Motif"</span><span class="op">=</span><span class="cn">NA</span>,<span class="st">"Score"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">wb_vis</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span><span class="op">)</span>,stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">lcl_vis</span><span class="op">)</span>
<span class="co">#&gt;        TF  Gene Motif     Score</span>
<span class="co">#&gt; 1   ASCL1 RPAP3    NA 0.9617341</span>
<span class="co">#&gt; 2   BARX1 RPAP3    NA 0.9616993</span>
<span class="co">#&gt; 3 BHLHE41 RPAP3    NA 0.9617121</span>
<span class="co">#&gt; 4   CEBPD RPAP3    NA 0.9617139</span>
<span class="co">#&gt; 5   CEBPZ RPAP3    NA 0.9617043</span>
<span class="co">#&gt; 6   CENPB RPAP3    NA 0.9617101</span></pre></div>
</div>
<div id="plot-the-200-highest-edge-weights" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-200-highest-edge-weights" class="anchor"></a>Plot the 200 highest edge weights</h2>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="va">n</span><span class="op">=</span><span class="fl">200</span> <span class="co"># number of edges to plot </span>
<span class="va">top</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html">order</a></span><span class="op">(</span><span class="va">lcl_vis</span><span class="op">$</span><span class="va">Score</span>,decreasing<span class="op">=</span><span class="cn">T</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n</span><span class="op">]</span>
<span class="va">lcl_vis_top</span> <span class="op">&lt;-</span> <span class="va">lcl_vis</span><span class="op">[</span><span class="va">top</span>,<span class="op">]</span>
<span class="co"># Plot in cytoscape (open the cytoscape software before running this command)</span>
<span class="fu"><a href="../reference/vis.panda.in.cytoscape.html">vis.panda.in.cytoscape</a></span><span class="op">(</span><span class="va">lcl_vis_top</span>, network_name<span class="op">=</span><span class="st">"LCL"</span><span class="op">)</span>
<span class="co"># Here we will load a customized visual style for our network, in which TF nodes are orange circles, target gene nodes are blue squares, and edges shade and width are the edge weight (likelyhood of regulatory interaction between the TF and gene). You can further customize the network style directly from Cytoscape.</span>
<span class="fu"><a href="../reference/create.panda.style.html">create.panda.style</a></span><span class="op">(</span>style_name<span class="op">=</span><span class="st">"PandaStyle"</span><span class="op">)</span></pre></div>
</div>
<div id="plot-the-top-differential-edges-betwen-lcl-and-wb" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-top-differential-edges-betwen-lcl-and-wb" class="anchor"></a>Plot the top differential edges betwen LCL and WB</h2>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="co"># Select the top differential edge weights betweeen LCL and whole blood</span>
<span class="va">diffRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/panda.diff.edges.html">panda.diff.edges</a></span><span class="op">(</span><span class="va">lcl_vis</span>, <span class="va">wb_vis</span>, condition_name<span class="op">=</span><span class="st">"LCL"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">diffRes</span><span class="op">)</span>
<span class="co"># Number of differential edges is:</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">diffRes</span><span class="op">)</span>
<span class="co"># Select the top differential edges higher in LCL to plot in Cytoscape</span>
<span class="va">n</span><span class="op">=</span><span class="fl">200</span> <span class="co"># number of edges to select from each condition</span>
<span class="va">diffResLCL</span> <span class="op">&lt;-</span> <span class="va">diffRes</span><span class="op">[</span><span class="va">diffRes</span><span class="op">$</span><span class="va">LCL</span><span class="op">==</span><span class="st">"T"</span>,<span class="op">]</span>
<span class="va">diffResLCL</span> <span class="op">&lt;-</span> <span class="va">diffResLCL</span><span class="op">[</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html">order</a></span><span class="op">(</span><span class="va">diffResLCL</span><span class="op">$</span><span class="va">Score</span>,decreasing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n</span>,<span class="op">]</span>
<span class="co"># Select the top differential edges higher in whole blood to plot in Cytoscape</span>
<span class="va">diffResWB</span> <span class="op">&lt;-</span> <span class="va">diffRes</span><span class="op">[</span><span class="va">diffRes</span><span class="op">$</span><span class="va">LCL</span><span class="op">==</span><span class="st">"F"</span>,<span class="op">]</span>
<span class="va">diffResWB</span> <span class="op">&lt;-</span> <span class="va">diffResWB</span><span class="op">[</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html">order</a></span><span class="op">(</span><span class="va">diffResWB</span><span class="op">$</span><span class="va">Score</span>,decreasing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n</span>,<span class="op">]</span>
<span class="co"># Combine top differential edges in LCL and WB to plot in Cytoscape</span>
<span class="va">diffRes_vis</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/rbindlist.html">rbind</a></span><span class="op">(</span><span class="va">diffResLCL</span>, <span class="va">diffResWB</span><span class="op">)</span>
<span class="co"># Plot the network</span>
<span class="co"># Purple edges indicate higher edge weight in the defined "condition_name" parameter (LCL in our example), and green edges indicate higher edge weight in the other condition (whole blood in our example).</span>
<span class="fu"><a href="../reference/vis.diff.panda.in.cytoscape.html">vis.diff.panda.in.cytoscape</a></span><span class="op">(</span><span class="va">diffRes_vis</span>, condition_name <span class="op">=</span> <span class="st">"LCL"</span>, network_name<span class="op">=</span><span class="st">"diff.PANDA"</span><span class="op">)</span>
<span class="co"># Apply the style to the network</span>
<span class="fu"><a href="../reference/create.diff.panda.style.html">create.diff.panda.style</a></span><span class="op">(</span>style_name<span class="op">=</span><span class="st">"Diff.PandaStyle"</span>, condition_name<span class="op">=</span><span class="st">"LCL"</span><span class="op">)</span></pre></div>
</div>
</div>
<div id="calculating-degree" class="section level1">
<h1 class="hasAnchor">
<a href="#calculating-degree" class="anchor"></a>Calculating degree</h1>
<ul>
<li>out-degrees of TFs: sum of the weights of edges pointing to a TF</li>
<li>in-degrees of genes: sum of the weights of edges pointing to a gene</li>
</ul>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="va">lcl_outdegree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">otterLCL</span>, <span class="fl">1</span>, <span class="va">sum</span><span class="op">)</span> 
<span class="va">wb_outdegree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">otterWB</span>, <span class="fl">1</span>, <span class="va">sum</span><span class="op">)</span> 
<span class="va">lcl_indegree</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">otterLCL</span>, <span class="fl">2</span>, <span class="va">sum</span><span class="op">)</span> 
<span class="va">wb_indegree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">otterWB</span>, <span class="fl">2</span>, <span class="va">sum</span><span class="op">)</span> 
<span class="va">degreeDiff</span> <span class="op">&lt;-</span> <span class="va">lcl_indegree</span><span class="op">-</span><span class="va">wb_indegree</span> 
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">degreeDiff</span><span class="op">)</span>
<span class="co">#&gt;         RPAP3         ACSM3        CREBBP        ZNF263          RALA </span>
<span class="co">#&gt; -1.644496e-04  1.960062e-04 -1.709337e-04 -3.284675e-04 -6.335936e-05 </span>
<span class="co">#&gt;         PSMB1 </span>
<span class="co">#&gt; -3.213199e-04</span></pre></div>
</div>
<div id="gene-set-enrichment-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#gene-set-enrichment-analysis" class="anchor"></a>Gene Set Enrichment Analysis</h1>
<p>Well will use the fgsea package to perform a gene set enrichment analysis. Inputs are a ranked gene list (for example the gene in-degree difference between LCL and whole blood) and a list of gene sets (or signatures) in gmt format to test for enrichment. The gene sets can be downloaded from MSigDB: <a href="http://software.broadinstitute.org/gsea/msigdb" class="uri">http://software.broadinstitute.org/gsea/msigdb</a> The same gene annotation should be used in the ranked gene list and the gene sets. In our example, we will use the KEGG pathways downloaded from MSigDB.</p>
<div id="run-fgsea" class="section level2">
<h2 class="hasAnchor">
<a href="#run-fgsea" class="anchor"></a>Run fgsea</h2>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/system.html">system</a></span><span class="op">(</span><span class="st">"curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/c2.cp.kegg.v7.0.symbols.gmt"</span><span class="op">)</span>
<span class="va">pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/fgsea/man/gmtPathways.html">gmtPathways</a></span><span class="op">(</span><span class="st">"./c2.cp.kegg.v7.0.symbols.gmt"</span><span class="op">)</span>
<span class="co"># To retrieve biological-relevant processes, we will load and use the complete ranked gene list (consisting of 26,077 out of 27,174 genes). The in-degree difference has been calculated based on the complete networks instead of the subnetworks, which we constructed in this tutorial as small examples (with reduced run time).</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.html">system</a></span><span class="op">(</span><span class="st">"curl -O  https://netzoo.s3.us-east-2.amazonaws.com/netZooR/tutorial_datasets/lclWB_indegreeDifference_otter.txt"</span><span class="op">)</span>
<span class="va">degreeDiff_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="st">"./lclWB_indegreeDifference_otter.txt"</span>,stringsAsFactors <span class="op">=</span> <span class="cn">F</span>,header<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">degreeDiff_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="va">degreeDiff_all</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>, <span class="va">degreeDiff_all</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="va">fgseaRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/fgsea/man/fgsea.html">fgsea</a></span><span class="op">(</span><span class="va">pathways</span>, <span class="va">degreeDiff_all</span>, minSize<span class="op">=</span><span class="fl">15</span>, maxSize<span class="op">=</span><span class="fl">500</span>, nperm<span class="op">=</span><span class="fl">1000</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">fgseaRes</span><span class="op">)</span>
<span class="co">#&gt;                                          pathway        pval       padj</span>
<span class="co">#&gt; 1:               KEGG_GLYCOLYSIS_GLUCONEOGENESIS 0.002083333 0.01439493</span>
<span class="co">#&gt; 2:                  KEGG_CITRATE_CYCLE_TCA_CYCLE 0.208171206 0.35760025</span>
<span class="co">#&gt; 3:                KEGG_PENTOSE_PHOSPHATE_PATHWAY 0.071428571 0.15418118</span>
<span class="co">#&gt; 4: KEGG_PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS 0.446581197 0.62270415</span>
<span class="co">#&gt; 5:          KEGG_FRUCTOSE_AND_MANNOSE_METABOLISM 0.058467742 0.13099735</span>
<span class="co">#&gt; 6:                     KEGG_GALACTOSE_METABOLISM 0.381355932 0.55327869</span>
<span class="co">#&gt;            ES       NES nMoreExtreme size                          leadingEdge</span>
<span class="co">#&gt; 1:  0.4360719  1.768400            0   58    HK1,ENO2,ENO1,ALDOC,BPGM,PFKM,...</span>
<span class="co">#&gt; 2: -0.3428560 -1.187950          106   29  DLST,PCK2,SUCLG2P2,MDH2,FH,IDH2,...</span>
<span class="co">#&gt; 3:  0.4435710  1.404766           33   21   ALDOC,PFKM,PFKP,ALDOA,PFKL,GPI,...</span>
<span class="co">#&gt; 4:  0.3180477  1.014455          208   22   UGT1A1,UGT1A10,UGT1A6,CRYL1,UGT2B7</span>
<span class="co">#&gt; 5:  0.4118882  1.410368           28   31 HK1,ALDOC,PFKFB4,PFKM,MTMR2,PFKP,...</span>
<span class="co">#&gt; 6:  0.3247973  1.067120          179   25      GLB1,HK1,PFKM,PFKP,PFKL,GAA,...</span>
<span class="co"># Subset to pathways with FDR &lt; 0.05</span>
<span class="va">sig</span> <span class="op">&lt;-</span> <span class="va">fgseaRes</span><span class="op">[</span><span class="va">fgseaRes</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span>,<span class="op">]</span>
<span class="co"># Get the top 10 significant pathways enriched for genes having lower targeting in LCLs  </span>
<span class="va">sig</span><span class="op">[</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html">order</a></span><span class="op">(</span><span class="va">sig</span><span class="op">$</span><span class="va">NES</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>,<span class="op">]</span>
<span class="co">#&gt;                             pathway        pval       padj         ES       NES</span>
<span class="co">#&gt;  1:            KEGG_DNA_REPLICATION 0.001984127 0.01439493 -0.7725018 -2.791509</span>
<span class="co">#&gt;  2:                 KEGG_CELL_CYCLE 0.001845018 0.01439493 -0.5948628 -2.717545</span>
<span class="co">#&gt;  3:                KEGG_SPLICEOSOME 0.001831502 0.01439493 -0.5766563 -2.638763</span>
<span class="co">#&gt;  4:             KEGG_PROTEIN_EXPORT 0.001890359 0.01439493 -0.6851677 -2.236534</span>
<span class="co">#&gt;  5: KEGG_NUCLEOTIDE_EXCISION_REPAIR 0.001988072 0.01439493 -0.5785505 -2.192541</span>
<span class="co">#&gt;  6:       KEGG_BASE_EXCISION_REPAIR 0.001976285 0.01439493 -0.5965607 -2.137049</span>
<span class="co">#&gt;  7:      KEGG_PYRIMIDINE_METABOLISM 0.001904762 0.01439493 -0.4606911 -1.990125</span>
<span class="co">#&gt;  8:            KEGG_MISMATCH_REPAIR 0.003745318 0.02025635 -0.6107380 -1.977788</span>
<span class="co">#&gt;  9:             KEGG_RNA_POLYMERASE 0.001980198 0.01439493 -0.5731001 -1.959927</span>
<span class="co">#&gt; 10:      KEGG_N_GLYCAN_BIOSYNTHESIS 0.001956947 0.01439493 -0.4815544 -1.846545</span>
<span class="co">#&gt;     nMoreExtreme size                                    leadingEdge</span>
<span class="co">#&gt;  1:            0   34             RFC5,RFC3,FEN1,MCM7,POLE2,DNA2,...</span>
<span class="co">#&gt;  2:            0  121          ORC6,CDC7,CDC20,PKMYT1,CCNE1,ORC1,...</span>
<span class="co">#&gt;  3:            0  118          PHF5A,SRSF3,LSM7,SRSF7,LSM4,U2AF2,...</span>
<span class="co">#&gt;  4:            0   23       IMMP1L,SRP9,SRPRB,SPCS3,HSPA5,IMMP2L,...</span>
<span class="co">#&gt;  5:            0   41             RFC5,RFC3,POLE2,CCNH,RFC4,PCNA,...</span>
<span class="co">#&gt;  6:            0   33           FEN1,HMGB1,MUTYH,UNG,POLE2,NEIL3,...</span>
<span class="co">#&gt;  7:            0   90         POLR2D,DUT,POLE2,PRIM1,DTYMK,DHODH,...</span>
<span class="co">#&gt;  8:            1   22              RFC5,RFC3,RFC4,PCNA,EXO1,RFC2,...</span>
<span class="co">#&gt;  9:            0   28 POLR2D,POLR3B,POLR2H,POLR2L,POLR3GL,POLR2J,...</span>
<span class="co">#&gt; 10:            0   45         B4GALT2,DPM3,ALG14,FUT8,ALG6,MGAT2,...</span></pre></div>
</div>
<div id="bubble-plot-of-top-differentially-targeted-pathways" class="section level2">
<h2 class="hasAnchor">
<a href="#bubble-plot-of-top-differentially-targeted-pathways" class="anchor"></a>Bubble plot of top differentially targeted pathways</h2>
<p>Bubble plot of gene sets (KEGG pathways) on y-axis and adjusted p-value (padj) on x-axis. Bubble size indicates the number of genes in each gene set, and bubble color indicates the normalized enrichment score (NES). Blue is for negative NES (enrichment of higher targeted genes in whole blood), and red is for positive NES (enrichment of higher targeted genes in LCL).</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">fgseaRes</span><span class="op">)</span>
<span class="co"># Settings</span>
<span class="va">fdrcut</span> <span class="op">&lt;-</span> <span class="fl">0.05</span> <span class="co"># FDR cut-off to use as output for significant signatures</span>
<span class="va">dencol_neg</span> <span class="op">&lt;-</span> <span class="st">"blue"</span> <span class="co"># bubble plot color for negative ES</span>
<span class="va">dencol_pos</span> <span class="op">&lt;-</span> <span class="st">"red"</span> <span class="co"># bubble plot color for positive ES</span>
<span class="va">signnamelength</span> <span class="op">&lt;-</span> <span class="fl">4</span> <span class="co"># set to remove prefix from signature names (2 for "GO", 4 for "KEGG", 8 for "REACTOME")</span>
<span class="va">asp</span> <span class="op">&lt;-</span> <span class="fl">3</span> <span class="co"># aspect ratio of bubble plot</span>
<span class="va">charcut</span> <span class="op">&lt;-</span> <span class="fl">100</span> <span class="co"># cut signature name in heatmap to this nr of characters</span>
<span class="co"># Make signature names more readable</span>
<span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">dat</span><span class="op">$</span><span class="va">pathway</span><span class="op">)</span> <span class="co"># 'a' is a great variable name to substitute row names with something more readable</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="va">a</span><span class="op">[</span><span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">a</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>, <span class="va">signnamelength</span><span class="op">+</span><span class="fl">2</span>, <span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="va">a</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html">tolower</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span> <span class="co"># convert to lower case (you may want to comment this out, it really depends on what signatures you are looking at, c6 signatures contain gene names, and converting those to lower case may be confusing)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="va">a</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">)</span><span class="op">&gt;</span><span class="va">charcut</span><span class="op">)</span> <span class="op">{</span> <span class="va">a</span><span class="op">[</span><span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">a</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>, <span class="fl">1</span>, <span class="va">charcut</span><span class="op">)</span>, <span class="st">"..."</span>, sep<span class="op">=</span><span class="st">" "</span><span class="op">)</span><span class="op">}</span>
<span class="op">}</span> <span class="co"># cut signature names that have more characters than charcut, and add "..."</span>
<span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_"</span>, <span class="st">" "</span>, <span class="va">a</span><span class="op">)</span>
<span class="va">dat</span><span class="op">$</span><span class="va">NAME</span> <span class="op">&lt;-</span> <span class="va">a</span>
<span class="co"># Determine what signatures to plot (based on FDR cut)</span>
<span class="va">dat2</span> <span class="op">&lt;-</span> <span class="va">dat</span><span class="op">[</span><span class="va">dat</span><span class="op">[</span>,<span class="st">"padj"</span><span class="op">]</span><span class="op">&lt;</span><span class="va">fdrcut</span>,<span class="op">]</span>
<span class="va">dat2</span> <span class="op">&lt;-</span> <span class="va">dat2</span><span class="op">[</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html">order</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">[</span>,<span class="st">"padj"</span><span class="op">]</span><span class="op">)</span>,<span class="op">]</span> 
<span class="va">dat2</span><span class="op">$</span><span class="va">signature</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">$</span><span class="va">NAME</span>, <span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">$</span><span class="va">NAME</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co"># Determine what labels to color</span>
<span class="va">sign_neg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">[</span>,<span class="st">"NES"</span><span class="op">]</span><span class="op">&lt;</span><span class="fl">0</span><span class="op">)</span>
<span class="va">sign_pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">[</span>,<span class="st">"NES"</span><span class="op">]</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span>
<span class="co"># Color labels</span>
<span class="va">signcol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">$</span><span class="va">signature</span><span class="op">)</span><span class="op">)</span>
<span class="va">signcol</span><span class="op">[</span><span class="va">sign_neg</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">dencol_neg</span> <span class="co"># text color of negative signatures</span>
<span class="va">signcol</span><span class="op">[</span><span class="va">sign_pos</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">dencol_pos</span> <span class="co"># text color of positive signatures</span>
<span class="va">signcol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span><span class="op">(</span><span class="va">signcol</span><span class="op">)</span> <span class="co"># need to revert vector of colors, because ggplot starts plotting these from below</span>
<span class="co"># Plot bubble plot</span>
<span class="va">g</span><span class="op">&lt;-</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">dat2</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">padj</span>,y<span class="op">=</span><span class="va">signature</span>,size<span class="op">=</span><span class="va">size</span><span class="op">)</span><span class="op">)</span>
<span class="va">g</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">NES</span><span class="op">)</span>, shape<span class="op">=</span><span class="fl">21</span>, colour<span class="op">=</span><span class="st">"white"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span> <span class="co"># white background, needs to be placed before the "signcol" line</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="va">fdrcut</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html">scale_size_area</a></span><span class="op">(</span>max_size<span class="op">=</span><span class="fl">10</span>,guide<span class="op">=</span><span class="st">"none"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html">scale_fill_gradient2</a></span><span class="op">(</span>low<span class="op">=</span><span class="va">dencol_neg</span>, high<span class="op">=</span><span class="va">dencol_pos</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>colour<span class="op">=</span><span class="va">signcol</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="va">asp</span>, axis.title.y<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="co"># test aspect.ratio</span>
<span class="co">#&gt; Warning: Vectorized input to `element_text()` is not officially supported.</span>
<span class="co">#&gt; Results may be unexpected or may change in future versions of ggplot2.</span></pre></div>
<p><img src="TutorialOTTER_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Tian Wang, Marouen Ben Guebila, John Platig, Marieke Kuijjer, Magha Padi, Rebekka Burkholz, Deborah Weighill.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
