<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization • netZooR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization">
<meta property="og:description" content="netZooR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    </li>
    <li>
      <a href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    </li>
    <li>
      <a href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    </li>
    <li>
      <a href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    </li>
    <li>
      <a href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    </li>
    <li>
      <a href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/netZoo/netZooR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="yarn_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</h1>
                        <h4 class="author">Joseph N. Paulson &amp; John Quackenbush</h4>
            
            <h4 class="date">2022-06-16</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/netZoo/netZooR/blob/master/vignettes/yarn.Rmd"><code>vignettes/yarn.Rmd</code></a></small>
      <div class="hidden name"><code>yarn.Rmd</code></div>

    </div>

    
    
<div id="yarn---yet-another-rna-seq-package" class="section level2">
<h2 class="hasAnchor">
<a href="#yarn---yet-another-rna-seq-package" class="anchor"></a>YARN - Yet Another RNa-seq package</h2>
<p>The goal of yarn is to expedite large RNA-seq analyses using a combination of previously developed tools. Yarn is meant to make it easier for the user to perform accurate comparison of conditions by leveraging many Bioconductor tools and various statistical and normalization techniques while accounting for the large heterogeneity and sparsity found in very large RNA-seq experiments.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install yarn from github through netZooR with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"netZooR"</span><span class="op">)</span></code></pre></div>
</div>
<div id="quick-introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#quick-introduction" class="anchor"></a>Quick Introduction</h2>
<p>If you’re here to grab the GTEx version 6.0 data then look no further than this <a href="https://gist.github.com/jnpaulson/8c2ccfb0185dc490ff72e51aef86678c">gist</a> that uses yarn to download all the data and preprocess it for you.</p>
</div>
<div id="preprocessing" class="section level2">
<h2 class="hasAnchor">
<a href="#preprocessing" class="anchor"></a>Preprocessing</h2>
<p>Below are a few of the functions we can use to preprocess a large RNA-seq experiment. We follow a particular procedure where we:</p>
<ol style="list-style-type: decimal">
<li>Filter poor quality samples</li>
<li>Merge samples of similar conditions for increased power</li>
<li>Filter genes while preserving tissue or group specificity</li>
<li>Normalize while accounting for global differences in tissue distribution</li>
</ol>
<p>We will make use of the <code>skin</code> dataset for examples. The <code>skin</code> dataset is a small sample of the full GTEx data that can be downloaded using the <code>downloadGTEx</code> function. The <code>skin</code> dataset looks like this:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">skin</span></code></pre></div>
<pre><code>## ExpressionSet (storageMode: lockedEnvironment)
## assayData: 40824 features, 20 samples 
##   element names: exprs 
## protocolData: none
## phenoData
##   rowNames: GTEX-OHPL-0008-SM-4E3I9 GTEX-145MN-1526-SM-5SI9T ...
##     GTEX-144FL-0626-SM-5LU43 (20 total)
##   varLabels: SAMPID SMATSSCR ... DTHHRDY (65 total)
##   varMetadata: labelDescription
## featureData
##   featureNames: 48350 48365 ... 7565 (40824 total)
##   fvarLabels: ensembl_gene_id hgnc_symbol ... gene_biotype (6 total)
##   fvarMetadata: labelDescription
## experimentData: use 'experimentData(object)'
## Annotation:</code></pre>
<p>This is a basic workflow. Details will be fleshed out:</p>
<ol start="0" style="list-style-type: decimal">
<li>First always remember to have the library loaded.</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">netZooR</span><span class="op">)</span></code></pre></div>
<ol style="list-style-type: decimal">
<li>Download the GTEx gene count data as an ExpressionSet object or load the sample skin dataset.</li>
</ol>
<p>For computational reasons we load the sample skin data instead of having the user download the</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">netZooR</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"skin.rdata"</span><span class="op">)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Check mis-annotation of gender or other phenotypes using group-specific genes</li>
</ol>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">yarn</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yarn/man/checkMisAnnotation.html">checkMisAnnotation</a></span><span class="op">(</span><span class="va">skin</span>,<span class="st">"GENDER"</span>,controlGenes<span class="op">=</span><span class="st">"Y"</span>,legendPosition<span class="op">=</span><span class="st">"topleft"</span><span class="op">)</span></code></pre></div>
<p><img src="yarn_files/figure-html/checkMisAnnotation-1.png" width="700"></p>
<ol start="3" style="list-style-type: decimal">
<li>Decide what sub-groups should be merged</li>
</ol>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">yarn</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yarn/man/checkTissuesToMerge.html">checkTissuesToMerge</a></span><span class="op">(</span><span class="va">skin</span>,<span class="st">"SMTS"</span>,<span class="st">"SMTSD"</span><span class="op">)</span></code></pre></div>
<p><img src="yarn_files/figure-html/checkTissuesToMerge-1.png" width="700"></p>
<ol start="4" style="list-style-type: decimal">
<li>Filter lowly expressed genes</li>
</ol>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">skin_filtered</span> <span class="op">=</span> <span class="fu">yarn</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yarn/man/filterLowGenes.html">filterLowGenes</a></span><span class="op">(</span><span class="va">skin</span>,<span class="st">"SMTSD"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">skin</span><span class="op">)</span></code></pre></div>
<pre><code>## Features  Samples 
##    40824       20</code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">skin_filtered</span><span class="op">)</span></code></pre></div>
<pre><code>## Features  Samples 
##    19933       20</code></pre>
<p>Or group specific genes</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tmp</span> <span class="op">=</span> <span class="fu">yarn</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yarn/man/filterGenes.html">filterGenes</a></span><span class="op">(</span><span class="va">skin</span>,labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"X"</span>,<span class="st">"Y"</span>,<span class="st">"MT"</span><span class="op">)</span>,featureName <span class="op">=</span> <span class="st">"chromosome_name"</span><span class="op">)</span>
<span class="co"># Keep only the sex names</span>
<span class="va">tmp</span> <span class="op">=</span> <span class="fu">yarn</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yarn/man/filterGenes.html">filterGenes</a></span><span class="op">(</span><span class="va">skin</span>,labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"X"</span>,<span class="st">"Y"</span>,<span class="st">"MT"</span><span class="op">)</span>,featureName <span class="op">=</span> <span class="st">"chromosome_name"</span>,keepOnly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<ol start="5" style="list-style-type: decimal">
<li>Normalize in a tissue or group-aware manner</li>
</ol>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">yarn</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yarn/man/plotDensity.html">plotDensity</a></span><span class="op">(</span><span class="va">skin_filtered</span>,<span class="st">"SMTSD"</span>,main<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="st">'log'</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">*</span><span class="st">' raw expression'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="yarn_files/figure-html/density-1.png" width="700"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">skin_filtered</span> <span class="op">=</span> <span class="fu">yarn</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yarn/man/normalizeTissueAware.html">normalizeTissueAware</a></span><span class="op">(</span><span class="va">skin_filtered</span>,<span class="st">"SMTSD"</span><span class="op">)</span>
<span class="fu">yarn</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yarn/man/plotDensity.html">plotDensity</a></span><span class="op">(</span><span class="va">skin_filtered</span>,<span class="st">"SMTSD"</span>,normalized<span class="op">=</span><span class="cn">TRUE</span>,main<span class="op">=</span><span class="st">"Normalized"</span><span class="op">)</span></code></pre></div>
<pre><code>## normalizedMatrix is assumed to already be log-transformed</code></pre>
<p><img src="yarn_files/figure-html/density-2.png" width="700"></p>
</div>
<div id="helper-functions" class="section level2">
<h2 class="hasAnchor">
<a href="#helper-functions" class="anchor"></a>Helper functions</h2>
<p>Other than <code>checkMisAnnotation</code> and <code>checkTissuesToMerge</code> we provide a few plotting function. We include, <code>plotCMDS</code>, <code>plotDensity</code>, <code>plotHeatmap</code>.</p>
<p><code>plotCMDS</code> - PCoA / Classical Multi-Dimensional Scaling of the most variable genes.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">skin</span><span class="op">)</span>
<span class="va">res</span> <span class="op">=</span> <span class="fu">yarn</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yarn/man/plotCMDS.html">plotCMDS</a></span><span class="op">(</span><span class="va">skin</span>,pch<span class="op">=</span><span class="fl">21</span>,bg<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span><span class="op">(</span><span class="va">skin</span><span class="op">)</span><span class="op">$</span><span class="va">SMTSD</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="yarn_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<p><code>plotDensity</code> - Density plots colored by phenotype of choosing. Allows for inspection of global trend differences.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filtData</span> <span class="op">=</span> <span class="fu">yarn</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yarn/man/filterLowGenes.html">filterLowGenes</a></span><span class="op">(</span><span class="va">skin</span>,<span class="st">"SMTSD"</span><span class="op">)</span>
<span class="fu">yarn</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yarn/man/plotDensity.html">plotDensity</a></span><span class="op">(</span><span class="va">filtData</span>,groups<span class="op">=</span><span class="st">"SMTSD"</span>,legendPos<span class="op">=</span><span class="st">"topleft"</span><span class="op">)</span></code></pre></div>
<p><img src="yarn_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p><code>plotHeatmap</code> - Heatmap of the most variable genes.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span>
<span class="va">tissues</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span><span class="op">(</span><span class="va">skin</span><span class="op">)</span><span class="op">$</span><span class="va">SMTSD</span>
<span class="va">heatmapColColors</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">12</span>,<span class="st">"Set3"</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">tissues</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="va">heatmapCols</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">9</span>, <span class="st">"RdBu"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">50</span><span class="op">)</span>
<span class="fu">yarn</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yarn/man/plotHeatmap.html">plotHeatmap</a></span><span class="op">(</span><span class="va">skin</span>,normalized<span class="op">=</span><span class="cn">FALSE</span>,log<span class="op">=</span><span class="cn">TRUE</span>,trace<span class="op">=</span><span class="st">"none"</span>,n<span class="op">=</span><span class="fl">10</span>,
 col <span class="op">=</span> <span class="va">heatmapCols</span>,ColSideColors <span class="op">=</span> <span class="va">heatmapColColors</span>,cexRow <span class="op">=</span> <span class="fl">0.25</span>,cexCol <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></code></pre></div>
<p><img src="yarn_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
<div id="information" class="section level2">
<h2 class="hasAnchor">
<a href="#information" class="anchor"></a>Information</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.1 (2021-08-10)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Mojave 10.14.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
## [1] RColorBrewer_1.1-2  netZooR_1.0.4       yarn_1.18.0        
## [4] pandaR_1.19.5       Biobase_2.52.0      BiocGenerics_0.38.0
## [7] reticulate_1.22     igraph_1.3.2       
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.2.2                  proto_1.0.0                
##   [3] R.utils_2.11.0              RUnit_0.4.32               
##   [5] tidyselect_1.1.1            RSQLite_2.2.8              
##   [7] AnnotationDbi_1.54.1        grid_4.1.1                 
##   [9] BiocParallel_1.26.2         base64url_1.4              
##  [11] munsell_0.5.0               codetools_0.2-18           
##  [13] ragg_1.1.3                  preprocessCore_1.54.0      
##  [15] chron_2.3-56                pbdZMQ_0.3-5               
##  [17] colorspace_2.0-2            Category_2.58.0            
##  [19] filelock_1.0.2              highr_0.9                  
##  [21] dplR_1.7.2                  knitr_1.36                 
##  [23] uuid_0.1-4                  stats4_4.1.1               
##  [25] MatrixGenerics_1.4.3        repr_1.1.3                 
##  [27] GenomeInfoDbData_1.2.6      bit64_4.0.5                
##  [29] rhdf5_2.36.0                rprojroot_2.0.2            
##  [31] downloader_0.4              vctrs_0.3.8                
##  [33] generics_0.1.0              xfun_0.26                  
##  [35] BiocFileCache_2.0.0         R6_2.5.1                   
##  [37] doParallel_1.0.16           GenomeInfoDb_1.28.4        
##  [39] illuminaio_0.34.0           RJSONIO_1.3-1.6            
##  [41] locfit_1.5-9.4              bitops_1.0-7               
##  [43] rhdf5filters_1.4.0          cachem_1.0.6               
##  [45] reshape_0.8.8               DelayedArray_0.18.0        
##  [47] assertthat_0.2.1            BiocIO_1.2.0               
##  [49] scales_1.1.1                nnet_7.3-16                
##  [51] gtable_0.3.0                rlang_0.4.11               
##  [53] genefilter_1.74.0           systemfonts_1.0.2          
##  [55] splines_4.1.1               rtracklayer_1.52.1         
##  [57] GEOquery_2.60.0             hexbin_1.28.2              
##  [59] reshape2_1.4.4              yaml_2.2.1                 
##  [61] backports_1.2.1             GenomicFeatures_1.44.2     
##  [63] RBGL_1.68.0                 tools_4.1.1                
##  [65] nor1mix_1.3-0               ggplot2_3.3.6              
##  [67] ellipsis_0.3.2              gplots_3.1.1               
##  [69] jquerylib_0.1.4             ggdendro_0.1.22            
##  [71] siggenes_1.66.0             STRINGdb_2.4.1             
##  [73] gsubfn_0.7                  hash_2.2.6.1               
##  [75] Rcpp_1.0.7                  plyr_1.8.6                 
##  [77] base64enc_0.1-3             sparseMatrixStats_1.4.2    
##  [79] progress_1.2.2              zlibbioc_1.38.0            
##  [81] purrr_0.3.4                 RCurl_1.98-1.5             
##  [83] prettyunits_1.1.1           sqldf_0.4-11               
##  [85] openssl_1.4.5               bumphunter_1.34.0          
##  [87] S4Vectors_0.30.2            cluster_2.1.2              
##  [89] SummarizedExperiment_1.22.0 fs_1.5.0                   
##  [91] magrittr_2.0.1              data.table_1.14.2          
##  [93] matrixStats_0.61.0          hms_1.1.1                  
##  [95] evaluate_0.14               xtable_1.8-4               
##  [97] XML_3.99-0.8                mclust_5.4.7               
##  [99] IRanges_2.26.0              compiler_4.1.1             
## [101] biomaRt_2.48.3              minfi_1.38.0               
## [103] tibble_3.1.5                RCy3_2.12.4                
## [105] KernSmooth_2.23-20          crayon_1.4.1               
## [107] R.oo_1.24.0                 htmltools_0.5.2            
## [109] GOstats_2.58.0              mgcv_1.8-36                
## [111] tzdb_0.1.2                  tidyr_1.1.4                
## [113] DBI_1.1.1                   dbplyr_2.1.1               
## [115] MASS_7.3-54                 rappdirs_0.3.3             
## [117] Matrix_1.3-4                readr_2.0.2                
## [119] permute_0.9-5               quantro_1.26.0             
## [121] uchardet_1.1.0              R.methodsS3_1.8.1          
## [123] quadprog_1.5-8              GenomicRanges_1.44.0       
## [125] pkgconfig_2.0.3             pkgdown_1.6.1              
## [127] GenomicAlignments_1.28.0    signal_0.7-7               
## [129] IRdisplay_1.0               xml2_1.3.2                 
## [131] foreach_1.5.1               annotate_1.70.0            
## [133] penalized_0.9-51            bslib_0.3.1                
## [135] rngtools_1.5                multtest_2.48.0            
## [137] beanplot_1.2                XVector_0.32.0             
## [139] AnnotationForge_1.34.0      doRNG_1.8.2                
## [141] scrime_1.3.5                stringr_1.4.0              
## [143] digest_0.6.28               vegan_2.5-7                
## [145] graph_1.70.0                Biostrings_2.60.2          
## [147] rmarkdown_2.11              base64_2.0                 
## [149] edgeR_3.34.1                DelayedMatrixStats_1.14.3  
## [151] GSEABase_1.54.0             restfulr_0.0.13            
## [153] curl_4.3.2                  Rsamtools_2.8.0            
## [155] gtools_3.9.2                rjson_0.2.20               
## [157] lifecycle_1.0.1             nlme_3.1-152               
## [159] jsonlite_1.7.2              Rhdf5lib_1.14.2            
## [161] viridisLite_0.4.0           desc_1.4.0                 
## [163] askpass_1.1                 limma_3.48.3               
## [165] fansi_0.5.0                 pillar_1.6.3               
## [167] lattice_0.20-44             plotrix_3.8-2              
## [169] KEGGREST_1.32.0             fastmap_1.1.0              
## [171] httr_1.4.2                  survival_3.2-11            
## [173] GO.db_3.13.0                glue_1.4.2                 
## [175] png_0.1-7                   iterators_1.0.13           
## [177] bit_4.0.4                   Rgraphviz_2.36.0           
## [179] stringi_1.7.5               sass_0.4.0                 
## [181] HDF5Array_1.20.0            blob_1.2.2                 
## [183] textshaping_0.3.5           org.Hs.eg.db_3.13.0        
## [185] caTools_1.18.2              memoise_2.0.0              
## [187] IRkernel_1.2                dplyr_1.0.7</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Deborah Weighill.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
