<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='This function is based on the bipartite modularity
as defined in "Modularity and community detection in bipartite networks"
by Michael J. Barber, Phys. Rev. E 76, 066102 (2007)
This function uses a slightly different implementation from the paper. It 
does not use the "adaptive BRIM" method for identifying the number of 
modules. Rather, it simply continues to iterate until the difference in 
modularity between iterations is less that 10^-4. Starting from a random 
initial condition, this could take some time. Use 
condorCluster for quicker runtimes and likely better 
clustering, it initializes the blue 
node memberships by projecting the blue nodes into a unipartite "blue" 
network and then identify communities in that network using a standard 
unipartite community detection algorithm run on the projected network.
See condorCluster for more details on that.
This function loads the entire adjacency matrix in memory, so if your
network has more than ~50,000 nodes, you may want to use
condorModularityMax, which is slower, but does not store
the matrices in memory. Or, of course, you could move to a larger machine.'><title>Iteratively maximize bipartite modularity. — condorMatrixModularity • netZooR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Iteratively maximize bipartite modularity. — condorMatrixModularity"><meta property="og:description" content='This function is based on the bipartite modularity
as defined in "Modularity and community detection in bipartite networks"
by Michael J. Barber, Phys. Rev. E 76, 066102 (2007)
This function uses a slightly different implementation from the paper. It 
does not use the "adaptive BRIM" method for identifying the number of 
modules. Rather, it simply continues to iterate until the difference in 
modularity between iterations is less that 10^-4. Starting from a random 
initial condition, this could take some time. Use 
condorCluster for quicker runtimes and likely better 
clustering, it initializes the blue 
node memberships by projecting the blue nodes into a unipartite "blue" 
network and then identify communities in that network using a standard 
unipartite community detection algorithm run on the projected network.
See condorCluster for more details on that.
This function loads the entire adjacency matrix in memory, so if your
network has more than ~50,000 nodes, you may want to use
condorModularityMax, which is slower, but does not store
the matrices in memory. Or, of course, you could move to a larger machine.'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">netZooR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.17</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    <a class="dropdown-item" href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    <a class="dropdown-item" href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    <a class="dropdown-item" href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    <a class="dropdown-item" href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    <a class="dropdown-item" href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    <a class="dropdown-item" href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    <a class="dropdown-item" href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    <a class="dropdown-item" href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    <a class="dropdown-item" href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/netZoo/netZooR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Iteratively maximize bipartite modularity.</h1>
      <small class="dont-index">Source: <a href="https://github.com/netZoo/netZooR/blob/HEAD/R/CONDOR.R" class="external-link"><code>R/CONDOR.R</code></a></small>
      <div class="d-none name"><code>condorMatrixModularity.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function is based on the bipartite modularity
as defined in "Modularity and community detection in bipartite networks"
by Michael J. Barber, Phys. Rev. E 76, 066102 (2007)
This function uses a slightly different implementation from the paper. It 
does not use the "adaptive BRIM" method for identifying the number of 
modules. Rather, it simply continues to iterate until the difference in 
modularity between iterations is less that 10^-4. Starting from a random 
initial condition, this could take some time. Use 
<code><a href="condorCluster.html">condorCluster</a></code> for quicker runtimes and likely better 
clustering, it initializes the blue 
node memberships by projecting the blue nodes into a unipartite "blue" 
network and then identify communities in that network using a standard 
unipartite community detection algorithm run on the projected network.
See <code><a href="condorCluster.html">condorCluster</a></code> for more details on that.
This function loads the entire adjacency matrix in memory, so if your
network has more than ~50,000 nodes, you may want to use
<code><a href="condorModularityMax.html">condorModularityMax</a></code>, which is slower, but does not store
the matrices in memory. Or, of course, you could move to a larger machine.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">condorMatrixModularity</span><span class="op">(</span></span>
<span>  <span class="va">condor.object</span>,</span>
<span>  T0 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">q</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">q</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  weights <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  deltaQmin <span class="op">=</span> <span class="st">"default"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>condor.object</dt>
<dd><p>is a list created by 
<code><a href="createCondorObject.html">createCondorObject</a></code>. <code>condor.object$edges</code> must 
contain the edges in the giant connected component of a bipartite network</p></dd>


<dt>T0</dt>
<dd><p>is a two column data.frame with the initial community 
assignment for each "blue" node, assuming there are more reds than blues, 
though this is not strictly necessary. The first column contains the 
node name, the second column the community assignment.</p></dd>


<dt>weights</dt>
<dd><p>edgeweights for each edge in <code>edgelist</code>.</p></dd>


<dt>deltaQmin</dt>
<dd><p>convergence parameter determining the minimum required increase
in the modularity for each iteration. Default is min(10^-4,1/(number of edges)),
with number of edges determined by <code>nrow(condor.object$edges)</code>. User can
set this parameter by passing a numeric value to deltaQmin.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Qcoms data.frame with modularity of each community.</p>


<p>modularity modularity value after each iteration.</p>


<p>red.memb community membership of the red nodes</p>


<p>blue.memb community membership of the blue.nodes</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">r</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">3</span>,<span class="fl">3</span>,<span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span>;</span>
<span><span class="va">b</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">4</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">4</span>,<span class="fl">3</span>,<span class="fl">4</span><span class="op">)</span>;</span>
<span><span class="va">reds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Alice"</span>,<span class="st">"Sue"</span>,<span class="st">"Janine"</span>,<span class="st">"Mary"</span><span class="op">)</span></span>
<span><span class="va">blues</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Bob"</span>,<span class="st">"John"</span>,<span class="st">"Ed"</span>,<span class="st">"Hank"</span><span class="op">)</span></span>
<span><span class="va">elist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>red<span class="op">=</span><span class="va">reds</span><span class="op">[</span><span class="va">r</span><span class="op">]</span>,blue<span class="op">=</span><span class="va">blues</span><span class="op">[</span><span class="va">b</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">condor.object</span> <span class="op">&lt;-</span> <span class="fu"><a href="createCondorObject.html">createCondorObject</a></span><span class="op">(</span><span class="va">elist</span><span class="op">)</span></span>
<span><span class="co">#randomly assign blues to their own community</span></span>
<span><span class="va">T0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>nodes<span class="op">=</span><span class="va">blues</span>,coms<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">condor.object</span> <span class="op">&lt;-</span> <span class="fu">condorMatrixModularity</span><span class="op">(</span><span class="va">condor.object</span>,T0<span class="op">=</span><span class="va">T0</span><span class="op">)</span></span></code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill, Kate Shutta.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

