<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference â€¢ netZooR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">netZooR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.17</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    <a class="dropdown-item" href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    <a class="dropdown-item" href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    <a class="dropdown-item" href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    <a class="dropdown-item" href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    <a class="dropdown-item" href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    <a class="dropdown-item" href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    <a class="dropdown-item" href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    <a class="dropdown-item" href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    <a class="dropdown-item" href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/netZoo/netZooR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="main-functions">Main functions<a class="anchor" aria-label="anchor" href="#main-functions"></a></h2>
      
      <p class="section-desc"></p><p>main functions to run each algorithm</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="pandaPy.html">pandaPy()</a></code> 
        </dt>
        <dd>Run Python implementation PANDA in R</dd>
      </dl><dl><dt>
          
          <code><a href="createCondorObject.html">createCondorObject()</a></code> 
        </dt>
        <dd>Create list amenable to analysis using <code>condor</code> package.</dd>
      </dl><dl><dt>
          
          <code><a href="lioness.html">lioness()</a></code> 
        </dt>
        <dd>Compute LIONESS (Linear Interpolation to Obtain Network Estimates for Single Samples)</dd>
      </dl><dl><dt>
          
          <code><a href="alpaca.html">alpaca()</a></code> 
        </dt>
        <dd>Main ALPACA function</dd>
      </dl><dl><dt>
          
          <code><a href="sambar.html">sambar()</a></code> 
        </dt>
        <dd>Main SAMBAR function.</dd>
      </dl><dl><dt>
          
          <code><a href="monster.html">monster()</a></code> 
        </dt>
        <dd>MOdeling Network State Transitions from Expression and Regulatory data (MONSTER)</dd>
      </dl><dl><dt>
          
          <code><a href="otter.html">otter()</a></code> 
        </dt>
        <dd>Run OTTER in R</dd>
      </dl><dl><dt>
          
          <code><a href="cobra.html">cobra()</a></code> 
        </dt>
        <dd>Run COBRA in R</dd>
      </dl><dl><dt>
          
          <code><a href="puma.html">puma()</a></code> 
        </dt>
        <dd>PANDA using microRNA associations</dd>
      </dl><dl><dt>
          
          <code><a href="spider.html">spider()</a></code> 
        </dt>
        <dd>Seeding PANDA Interactions to Derive Epigenetic Regulation</dd>
      </dl><dl><dt>
          
          <code><a href="tiger.html">tiger()</a></code> 
        </dt>
        <dd>TIGER main function</dd>
      </dl><dl><dt>
          
          <code><a href="runEgret.html">runEgret()</a></code> 
        </dt>
        <dd>Run EGRET in R</dd>
      </dl><dl><dt>
          
          <code><a href="dragon.html">dragon()</a></code> 
        </dt>
        <dd>Run DRAGON in R.</dd>
      </dl><dl><dt>
          
          <code><a href="craneBipartite.html">craneBipartite()</a></code> 
        </dt>
        <dd>Pertrubs the bipartite network with fixed node strength</dd>
      </dl></div><div class="section level2">
      <h2 id="panda">PANDA<a class="anchor" aria-label="anchor" href="#panda"></a></h2>
      
      <p class="section-desc"></p><p>All functions of PANDA</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="pandaPy.html">pandaPy()</a></code> 
        </dt>
        <dd>Run Python implementation PANDA in R</dd>
      </dl><dl><dt>
          
          <code><a href="pandaToCondorObject.html">pandaToCondorObject()</a></code> 
        </dt>
        <dd>Turn PANDA network into a CONDOR object</dd>
      </dl><dl><dt>
          
          <code><a href="pandaToAlpaca.html">pandaToAlpaca()</a></code> 
        </dt>
        <dd>Use two PANDA network to generate an ALPACA result</dd>
      </dl><dl><dt>
          
          <code><a href="pandaDiffEdges.html">pandaDiffEdges()</a></code> 
        </dt>
        <dd>Identify differential edges in two PANDA networks</dd>
      </dl><dl><dt>
          
          <code><a href="visPandaInCytoscape.html">visPandaInCytoscape()</a></code> 
        </dt>
        <dd>Plot PANDA network in Cytoscape</dd>
      </dl><dl><dt>
          
          <code><a href="createPandaStyle.html">createPandaStyle()</a></code> 
        </dt>
        <dd>Create a Cytoscape visual style for PANDA network</dd>
      </dl></div><div class="section level2">
      <h2 id="condor">CONDOR<a class="anchor" aria-label="anchor" href="#condor"></a></h2>
      
      <p class="section-desc"></p><p>All functions of CONDOR</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="createCondorObject.html">createCondorObject()</a></code> 
        </dt>
        <dd>Create list amenable to analysis using <code>condor</code> package.</dd>
      </dl><dl><dt>
          
          <code><a href="condorCluster.html">condorCluster()</a></code> 
        </dt>
        <dd>Main clustering function for condor.</dd>
      </dl><dl><dt>
          
          <code><a href="condorCoreEnrich.html">condorCoreEnrich()</a></code> 
        </dt>
        <dd>Compare qscore distribution of a subset of nodes to all other nodes.</dd>
      </dl><dl><dt>
          
          <code><a href="condorCreateObject.html">condorCreateObject()</a></code> 
        </dt>
        <dd>creates condor object</dd>
      </dl><dl><dt>
          
          <code><a href="condorMatrixModularity.html">condorMatrixModularity()</a></code> 
        </dt>
        <dd>Iteratively maximize bipartite modularity.</dd>
      </dl><dl><dt>
          
          <code><a href="condorModularityMax.html">condorModularityMax()</a></code> 
        </dt>
        <dd>Iteratively maximize bipartite modularity.</dd>
      </dl><dl><dt>
          
          <code><a href="condorPlotCommunities.html">condorPlotCommunities()</a></code> 
        </dt>
        <dd>Plot adjacency matrix with links grouped and colored by community</dd>
      </dl><dl><dt>
          
          <code><a href="condorPlotHeatmap.html">condorPlotHeatmap()</a></code> 
        </dt>
        <dd>Plot weighted adjacency matrix with links grouped by community</dd>
      </dl><dl><dt>
          
          <code><a href="condorQscore.html">condorQscore()</a></code> 
        </dt>
        <dd>Calculate Qscore for all nodes</dd>
      </dl><dl><dt>
          
          <code><a href="condorRun.html">condorRun()</a></code> 
        </dt>
        <dd>Run CONDOR clustering</dd>
      </dl></div><div class="section level2">
      <h2 id="lioness">LIONESS<a class="anchor" aria-label="anchor" href="#lioness"></a></h2>
      
      <p class="section-desc"></p><p>All functions of LIONESS</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="lionessPy.html">lionessPy()</a></code> 
        </dt>
        <dd>Run python implementation of LIONESS</dd>
      </dl><dl><dt>
          
          <code><a href="lioness.html">lioness()</a></code> 
        </dt>
        <dd>Compute LIONESS (Linear Interpolation to Obtain Network Estimates for Single Samples)</dd>
      </dl></div><div class="section level2">
      <h2 id="alpaca">ALPACA<a class="anchor" aria-label="anchor" href="#alpaca"></a></h2>
      
      <p class="section-desc"></p><p>All functions of ALPACA</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="alpaca.html">alpaca()</a></code> 
        </dt>
        <dd>Main ALPACA function</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaCommunityStructureRotation.html">alpacaCommunityStructureRotation()</a></code> 
        </dt>
        <dd>Comparing node community membership between two networks</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaComputeDWBMmatmScale.html">alpacaComputeDWBMmatmScale()</a></code> 
        </dt>
        <dd>Differential modularity matrix</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaComputeDifferentialScoreFromDWBM.html">alpacaComputeDifferentialScoreFromDWBM()</a></code> 
        </dt>
        <dd>Compute Differential modularity score from differential modularity matrix</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaComputeWBMmat.html">alpacaComputeWBMmat()</a></code> 
        </dt>
        <dd>Compute modularity matrix for weighted bipartite network</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaCrane.html">alpacaCrane()</a></code> 
        </dt>
        <dd>Find the robust nodes in ALPACA community using CRANE</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaDeltaZAnalysis.html">alpacaDeltaZAnalysis()</a></code> 
        </dt>
        <dd>Edge subtraction method (CONDOR optimizaton)</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaDeltaZAnalysisLouvain.html">alpacaDeltaZAnalysisLouvain()</a></code> 
        </dt>
        <dd>Edge subtraction method (Louvain optimizaton)</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaExtractTopGenes.html">alpacaExtractTopGenes()</a></code> 
        </dt>
        <dd>Extract core target genes in differential modules</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaGOtabtogenes.html">alpacaGOtabtogenes()</a></code> 
        </dt>
        <dd>The top GO term associated genes in each module</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaGenLouvain.html">alpacaGenLouvain()</a></code> 
        </dt>
        <dd>Generalized Louvain optimization</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaGetMember.html">alpacaGetMember()</a></code> 
        </dt>
        <dd>get the member vector from alpaca object</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaGoToGenes.html">alpacaGoToGenes()</a></code> 
        </dt>
        <dd>Map GO terms to gene symbols</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaListToGo.html">alpacaListToGo()</a></code> 
        </dt>
        <dd>GO term enrichment for a list of gene sets</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaMetaNetwork.html">alpacaMetaNetwork()</a></code> 
        </dt>
        <dd>Create alpacaMetaNetwork for Louvain optimization</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaNodeToGene.html">alpacaNodeToGene()</a></code> 
        </dt>
        <dd>Remove tags from gene names</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaObjectToDfList.html">alpacaObjectToDfList()</a></code> 
        </dt>
        <dd>Converts alpaca output into list of data frames</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaRotationAnalysis.html">alpacaRotationAnalysis()</a></code> 
        </dt>
        <dd>Community comparison method (CONDOR optimizaton)</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaRotationAnalysisLouvain.html">alpacaRotationAnalysisLouvain()</a></code> 
        </dt>
        <dd>Community comparison method (CONDOR optimizaton)</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaSimulateNetwork.html">alpacaSimulateNetwork()</a></code> 
        </dt>
        <dd>Simulated networks</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaTestNodeRank.html">alpacaTestNodeRank()</a></code> 
        </dt>
        <dd>Enrichment in ranked list</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaTidyConfig.html">alpacaTidyConfig()</a></code> 
        </dt>
        <dd>Renumbering community membership vector</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaTopEnsembltoTopSym.html">alpacaTopEnsembltoTopSym()</a></code> 
        </dt>
        <dd>Translating gene identifiers to gene symbols</dd>
      </dl><dl><dt>
          
          <code><a href="alpacaWBMlouvain.html">alpacaWBMlouvain()</a></code> 
        </dt>
        <dd>Generalized Louvain method for bipartite networks</dd>
      </dl></div><div class="section level2">
      <h2 id="sambar">SAMBAR<a class="anchor" aria-label="anchor" href="#sambar"></a></h2>
      
      <p class="section-desc"></p><p>All functions of SAMBAR</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="sambar.html">sambar()</a></code> 
        </dt>
        <dd>Main SAMBAR function.</dd>
      </dl><dl><dt>
          
          <code><a href="sambarConvertgmt.html">sambarConvertgmt()</a></code> 
        </dt>
        <dd>Convert .gmt files into a binary matrix.</dd>
      </dl><dl><dt>
          
          <code><a href="sambarCorgenelength.html">sambarCorgenelength()</a></code> 
        </dt>
        <dd>Normalize gene mutation scores by gene length.</dd>
      </dl><dl><dt>
          
          <code><a href="sambarDesparsify.html">sambarDesparsify()</a></code> 
        </dt>
        <dd>De-sparsify gene-level mutation scores into gene set-level mutation scores.</dd>
      </dl></div><div class="section level2">
      <h2 id="monster">MONSTER<a class="anchor" aria-label="anchor" href="#monster"></a></h2>
      
      <p class="section-desc"></p><p>All functions of MONSTER</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="monster.html">monster()</a></code> 
        </dt>
        <dd>MOdeling Network State Transitions from Expression and Regulatory data (MONSTER)</dd>
      </dl><dl><dt>
          
          <code><a href="monsterBereFull.html">monsterBereFull()</a></code> 
        </dt>
        <dd>Bipartite Edge Reconstruction from Expression data 
(composite method with direct/indirect)</dd>
      </dl><dl><dt>
          
          <code><a href="monsterCalculateTmPValues.html">monsterCalculateTmPValues()</a></code> 
        </dt>
        <dd>Calculate p-values for a tranformation matrix</dd>
      </dl><dl><dt>
          
          <code><a href="monsterCheckDataType.html">monsterCheckDataType()</a></code> 
        </dt>
        <dd>Checks that data is something MONSTER can handle</dd>
      </dl><dl><dt>
          
          <code><a href="monsterGetTm.html">monsterGetTm()</a></code> 
        </dt>
        <dd>monsterGetTm</dd>
      </dl><dl><dt>
          
          <code><a href="monsterHclHeatmapPlot.html">monsterHclHeatmapPlot()</a></code> 
        </dt>
        <dd>Transformation matrix plot</dd>
      </dl><dl><dt>
          
          <code><a href="monsterMonsterNI.html">monsterMonsterNI()</a></code> 
        </dt>
        <dd>Bipartite Edge Reconstruction from Expression data</dd>
      </dl><dl><dt>
          
          <code><a href="monsterPlotMonsterAnalysis.html">monsterPlotMonsterAnalysis()</a></code> 
        </dt>
        <dd>monsterPlotMonsterAnalysis</dd>
      </dl><dl><dt>
          
          <code><a href="monsterPrintMonsterAnalysis.html">monsterPrintMonsterAnalysis()</a></code> 
        </dt>
        <dd>monsterPrintMonsterAnalysis</dd>
      </dl><dl><dt>
          
          <code><a href="monsterRes.html">monsterRes</a></code> 
        </dt>
        <dd>MONSTER results from example cell-cycle yeast transition</dd>
      </dl><dl><dt>
          
          <code><a href="monsterTransformationMatrix.html">monsterTransformationMatrix()</a></code> 
        </dt>
        <dd>Bi-partite network analysis tools</dd>
      </dl><dl><dt>
          
          <code><a href="monsterTransitionNetworkPlot.html">monsterTransitionNetworkPlot()</a></code> 
        </dt>
        <dd>This function uses igraph to plot the transition matrix (directed graph) as a network.
The edges in the network should be read as A 'positively/negatively contributes to' the 
targeting of B in the target state.</dd>
      </dl><dl><dt>
          
          <code><a href="monsterTransitionPCAPlot.html">monsterTransitionPCAPlot()</a></code> 
        </dt>
        <dd>Principal Components plot of transformation matrix</dd>
      </dl><dl><dt>
          
          <code><a href="monsterdTFIPlot.html">monsterdTFIPlot()</a></code> 
        </dt>
        <dd>This function plots the Off diagonal mass of an 
observed Transition Matrix compared to a set of null TMs</dd>
      </dl></div><div class="section level2">
      <h2 id="otter">OTTER<a class="anchor" aria-label="anchor" href="#otter"></a></h2>
      
      <p class="section-desc"></p><p>OTTER functions</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="otter.html">otter()</a></code> 
        </dt>
        <dd>Run OTTER in R</dd>
      </dl></div><div class="section level2">
      <h2 id="tiger">TIGER<a class="anchor" aria-label="anchor" href="#tiger"></a></h2>
      
      <p class="section-desc"></p><p>TIGER functions</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="tiger.html">tiger()</a></code> 
        </dt>
        <dd>TIGER main function</dd>
      </dl><dl><dt>
          
          <code><a href="TIGER_expr.html">TIGER_expr</a></code> 
        </dt>
        <dd>TIGER example expression matrix</dd>
      </dl><dl><dt>
          
          <code><a href="TIGER_prior.html">TIGER_prior</a></code> 
        </dt>
        <dd>TIGER example prior network</dd>
      </dl><dl><dt>
          
          <code><a href="adj2regulon.html">adj2regulon()</a></code> 
        </dt>
        <dd>Convert bipartite adjacency to regulon</dd>
      </dl><dl><dt>
          
          <code><a href="priorPp.html">priorPp()</a></code> 
        </dt>
        <dd>Filter low confident edge signs in the prior network using GeneNet</dd>
      </dl></div><div class="section level2">
      <h2 id="cobra">COBRA<a class="anchor" aria-label="anchor" href="#cobra"></a></h2>
      
      <p class="section-desc"></p><p>COBRA functions</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="cobra.html">cobra()</a></code> 
        </dt>
        <dd>Run COBRA in R</dd>
      </dl></div><div class="section level2">
      <h2 id="crane">CRANE<a class="anchor" aria-label="anchor" href="#crane"></a></h2>
      
      <p class="section-desc"></p><p>CRANE functions</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="craneBipartite.html">craneBipartite()</a></code> 
        </dt>
        <dd>Pertrubs the bipartite network with fixed node strength</dd>
      </dl><dl><dt>
          
          <code><a href="craneUnipartite.html">craneUnipartite()</a></code> 
        </dt>
        <dd>Pertrubs the unipartite network with fixed node strength from adjacency matrix</dd>
      </dl><dl><dt>
          
          <code><a href="elistToAdjMat.html">elistToAdjMat()</a></code> 
        </dt>
        <dd>Converts edge list to adjacency matrix</dd>
      </dl><dl><dt>
          
          <code><a href="elistSort.html">elistSort()</a></code> 
        </dt>
        <dd>Sorts the edge list based on first two columns in alphabetical order</dd>
      </dl><dl><dt>
          
          <code><a href="elistRemoveTags.html">elistRemoveTags()</a></code> 
        </dt>
        <dd>undo elistAddTags</dd>
      </dl><dl><dt>
          
          <code><a href="isElist.html">isElist()</a></code> 
        </dt>
        <dd>Check if data frame is an edge list</dd>
      </dl><dl><dt>
          
          <code><a href="adjMatToElist.html">adjMatToElist()</a></code> 
        </dt>
        <dd>converts adjacency matrix to edge list</dd>
      </dl><dl><dt>
          
          <code><a href="el2adj.html">el2adj()</a></code> 
        </dt>
        <dd>Convert bipartite edge list to adjacency mat</dd>
      </dl><dl><dt>
          
          <code><a href="el2regulon.html">el2regulon()</a></code> 
        </dt>
        <dd>Convert a bipartite edgelist to regulon</dd>
      </dl><dl><dt>
          
          <code><a href="elistIsEdgeOrderEqual.html">elistIsEdgeOrderEqual()</a></code> 
        </dt>
        <dd>check if first two columns are identical</dd>
      </dl><dl><dt>
          
          <code><a href="elistAddTags.html">elistAddTags()</a></code> 
        </dt>
        <dd>Adds "_A" to first column and "_B" to second column</dd>
      </dl><dl><dt>
          
          <code><a href="adj2el.html">adj2el()</a></code> 
        </dt>
        <dd>Convert a bipartite adjacency matrix to an edgelist</dd>
      </dl><dl><dt>
          
          <code><a href="jutterDegree.html">jutterDegree()</a></code> 
        </dt>
        <dd>CRANE Beta perturbation function. This function will add noice to the node strength sequence.</dd>
      </dl></div><div class="section level2">
      <h2 id="dragon">DRAGON<a class="anchor" aria-label="anchor" href="#dragon"></a></h2>
      
      <p class="section-desc"></p><p>DRAGON functions</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="dragon.html">dragon()</a></code> 
        </dt>
        <dd>Run DRAGON in R.</dd>
      </dl></div><div class="section level2">
      <h2 id="spider">SPIDER<a class="anchor" aria-label="anchor" href="#spider"></a></h2>
      
      <p class="section-desc"></p><p>SPIDER functions</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="spider.html">spider()</a></code> 
        </dt>
        <dd>Seeding PANDA Interactions to Derive Epigenetic Regulation</dd>
      </dl><dl><dt>
          
          <code><a href="degreeAdjust.html">degreeAdjust()</a></code> 
        </dt>
        <dd>Function to adjust the degree so that the hub nodes are not penalized in z-score transformation</dd>
      </dl></div><div class="section level2">
      <h2 id="puma">PUMA<a class="anchor" aria-label="anchor" href="#puma"></a></h2>
      
      <p class="section-desc"></p><p>PUMA functions</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="puma.html">puma()</a></code> 
        </dt>
        <dd>PANDA using microRNA associations</dd>
      </dl></div><div class="section level2">
      <h2 id="egret">EGRET<a class="anchor" aria-label="anchor" href="#egret"></a></h2>
      
      <p class="section-desc"></p><p>EGRET functions</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="runEgret.html">runEgret()</a></code> 
        </dt>
        <dd>Run EGRET in R</dd>
      </dl></div><div class="section level2">
      <h2 id="blobfish">BLOBFISH<a class="anchor" aria-label="anchor" href="#blobfish"></a></h2>
      
      <p class="section-desc"></p><p>BLOBFISH functions</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="RunBLOBFISH.html">RunBLOBFISH()</a></code> 
        </dt>
        <dd>Given a set of genes of interest, full bipartite networks with scores (one network for each sample), a significance
cutoff for statistical testing, and a hop constraint, BLOBFISH finds a subnetwork of
significant edges connecting the genes.</dd>
      </dl><dl><dt>
          
          <code><a href="GenerateNullPANDADistribution.html">GenerateNullPANDADistribution()</a></code> 
        </dt>
        <dd>Generate a null distribution of edge scores for PANDA-like networks; that is,
the set of edges where (1) the TF does not have a binding motif in the gene region,
(2) the TF does not form a complex with any other TF that has a binding motif in
the gene region, and (3) the genes regulated by the TF are not coexpressed with the
gene in question. We obtain this by inputting an empty prior and an identity coexpression
matrix.</dd>
      </dl><dl><dt>
          
          <code><a href="BuildSubnetwork.html">BuildSubnetwork()</a></code> 
        </dt>
        <dd>Find the subnetwork of significant edges connecting the genes.</dd>
      </dl><dl><dt>
          
          <code><a href="CalculatePValues.html">CalculatePValues()</a></code> 
        </dt>
        <dd>Calculate p-values for all edges in the network using a Wilcoxon two-sample test
for each edge.</dd>
      </dl><dl><dt>
          
          <code><a href="FindSignificantEdgesForHop.html">FindSignificantEdgesForHop()</a></code> 
        </dt>
        <dd>Find the subnetwork of significant edges n / 2 hops away from each gene.</dd>
      </dl><dl><dt>
          
          <code><a href="SignificantBreadthFirstSearch.html">SignificantBreadthFirstSearch()</a></code> 
        </dt>
        <dd>Find all significant edges adjacent to the starting nodes, excluding the nodes
specified.</dd>
      </dl><dl><dt>
          
          <code><a href="FindConnectionsForAllHopCounts.html">FindConnectionsForAllHopCounts()</a></code> 
        </dt>
        <dd>For all hop counts up to the maximum, find subnetworks connecting each pair of
genes by exactly that number of hops. For instance, find each</dd>
      </dl><dl><dt>
          
          <code><a href="PlotNetwork.html">PlotNetwork()</a></code> 
        </dt>
        <dd>Plot the networks, using different colors for transcription factors, genes of interest,
and additional genes.</dd>
      </dl></div><div class="section level2">
      <h2 id="yarn">YARN<a class="anchor" aria-label="anchor" href="#yarn"></a></h2>
      
      <p class="section-desc"></p><p>YARN functions</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="checkMisAnnotation.html">checkMisAnnotation()</a></code> 
        </dt>
        <dd>Check for wrong annotation of a sample using classical MDS and control genes.</dd>
      </dl><dl><dt>
          
          <code><a href="checkTissuesToMerge.html">checkTissuesToMerge()</a></code> 
        </dt>
        <dd>Check tissues to merge based on gene expression profile</dd>
      </dl><dl><dt>
          
          <code><a href="filterGenes.html">filterGenes()</a></code> 
        </dt>
        <dd>Filter specific genes</dd>
      </dl><dl><dt>
          
          <code><a href="filterLowGenes.html">filterLowGenes()</a></code> 
        </dt>
        <dd>Filter genes that have less than a minimum threshold CPM for a given group/tissue</dd>
      </dl><dl><dt>
          
          <code><a href="normalizeTissueAware.html">normalizeTissueAware()</a></code> 
        </dt>
        <dd>Normalize in a tissue aware context</dd>
      </dl><dl><dt>
          
          <code><a href="annotateFromBiomart.html">annotateFromBiomart()</a></code> 
        </dt>
        <dd>Annotate your Expression Set with biomaRt</dd>
      </dl><dl><dt>
          
          <code><a href="downloadGTEx.html">downloadGTEx()</a></code> 
        </dt>
        <dd>Download GTEx files and turn them into ExpressionSet object</dd>
      </dl><dl><dt>
          
          <code><a href="extractMatrix.html">extractMatrix()</a></code> 
        </dt>
        <dd>Extract the appropriate matrix</dd>
      </dl><dl><dt>
          
          <code><a href="filterMissingGenes.html">filterMissingGenes()</a></code> 
        </dt>
        <dd>Filter genes not expressed in any sample</dd>
      </dl><dl><dt>
          
          <code><a href="filterSamples.html">filterSamples()</a></code> 
        </dt>
        <dd>Filter samples</dd>
      </dl><dl><dt>
          
          <code><a href="plotCMDS.html">plotCMDS()</a></code> 
        </dt>
        <dd>Plot classical MDS of dataset</dd>
      </dl><dl><dt>
          
          <code><a href="plotDensity.html">plotDensity()</a></code> 
        </dt>
        <dd>Density plots of columns in a matrix</dd>
      </dl><dl><dt>
          
          <code><a href="plotHeatmap.html">plotHeatmap()</a></code> 
        </dt>
        <dd>Plot heatmap of most variable genes</dd>
      </dl><dl><dt>
          
          <code><a href="qsmooth.html">qsmooth()</a></code> 
        </dt>
        <dd>Quantile shrinkage normalization</dd>
      </dl><dl><dt>
          
          <code><a href="qstats.html">qstats()</a></code> 
        </dt>
        <dd>Compute quantile statistics</dd>
      </dl></div><div class="section level2">
      <h2 id="plotting-functions">Plotting functions<a class="anchor" aria-label="anchor" href="#plotting-functions"></a></h2>
      
      <p class="section-desc"></p><p>function to plot the networks and community structures.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="createPandaStyle.html">createPandaStyle()</a></code> 
        </dt>
        <dd>Create a Cytoscape visual style for PANDA network</dd>
      </dl><dl><dt>
          
          <code><a href="condorPlotCommunities.html">condorPlotCommunities()</a></code> 
        </dt>
        <dd>Plot adjacency matrix with links grouped and colored by community</dd>
      </dl><dl><dt>
          
          <code><a href="condorPlotHeatmap.html">condorPlotHeatmap()</a></code> 
        </dt>
        <dd>Plot weighted adjacency matrix with links grouped by community</dd>
      </dl></div><div class="section level2">
      <h2 id="helper-functions">Helper functions<a class="anchor" aria-label="anchor" href="#helper-functions"></a></h2>
      
      <p class="section-desc"></p><p>miscellaneous function</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="sourcePPI.html">sourcePPI()</a></code> 
        </dt>
        <dd>Source the Protein-Protein interaction in STRING database</dd>
      </dl><dl><dt>
          
          <code><a href="pandaToCondorObject.html">pandaToCondorObject()</a></code> 
        </dt>
        <dd>Turn PANDA network into a CONDOR object</dd>
      </dl><dl><dt>
          
          <code><a href="pandaToAlpaca.html">pandaToAlpaca()</a></code> 
        </dt>
        <dd>Use two PANDA network to generate an ALPACA result</dd>
      </dl></div><div class="section level2">
      <h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a></h2>
      
      <p class="section-desc"></p><p>Input example data files</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="yeast.html">yeast</a></code> 
        </dt>
        <dd>Toy data derived from three gene expression datasets and a mapping from transcription factors to genes.</dd>
      </dl><dl><dt>
          
          <code><a href="small1976.html">small1976</a></code> 
        </dt>
        <dd>Pollinator-plant interactions</dd>
      </dl><dl><dt>
          
          <code><a href="mut.ucec.html">mut.ucec</a></code> <code><a href="mut.ucec.html">mut.ucec</a></code> 
        </dt>
        <dd>Example of mutation data</dd>
      </dl><dl><dt>
          
          <code><a href="exon.size.html">exon.size</a></code> <code><a href="exon.size.html">exon.size</a></code> <code><a href="exon.size.html">genes</a></code> 
        </dt>
        <dd>Gene length</dd>
      </dl><dl><dt>
          
          <code><a href="genes.html">genes</a></code> 
        </dt>
        <dd>Example of a gene list</dd>
      </dl><dl><dt>
          
          <code><a href="Bladder_data.html">bladder</a></code> 
        </dt>
        <dd>Bladder RNA-seq data from the GTEx consortium</dd>
      </dl><dl><dt>
          
          <code><a href="Skin_data.html">skin</a></code> 
        </dt>
        <dd>Skin RNA-seq data from the GTEx consortium</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill, Kate Shutta.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

