<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ netZooR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.58</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/netZoo/netZooR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-main-functions" class="hasAnchor"><a href="#section-main-functions" class="anchor"></a>Main functions</h2>
          <p class="section-desc"><p>main functions to run each algorithm</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="alpaca.html">alpaca()</a></code> </p>
        </td>
        <td><p>Main ALPACA function</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaCommunityStructureRotation.html">alpacaCommunityStructureRotation()</a></code> </p>
        </td>
        <td><p>Comparing node community membership between two networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaDeltaZAnalysis.html">alpacaDeltaZAnalysis()</a></code> </p>
        </td>
        <td><p>Edge subtraction method (CONDOR optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaDeltaZAnalysisLouvain.html">alpacaDeltaZAnalysisLouvain()</a></code> </p>
        </td>
        <td><p>Edge subtraction method (Louvain optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaExtractTopGenes.html">alpacaExtractTopGenes()</a></code> </p>
        </td>
        <td><p>Extract core target genes in differential modules</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaGOtabtogenes.html">alpacaGOtabtogenes()</a></code> </p>
        </td>
        <td><p>The top GO term associated genes in each module</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaRotationAnalysis.html">alpacaRotationAnalysis()</a></code> </p>
        </td>
        <td><p>Community comparison method (CONDOR optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaRotationAnalysisLouvain.html">alpacaRotationAnalysisLouvain()</a></code> </p>
        </td>
        <td><p>Community comparison method (CONDOR optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaTestNodeRank.html">alpacaTestNodeRank()</a></code> </p>
        </td>
        <td><p>Enrichment in ranked list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaTopEnsembltoTopSym.html">alpacaTopEnsembltoTopSym()</a></code> </p>
        </td>
        <td><p>Translating gene identifiers to gene symbols</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaWBMlouvain.html">alpacaWBMlouvain()</a></code> </p>
        </td>
        <td><p>Generalized Louvain method for bipartite networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacacomputeDWBMmatmscale.html">alpacacomputeDWBMmatmscale()</a></code> </p>
        </td>
        <td><p>Differential modularity matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacacomputeWBMmat.html">alpacacomputeWBMmat()</a></code> </p>
        </td>
        <td><p>Compute modularity matrix for weighted bipartite network</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacagenlouvain.html">alpacagenlouvain()</a></code> </p>
        </td>
        <td><p>Generalized Louvain optimization</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacagotogenes.html">alpacagotogenes()</a></code> </p>
        </td>
        <td><p>Map GO terms to gene symbols</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacalisttogo.html">alpacalisttogo()</a></code> </p>
        </td>
        <td><p>GO term enrichment for a list of gene sets</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacametanetwork.html">alpacametanetwork()</a></code> </p>
        </td>
        <td><p>Create alpacametanetwork for Louvain optimization</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacanodetogene.html">alpacanodetogene()</a></code> </p>
        </td>
        <td><p>Remove tags from gene names</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacasimulateNetwork.html">alpacasimulateNetwork()</a></code> </p>
        </td>
        <td><p>Simulated networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacatidyconfig.html">alpacatidyconfig()</a></code> </p>
        </td>
        <td><p>Renumbering community membership vector</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorCluster.html">condorCluster()</a></code> </p>
        </td>
        <td><p>Main clustering function for condor.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorCoreEnrich.html">condorCoreEnrich()</a></code> </p>
        </td>
        <td><p>Compare qscore distribution of a subset of nodes to all other nodes.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorMatrixModularity.html">condorMatrixModularity()</a></code> </p>
        </td>
        <td><p>Iteratively maximize bipartite modularity.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorModularityMax.html">condorModularityMax()</a></code> </p>
        </td>
        <td><p>Iteratively maximize bipartite modularity.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorPlotCommunities.html">condorPlotCommunities()</a></code> </p>
        </td>
        <td><p>Plot adjacency matrix with links grouped and colored by community</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorPlotHeatmap.html">condorPlotHeatmap()</a></code> </p>
        </td>
        <td><p>Plot weighted adjacency matrix with links grouped by community</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorQscore.html">condorQscore()</a></code> </p>
        </td>
        <td><p>Calculate Qscore for all nodes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="createCondorObject.html">createCondorObject()</a></code> </p>
        </td>
        <td><p>Create list amenable to analysis using <code>condor</code> package.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="createDiffPandaStyle.html">createDiffPandaStyle()</a></code> </p>
        </td>
        <td><p>Create a Cytoscape visual style for two PANDA networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="createPandaStyle.html">createPandaStyle()</a></code> </p>
        </td>
        <td><p>Create a Cytoscape visual style for PANDA network</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="exon.size.html">exon.size</a></code> </p>
        </td>
        <td><p>Gene length</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="genes.html">genes</a></code> </p>
        </td>
        <td><p>Example of a gene list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="lioness.html">lioness()</a></code> </p>
        </td>
        <td><p>Compute LIONESS (Linear Interpolation to Obtain Network Estimates for Single Samples)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="lionessPy.html">lionessPy()</a></code> </p>
        </td>
        <td><p>Run python implementation of LIONESS</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monster.html">monster()</a></code> </p>
        </td>
        <td><p>MOdeling Network State Transitions from Expression and Regulatory data (MONSTER)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterBereFull.html">monsterBereFull()</a></code> </p>
        </td>
        <td><p>Bipartite Edge Reconstruction from Expression data 
(composite method with direct/indirect)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterCalculateTmPvalues.html">monsterCalculateTmPvalues()</a></code> </p>
        </td>
        <td><p>Calculate p-values for a tranformation matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterCheckDataType.html">monsterCheckDataType()</a></code> </p>
        </td>
        <td><p>Checks that data is something MONSTER can handle</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterHclHeatmapPlot.html">monsterHclHeatmapPlot()</a></code> </p>
        </td>
        <td><p>Transformation matrix plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterMonsterNI.html">monsterMonsterNI()</a></code> </p>
        </td>
        <td><p>Bipartite Edge Reconstruction from Expression data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterPlotMonsterAnalysis.html">monsterPlotMonsterAnalysis()</a></code> </p>
        </td>
        <td><p>monsterPlotMonsterAnalysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterPrintMonsterAnalysis.html">monsterPrintMonsterAnalysis()</a></code> </p>
        </td>
        <td><p>monsterPrintMonsterAnalysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterRes.html">monsterRes</a></code> </p>
        </td>
        <td><p>MONSTER results from example cell-cycle yeast transition</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterTransformationMatrix.html">monsterTransformationMatrix()</a></code> </p>
        </td>
        <td><p>Bi-partite network analysis tools</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterTransitionPCAPlot.html">monsterTransitionPCAPlot()</a></code> </p>
        </td>
        <td><p>Principal Components plot of transformation matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterdTFIPlot.html">monsterdTFIPlot()</a></code> </p>
        </td>
        <td><p>This function plots the Off diagonal mass of an 
observed Transition Matrix compared to a set of null TMs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monstertransitionNetworkPlot.html">monstertransitionNetworkPlot()</a></code> </p>
        </td>
        <td><p>This function uses igraph to plot the transition matrix (directed graph) as a network.
The edges in the network should be read as A 'positively/negatively contributes to' the 
targeting of B in the target state.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mut.ucec.html">mut.ucec</a></code> </p>
        </td>
        <td><p>Example of mutation data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="otter.html">otter()</a></code> </p>
        </td>
        <td><p>Run OTTER in R</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pandaDiffEdges.html">pandaDiffEdges()</a></code> </p>
        </td>
        <td><p>Identify differential edges in two PANDA networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pandaPy.html">pandaPy()</a></code> </p>
        </td>
        <td><p>Run Python implementation PANDA in R</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pandaToAlpaca.html">pandaToAlpaca()</a></code> </p>
        </td>
        <td><p>Use two PANDA network to generate an ALPACA result</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pandaToCondorObject.html">pandaToCondorObject()</a></code> </p>
        </td>
        <td><p>Turn PANDA network into a CONDOR object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runEgret.html">runEgret()</a></code> </p>
        </td>
        <td><p>Run EGRET in R</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambar.html">sambar()</a></code> </p>
        </td>
        <td><p>Main SAMBAR function.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambarConvertgmt.html">sambarConvertgmt()</a></code> </p>
        </td>
        <td><p>Convert .gmt files into a binary matrix.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambarCorgenelength.html">sambarCorgenelength()</a></code> </p>
        </td>
        <td><p>Normalize gene mutation scores by gene length.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambarDesparsify.html">sambarDesparsify()</a></code> </p>
        </td>
        <td><p>De-sparsify gene-level mutation scores into gene set-level mutation scores.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="small1976.html">small1976</a></code> </p>
        </td>
        <td><p>Pollinator-plant interactions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sourcePPI.html">sourcePPI()</a></code> </p>
        </td>
        <td><p>Source the Protein-Protein interaction in STRING database</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="visDiffPandaInCytoscape.html">visDiffPandaInCytoscape()</a></code> </p>
        </td>
        <td><p>Plot two PANDA networsk in Cytoscape</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="visPandaInCytoscape.html">visPandaInCytoscape()</a></code> </p>
        </td>
        <td><p>Plot PANDA network in Cytoscape</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="yeast.html">yeast</a></code> </p>
        </td>
        <td><p>Toy data derived from three gene expression datasets and a mapping from transcription factors to genes.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-plot-functions" class="hasAnchor"><a href="#section-plot-functions" class="anchor"></a>Plot functions</h2>
          <p class="section-desc"><p>function to plot the networks and community structures.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="alpaca.html">alpaca()</a></code> </p>
        </td>
        <td><p>Main ALPACA function</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaCommunityStructureRotation.html">alpacaCommunityStructureRotation()</a></code> </p>
        </td>
        <td><p>Comparing node community membership between two networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaDeltaZAnalysis.html">alpacaDeltaZAnalysis()</a></code> </p>
        </td>
        <td><p>Edge subtraction method (CONDOR optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaDeltaZAnalysisLouvain.html">alpacaDeltaZAnalysisLouvain()</a></code> </p>
        </td>
        <td><p>Edge subtraction method (Louvain optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaExtractTopGenes.html">alpacaExtractTopGenes()</a></code> </p>
        </td>
        <td><p>Extract core target genes in differential modules</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaGOtabtogenes.html">alpacaGOtabtogenes()</a></code> </p>
        </td>
        <td><p>The top GO term associated genes in each module</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaRotationAnalysis.html">alpacaRotationAnalysis()</a></code> </p>
        </td>
        <td><p>Community comparison method (CONDOR optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaRotationAnalysisLouvain.html">alpacaRotationAnalysisLouvain()</a></code> </p>
        </td>
        <td><p>Community comparison method (CONDOR optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaTestNodeRank.html">alpacaTestNodeRank()</a></code> </p>
        </td>
        <td><p>Enrichment in ranked list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaTopEnsembltoTopSym.html">alpacaTopEnsembltoTopSym()</a></code> </p>
        </td>
        <td><p>Translating gene identifiers to gene symbols</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaWBMlouvain.html">alpacaWBMlouvain()</a></code> </p>
        </td>
        <td><p>Generalized Louvain method for bipartite networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacacomputeDWBMmatmscale.html">alpacacomputeDWBMmatmscale()</a></code> </p>
        </td>
        <td><p>Differential modularity matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacacomputeWBMmat.html">alpacacomputeWBMmat()</a></code> </p>
        </td>
        <td><p>Compute modularity matrix for weighted bipartite network</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacagenlouvain.html">alpacagenlouvain()</a></code> </p>
        </td>
        <td><p>Generalized Louvain optimization</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacagotogenes.html">alpacagotogenes()</a></code> </p>
        </td>
        <td><p>Map GO terms to gene symbols</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacalisttogo.html">alpacalisttogo()</a></code> </p>
        </td>
        <td><p>GO term enrichment for a list of gene sets</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacametanetwork.html">alpacametanetwork()</a></code> </p>
        </td>
        <td><p>Create alpacametanetwork for Louvain optimization</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacanodetogene.html">alpacanodetogene()</a></code> </p>
        </td>
        <td><p>Remove tags from gene names</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacasimulateNetwork.html">alpacasimulateNetwork()</a></code> </p>
        </td>
        <td><p>Simulated networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacatidyconfig.html">alpacatidyconfig()</a></code> </p>
        </td>
        <td><p>Renumbering community membership vector</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorCluster.html">condorCluster()</a></code> </p>
        </td>
        <td><p>Main clustering function for condor.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorCoreEnrich.html">condorCoreEnrich()</a></code> </p>
        </td>
        <td><p>Compare qscore distribution of a subset of nodes to all other nodes.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorMatrixModularity.html">condorMatrixModularity()</a></code> </p>
        </td>
        <td><p>Iteratively maximize bipartite modularity.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorModularityMax.html">condorModularityMax()</a></code> </p>
        </td>
        <td><p>Iteratively maximize bipartite modularity.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorPlotCommunities.html">condorPlotCommunities()</a></code> </p>
        </td>
        <td><p>Plot adjacency matrix with links grouped and colored by community</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorPlotHeatmap.html">condorPlotHeatmap()</a></code> </p>
        </td>
        <td><p>Plot weighted adjacency matrix with links grouped by community</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorQscore.html">condorQscore()</a></code> </p>
        </td>
        <td><p>Calculate Qscore for all nodes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="createCondorObject.html">createCondorObject()</a></code> </p>
        </td>
        <td><p>Create list amenable to analysis using <code>condor</code> package.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="createDiffPandaStyle.html">createDiffPandaStyle()</a></code> </p>
        </td>
        <td><p>Create a Cytoscape visual style for two PANDA networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="createPandaStyle.html">createPandaStyle()</a></code> </p>
        </td>
        <td><p>Create a Cytoscape visual style for PANDA network</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="exon.size.html">exon.size</a></code> </p>
        </td>
        <td><p>Gene length</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="genes.html">genes</a></code> </p>
        </td>
        <td><p>Example of a gene list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="lioness.html">lioness()</a></code> </p>
        </td>
        <td><p>Compute LIONESS (Linear Interpolation to Obtain Network Estimates for Single Samples)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="lionessPy.html">lionessPy()</a></code> </p>
        </td>
        <td><p>Run python implementation of LIONESS</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monster.html">monster()</a></code> </p>
        </td>
        <td><p>MOdeling Network State Transitions from Expression and Regulatory data (MONSTER)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterBereFull.html">monsterBereFull()</a></code> </p>
        </td>
        <td><p>Bipartite Edge Reconstruction from Expression data 
(composite method with direct/indirect)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterCalculateTmPvalues.html">monsterCalculateTmPvalues()</a></code> </p>
        </td>
        <td><p>Calculate p-values for a tranformation matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterCheckDataType.html">monsterCheckDataType()</a></code> </p>
        </td>
        <td><p>Checks that data is something MONSTER can handle</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterHclHeatmapPlot.html">monsterHclHeatmapPlot()</a></code> </p>
        </td>
        <td><p>Transformation matrix plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterMonsterNI.html">monsterMonsterNI()</a></code> </p>
        </td>
        <td><p>Bipartite Edge Reconstruction from Expression data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterPlotMonsterAnalysis.html">monsterPlotMonsterAnalysis()</a></code> </p>
        </td>
        <td><p>monsterPlotMonsterAnalysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterPrintMonsterAnalysis.html">monsterPrintMonsterAnalysis()</a></code> </p>
        </td>
        <td><p>monsterPrintMonsterAnalysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterRes.html">monsterRes</a></code> </p>
        </td>
        <td><p>MONSTER results from example cell-cycle yeast transition</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterTransformationMatrix.html">monsterTransformationMatrix()</a></code> </p>
        </td>
        <td><p>Bi-partite network analysis tools</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterTransitionPCAPlot.html">monsterTransitionPCAPlot()</a></code> </p>
        </td>
        <td><p>Principal Components plot of transformation matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterdTFIPlot.html">monsterdTFIPlot()</a></code> </p>
        </td>
        <td><p>This function plots the Off diagonal mass of an 
observed Transition Matrix compared to a set of null TMs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monstertransitionNetworkPlot.html">monstertransitionNetworkPlot()</a></code> </p>
        </td>
        <td><p>This function uses igraph to plot the transition matrix (directed graph) as a network.
The edges in the network should be read as A 'positively/negatively contributes to' the 
targeting of B in the target state.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mut.ucec.html">mut.ucec</a></code> </p>
        </td>
        <td><p>Example of mutation data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="otter.html">otter()</a></code> </p>
        </td>
        <td><p>Run OTTER in R</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pandaDiffEdges.html">pandaDiffEdges()</a></code> </p>
        </td>
        <td><p>Identify differential edges in two PANDA networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pandaPy.html">pandaPy()</a></code> </p>
        </td>
        <td><p>Run Python implementation PANDA in R</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pandaToAlpaca.html">pandaToAlpaca()</a></code> </p>
        </td>
        <td><p>Use two PANDA network to generate an ALPACA result</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pandaToCondorObject.html">pandaToCondorObject()</a></code> </p>
        </td>
        <td><p>Turn PANDA network into a CONDOR object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runEgret.html">runEgret()</a></code> </p>
        </td>
        <td><p>Run EGRET in R</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambar.html">sambar()</a></code> </p>
        </td>
        <td><p>Main SAMBAR function.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambarConvertgmt.html">sambarConvertgmt()</a></code> </p>
        </td>
        <td><p>Convert .gmt files into a binary matrix.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambarCorgenelength.html">sambarCorgenelength()</a></code> </p>
        </td>
        <td><p>Normalize gene mutation scores by gene length.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambarDesparsify.html">sambarDesparsify()</a></code> </p>
        </td>
        <td><p>De-sparsify gene-level mutation scores into gene set-level mutation scores.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="small1976.html">small1976</a></code> </p>
        </td>
        <td><p>Pollinator-plant interactions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sourcePPI.html">sourcePPI()</a></code> </p>
        </td>
        <td><p>Source the Protein-Protein interaction in STRING database</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="visDiffPandaInCytoscape.html">visDiffPandaInCytoscape()</a></code> </p>
        </td>
        <td><p>Plot two PANDA networsk in Cytoscape</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="visPandaInCytoscape.html">visPandaInCytoscape()</a></code> </p>
        </td>
        <td><p>Plot PANDA network in Cytoscape</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="yeast.html">yeast</a></code> </p>
        </td>
        <td><p>Toy data derived from three gene expression datasets and a mapping from transcription factors to genes.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-panda" class="hasAnchor"><a href="#section-panda" class="anchor"></a>PANDA</h2>
          <p class="section-desc"><p>All function of PANDA</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-condor" class="hasAnchor"><a href="#section-condor" class="anchor"></a>CONDOR</h2>
          <p class="section-desc"><p>All fucntions of CONDOR</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="condorCluster.html">condorCluster()</a></code> </p>
        </td>
        <td><p>Main clustering function for condor.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorCoreEnrich.html">condorCoreEnrich()</a></code> </p>
        </td>
        <td><p>Compare qscore distribution of a subset of nodes to all other nodes.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorMatrixModularity.html">condorMatrixModularity()</a></code> </p>
        </td>
        <td><p>Iteratively maximize bipartite modularity.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorModularityMax.html">condorModularityMax()</a></code> </p>
        </td>
        <td><p>Iteratively maximize bipartite modularity.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorPlotCommunities.html">condorPlotCommunities()</a></code> </p>
        </td>
        <td><p>Plot adjacency matrix with links grouped and colored by community</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorPlotHeatmap.html">condorPlotHeatmap()</a></code> </p>
        </td>
        <td><p>Plot weighted adjacency matrix with links grouped by community</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorQscore.html">condorQscore()</a></code> </p>
        </td>
        <td><p>Calculate Qscore for all nodes</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-lioness" class="hasAnchor"><a href="#section-lioness" class="anchor"></a>LIONESS</h2>
          <p class="section-desc"><p>All functions of LIONESS</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="alpaca.html">alpaca()</a></code> </p>
        </td>
        <td><p>Main ALPACA function</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaCommunityStructureRotation.html">alpacaCommunityStructureRotation()</a></code> </p>
        </td>
        <td><p>Comparing node community membership between two networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaDeltaZAnalysis.html">alpacaDeltaZAnalysis()</a></code> </p>
        </td>
        <td><p>Edge subtraction method (CONDOR optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaDeltaZAnalysisLouvain.html">alpacaDeltaZAnalysisLouvain()</a></code> </p>
        </td>
        <td><p>Edge subtraction method (Louvain optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaExtractTopGenes.html">alpacaExtractTopGenes()</a></code> </p>
        </td>
        <td><p>Extract core target genes in differential modules</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaGOtabtogenes.html">alpacaGOtabtogenes()</a></code> </p>
        </td>
        <td><p>The top GO term associated genes in each module</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaRotationAnalysis.html">alpacaRotationAnalysis()</a></code> </p>
        </td>
        <td><p>Community comparison method (CONDOR optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaRotationAnalysisLouvain.html">alpacaRotationAnalysisLouvain()</a></code> </p>
        </td>
        <td><p>Community comparison method (CONDOR optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaTestNodeRank.html">alpacaTestNodeRank()</a></code> </p>
        </td>
        <td><p>Enrichment in ranked list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaTopEnsembltoTopSym.html">alpacaTopEnsembltoTopSym()</a></code> </p>
        </td>
        <td><p>Translating gene identifiers to gene symbols</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaWBMlouvain.html">alpacaWBMlouvain()</a></code> </p>
        </td>
        <td><p>Generalized Louvain method for bipartite networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacacomputeDWBMmatmscale.html">alpacacomputeDWBMmatmscale()</a></code> </p>
        </td>
        <td><p>Differential modularity matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacacomputeWBMmat.html">alpacacomputeWBMmat()</a></code> </p>
        </td>
        <td><p>Compute modularity matrix for weighted bipartite network</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacagenlouvain.html">alpacagenlouvain()</a></code> </p>
        </td>
        <td><p>Generalized Louvain optimization</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacagotogenes.html">alpacagotogenes()</a></code> </p>
        </td>
        <td><p>Map GO terms to gene symbols</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacalisttogo.html">alpacalisttogo()</a></code> </p>
        </td>
        <td><p>GO term enrichment for a list of gene sets</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacametanetwork.html">alpacametanetwork()</a></code> </p>
        </td>
        <td><p>Create alpacametanetwork for Louvain optimization</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacanodetogene.html">alpacanodetogene()</a></code> </p>
        </td>
        <td><p>Remove tags from gene names</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacasimulateNetwork.html">alpacasimulateNetwork()</a></code> </p>
        </td>
        <td><p>Simulated networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacatidyconfig.html">alpacatidyconfig()</a></code> </p>
        </td>
        <td><p>Renumbering community membership vector</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorCluster.html">condorCluster()</a></code> </p>
        </td>
        <td><p>Main clustering function for condor.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorCoreEnrich.html">condorCoreEnrich()</a></code> </p>
        </td>
        <td><p>Compare qscore distribution of a subset of nodes to all other nodes.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorMatrixModularity.html">condorMatrixModularity()</a></code> </p>
        </td>
        <td><p>Iteratively maximize bipartite modularity.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorModularityMax.html">condorModularityMax()</a></code> </p>
        </td>
        <td><p>Iteratively maximize bipartite modularity.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorPlotCommunities.html">condorPlotCommunities()</a></code> </p>
        </td>
        <td><p>Plot adjacency matrix with links grouped and colored by community</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorPlotHeatmap.html">condorPlotHeatmap()</a></code> </p>
        </td>
        <td><p>Plot weighted adjacency matrix with links grouped by community</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="condorQscore.html">condorQscore()</a></code> </p>
        </td>
        <td><p>Calculate Qscore for all nodes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="createCondorObject.html">createCondorObject()</a></code> </p>
        </td>
        <td><p>Create list amenable to analysis using <code>condor</code> package.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="createDiffPandaStyle.html">createDiffPandaStyle()</a></code> </p>
        </td>
        <td><p>Create a Cytoscape visual style for two PANDA networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="createPandaStyle.html">createPandaStyle()</a></code> </p>
        </td>
        <td><p>Create a Cytoscape visual style for PANDA network</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="exon.size.html">exon.size</a></code> </p>
        </td>
        <td><p>Gene length</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="genes.html">genes</a></code> </p>
        </td>
        <td><p>Example of a gene list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="lioness.html">lioness()</a></code> </p>
        </td>
        <td><p>Compute LIONESS (Linear Interpolation to Obtain Network Estimates for Single Samples)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="lionessPy.html">lionessPy()</a></code> </p>
        </td>
        <td><p>Run python implementation of LIONESS</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monster.html">monster()</a></code> </p>
        </td>
        <td><p>MOdeling Network State Transitions from Expression and Regulatory data (MONSTER)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterBereFull.html">monsterBereFull()</a></code> </p>
        </td>
        <td><p>Bipartite Edge Reconstruction from Expression data 
(composite method with direct/indirect)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterCalculateTmPvalues.html">monsterCalculateTmPvalues()</a></code> </p>
        </td>
        <td><p>Calculate p-values for a tranformation matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterCheckDataType.html">monsterCheckDataType()</a></code> </p>
        </td>
        <td><p>Checks that data is something MONSTER can handle</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterHclHeatmapPlot.html">monsterHclHeatmapPlot()</a></code> </p>
        </td>
        <td><p>Transformation matrix plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterMonsterNI.html">monsterMonsterNI()</a></code> </p>
        </td>
        <td><p>Bipartite Edge Reconstruction from Expression data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterPlotMonsterAnalysis.html">monsterPlotMonsterAnalysis()</a></code> </p>
        </td>
        <td><p>monsterPlotMonsterAnalysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterPrintMonsterAnalysis.html">monsterPrintMonsterAnalysis()</a></code> </p>
        </td>
        <td><p>monsterPrintMonsterAnalysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterRes.html">monsterRes</a></code> </p>
        </td>
        <td><p>MONSTER results from example cell-cycle yeast transition</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterTransformationMatrix.html">monsterTransformationMatrix()</a></code> </p>
        </td>
        <td><p>Bi-partite network analysis tools</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterTransitionPCAPlot.html">monsterTransitionPCAPlot()</a></code> </p>
        </td>
        <td><p>Principal Components plot of transformation matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterdTFIPlot.html">monsterdTFIPlot()</a></code> </p>
        </td>
        <td><p>This function plots the Off diagonal mass of an 
observed Transition Matrix compared to a set of null TMs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monstertransitionNetworkPlot.html">monstertransitionNetworkPlot()</a></code> </p>
        </td>
        <td><p>This function uses igraph to plot the transition matrix (directed graph) as a network.
The edges in the network should be read as A 'positively/negatively contributes to' the 
targeting of B in the target state.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mut.ucec.html">mut.ucec</a></code> </p>
        </td>
        <td><p>Example of mutation data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="otter.html">otter()</a></code> </p>
        </td>
        <td><p>Run OTTER in R</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pandaDiffEdges.html">pandaDiffEdges()</a></code> </p>
        </td>
        <td><p>Identify differential edges in two PANDA networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pandaPy.html">pandaPy()</a></code> </p>
        </td>
        <td><p>Run Python implementation PANDA in R</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pandaToAlpaca.html">pandaToAlpaca()</a></code> </p>
        </td>
        <td><p>Use two PANDA network to generate an ALPACA result</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pandaToCondorObject.html">pandaToCondorObject()</a></code> </p>
        </td>
        <td><p>Turn PANDA network into a CONDOR object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runEgret.html">runEgret()</a></code> </p>
        </td>
        <td><p>Run EGRET in R</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambar.html">sambar()</a></code> </p>
        </td>
        <td><p>Main SAMBAR function.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambarConvertgmt.html">sambarConvertgmt()</a></code> </p>
        </td>
        <td><p>Convert .gmt files into a binary matrix.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambarCorgenelength.html">sambarCorgenelength()</a></code> </p>
        </td>
        <td><p>Normalize gene mutation scores by gene length.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambarDesparsify.html">sambarDesparsify()</a></code> </p>
        </td>
        <td><p>De-sparsify gene-level mutation scores into gene set-level mutation scores.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="small1976.html">small1976</a></code> </p>
        </td>
        <td><p>Pollinator-plant interactions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sourcePPI.html">sourcePPI()</a></code> </p>
        </td>
        <td><p>Source the Protein-Protein interaction in STRING database</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="visDiffPandaInCytoscape.html">visDiffPandaInCytoscape()</a></code> </p>
        </td>
        <td><p>Plot two PANDA networsk in Cytoscape</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="visPandaInCytoscape.html">visPandaInCytoscape()</a></code> </p>
        </td>
        <td><p>Plot PANDA network in Cytoscape</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="yeast.html">yeast</a></code> </p>
        </td>
        <td><p>Toy data derived from three gene expression datasets and a mapping from transcription factors to genes.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-alpaca" class="hasAnchor"><a href="#section-alpaca" class="anchor"></a>ALPACA</h2>
          <p class="section-desc"><p>All functions of ALPACA</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="alpaca.html">alpaca()</a></code> </p>
        </td>
        <td><p>Main ALPACA function</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaCommunityStructureRotation.html">alpacaCommunityStructureRotation()</a></code> </p>
        </td>
        <td><p>Comparing node community membership between two networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaDeltaZAnalysis.html">alpacaDeltaZAnalysis()</a></code> </p>
        </td>
        <td><p>Edge subtraction method (CONDOR optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaDeltaZAnalysisLouvain.html">alpacaDeltaZAnalysisLouvain()</a></code> </p>
        </td>
        <td><p>Edge subtraction method (Louvain optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaExtractTopGenes.html">alpacaExtractTopGenes()</a></code> </p>
        </td>
        <td><p>Extract core target genes in differential modules</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaGOtabtogenes.html">alpacaGOtabtogenes()</a></code> </p>
        </td>
        <td><p>The top GO term associated genes in each module</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaRotationAnalysis.html">alpacaRotationAnalysis()</a></code> </p>
        </td>
        <td><p>Community comparison method (CONDOR optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaRotationAnalysisLouvain.html">alpacaRotationAnalysisLouvain()</a></code> </p>
        </td>
        <td><p>Community comparison method (CONDOR optimizaton)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaTestNodeRank.html">alpacaTestNodeRank()</a></code> </p>
        </td>
        <td><p>Enrichment in ranked list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaTopEnsembltoTopSym.html">alpacaTopEnsembltoTopSym()</a></code> </p>
        </td>
        <td><p>Translating gene identifiers to gene symbols</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacaWBMlouvain.html">alpacaWBMlouvain()</a></code> </p>
        </td>
        <td><p>Generalized Louvain method for bipartite networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacacomputeDWBMmatmscale.html">alpacacomputeDWBMmatmscale()</a></code> </p>
        </td>
        <td><p>Differential modularity matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacacomputeWBMmat.html">alpacacomputeWBMmat()</a></code> </p>
        </td>
        <td><p>Compute modularity matrix for weighted bipartite network</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacagenlouvain.html">alpacagenlouvain()</a></code> </p>
        </td>
        <td><p>Generalized Louvain optimization</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacagotogenes.html">alpacagotogenes()</a></code> </p>
        </td>
        <td><p>Map GO terms to gene symbols</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacalisttogo.html">alpacalisttogo()</a></code> </p>
        </td>
        <td><p>GO term enrichment for a list of gene sets</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacametanetwork.html">alpacametanetwork()</a></code> </p>
        </td>
        <td><p>Create alpacametanetwork for Louvain optimization</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacanodetogene.html">alpacanodetogene()</a></code> </p>
        </td>
        <td><p>Remove tags from gene names</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacasimulateNetwork.html">alpacasimulateNetwork()</a></code> </p>
        </td>
        <td><p>Simulated networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alpacatidyconfig.html">alpacatidyconfig()</a></code> </p>
        </td>
        <td><p>Renumbering community membership vector</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-sambar" class="hasAnchor"><a href="#section-sambar" class="anchor"></a>SAMBAR</h2>
          <p class="section-desc"><p>All functions of SAMBAR</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="sambar.html">sambar()</a></code> </p>
        </td>
        <td><p>Main SAMBAR function.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambarConvertgmt.html">sambarConvertgmt()</a></code> </p>
        </td>
        <td><p>Convert .gmt files into a binary matrix.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambarCorgenelength.html">sambarCorgenelength()</a></code> </p>
        </td>
        <td><p>Normalize gene mutation scores by gene length.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sambarDesparsify.html">sambarDesparsify()</a></code> </p>
        </td>
        <td><p>De-sparsify gene-level mutation scores into gene set-level mutation scores.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-monster" class="hasAnchor"><a href="#section-monster" class="anchor"></a>MONSTER</h2>
          <p class="section-desc"><p>All functions of MONSTER</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="monster.html">monster()</a></code> </p>
        </td>
        <td><p>MOdeling Network State Transitions from Expression and Regulatory data (MONSTER)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterBereFull.html">monsterBereFull()</a></code> </p>
        </td>
        <td><p>Bipartite Edge Reconstruction from Expression data 
(composite method with direct/indirect)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterCalculateTmPvalues.html">monsterCalculateTmPvalues()</a></code> </p>
        </td>
        <td><p>Calculate p-values for a tranformation matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterCheckDataType.html">monsterCheckDataType()</a></code> </p>
        </td>
        <td><p>Checks that data is something MONSTER can handle</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterHclHeatmapPlot.html">monsterHclHeatmapPlot()</a></code> </p>
        </td>
        <td><p>Transformation matrix plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterMonsterNI.html">monsterMonsterNI()</a></code> </p>
        </td>
        <td><p>Bipartite Edge Reconstruction from Expression data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterPlotMonsterAnalysis.html">monsterPlotMonsterAnalysis()</a></code> </p>
        </td>
        <td><p>monsterPlotMonsterAnalysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterPrintMonsterAnalysis.html">monsterPrintMonsterAnalysis()</a></code> </p>
        </td>
        <td><p>monsterPrintMonsterAnalysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterRes.html">monsterRes</a></code> </p>
        </td>
        <td><p>MONSTER results from example cell-cycle yeast transition</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterTransformationMatrix.html">monsterTransformationMatrix()</a></code> </p>
        </td>
        <td><p>Bi-partite network analysis tools</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterTransitionPCAPlot.html">monsterTransitionPCAPlot()</a></code> </p>
        </td>
        <td><p>Principal Components plot of transformation matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monsterdTFIPlot.html">monsterdTFIPlot()</a></code> </p>
        </td>
        <td><p>This function plots the Off diagonal mass of an 
observed Transition Matrix compared to a set of null TMs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="monstertransitionNetworkPlot.html">monstertransitionNetworkPlot()</a></code> </p>
        </td>
        <td><p>This function uses igraph to plot the transition matrix (directed graph) as a network.
The edges in the network should be read as A 'positively/negatively contributes to' the 
targeting of B in the target state.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-otter" class="hasAnchor"><a href="#section-otter" class="anchor"></a>OTTER</h2>
          <p class="section-desc"><p>All functions of OTTER</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="otter.html">otter()</a></code> </p>
        </td>
        <td><p>Run OTTER in R</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Tian Wang, Marouen Ben Guebila, John Platig, Marieke Kuijjer, Magha Padi, Rebekka Burkholz, Deborah Weighill.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


