<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>MOdeling Network State Transitions from Expression and Regulatory data (MONSTER) — monster • netZooR</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/all.min.css" integrity="sha256-PbSmjxuVAzJ6FPvNYsrXygfGhNJYyZ2GktDbkMBqQZg=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/v4-shims.min.css" integrity="sha256-A6jcAdwFD48VMjlI3GDxUd+eCQa7/KWy6G9oe/ovaPA=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="MOdeling Network State Transitions from Expression and Regulatory data (MONSTER) — monster" />

<meta property="og:description" content="This function runs the MONSTER algorithm.  Biological states are characterized by distinct patterns 
of gene expression that reflect each phenotype's active cellular processes. 
Driving these phenotypes are gene regulatory networks in which transcriptions factors control 
when and to what degree individual genes are expressed. Phenotypic transitions, such as those that 
occur when disease arises from healthy tissue, are associated with changes in these  networks. 
MONSTER is an approach to understanding these transitions. MONSTER models phenotypic-specific 
regulatory networks and then estimates a &quot;transition matrix&quot; that converts one state to another. 
By examining the properties of the transition matrix, we can gain insight into regulatory 
changes associated with phenotypic state transition." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.7.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ALPACA_MaghaP.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="../articles/ApplicationinGTExData_DeborahW.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/ApplicationwithTBdataset_TianW.html">Application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/MONSTER_DanielS.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="../articles/SAMBAR_MariekeK.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="../articles/lioness_limma_CamilaLR.html">Comparing LIONESS Regulatory Networks using limma</a>
    </li>
    <li>
      <a href="../articles/pandaR_DanielS.html">An Introduction to the pandaR Package</a>
    </li>
    <li>
      <a href="../articles/panda_CamilaLR.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>MOdeling Network State Transitions from Expression and Regulatory data (MONSTER)</h1>
    
    <div class="hidden name"><code>monster.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function runs the MONSTER algorithm.  Biological states are characterized by distinct patterns 
of gene expression that reflect each phenotype's active cellular processes. 
Driving these phenotypes are gene regulatory networks in which transcriptions factors control 
when and to what degree individual genes are expressed. Phenotypic transitions, such as those that 
occur when disease arises from healthy tissue, are associated with changes in these  networks. 
MONSTER is an approach to understanding these transitions. MONSTER models phenotypic-specific 
regulatory networks and then estimates a "transition matrix" that converts one state to another. 
By examining the properties of the transition matrix, we can gain insight into regulatory 
changes associated with phenotypic state transition.</p>
    
    </div>

    <pre class="usage"><span class='fu'>monster</span>(<span class='no'>expr</span>, <span class='no'>design</span>, <span class='no'>motif</span>, <span class='kw'>nullPerms</span> <span class='kw'>=</span> <span class='fl'>100</span>, <span class='kw'>ni_method</span> <span class='kw'>=</span> <span class='st'>"bere"</span>,
  <span class='kw'>ni.coefficient.cutoff</span> <span class='kw'>=</span> <span class='fl'>NA</span>, <span class='kw'>numMaxCores</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>outputDir</span> <span class='kw'>=</span> <span class='fl'>NA</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>expr</th>
      <td><p>Gene Expression dataset, can be matrix or data.frame of expression values or ExpressionSet</p></td>
    </tr>
    <tr>
      <th>design</th>
      <td><p>Binary vector indicating case control partition</p></td>
    </tr>
    <tr>
      <th>motif</th>
      <td><p>Regulatory data.frame consisting of three columns.  For each row, a transcription factor (column 1) 
regulates a gene (column 2) with a defined strength (column 3), usually taken to be 0 or 1</p></td>
    </tr>
    <tr>
      <th>nullPerms</th>
      <td><p>number of random permutations to run (default 100).  Set to 0 to only 
calculate observed transition matrix</p></td>
    </tr>
    <tr>
      <th>ni_method</th>
      <td><p>String to indicate algorithm method.  Must be one of "bere","pearson","cd","lda", or "wcd". Default is "bere"</p></td>
    </tr>
    <tr>
      <th>ni.coefficient.cutoff</th>
      <td><p>numeric to specify a p-value cutoff at the network
inference step.  Default is NA, indicating inclusion of all coefficients.</p></td>
    </tr>
    <tr>
      <th>numMaxCores</th>
      <td><p>requires doParallel, foreach.  Runs MONSTER in parallel computing 
environment.  Set to 1 to avoid parallelization.</p></td>
    </tr>
    <tr>
      <th>outputDir</th>
      <td><p>character vector specifying a directory or path in which 
which to save MONSTER results, default is NA and results are not saved.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of class "monsterAnalysis" containing results</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># \donttest{</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>yeast</span>)
<span class='no'>design</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span>(<span class='fl'>0</span>,<span class='fl'>20</span>),<span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span>(<span class='fl'>NA</span>,<span class='fl'>10</span>),<span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span>(<span class='fl'>1</span>,<span class='fl'>20</span>))
<span class='no'>monsterRes</span> <span class='kw'>&lt;-</span> <span class='fu'>monster</span>(<span class='no'>yeast</span>$<span class='no'>exp.cc</span>[<span class='fl'>1</span>:<span class='fl'>500</span>,], <span class='no'>design</span>, <span class='no'>yeast</span>$<span class='no'>motif</span>, <span class='kw'>nullPerms</span><span class='kw'>=</span><span class='fl'>10</span>, <span class='kw'>numMaxCores</span><span class='kw'>=</span><span class='fl'>4</span>)</div><div class='output co'>#&gt; [1] "Running null permutations in parallel"
#&gt; [1] "4 cores used"
#&gt; [1] "11 network transitions to be estimated"
#&gt; Time difference of 10.7082 secs</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/graphics/plot.html'>plot</a></span>(<span class='no'>monsterRes</span>)</div><div class='output co'>#&gt; <span class='error'>Error in as.double(y): cannot coerce type 'S4' to vector of type 'double'</span></div><div class='input'># }
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Tian Wang, Marouen Ben Guebila, John Platig, Marieke Kuijjer, Magha Padi, Rebekka Burkholz.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


