<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function runs the MONSTER algorithm.  Biological states are characterized by distinct patterns 
of gene expression that reflect each phenotype's active cellular processes. 
Driving these phenotypes are gene regulatory networks in which transcriptions factors control 
when and to what degree individual genes are expressed. Phenotypic transitions, such as those that 
occur when disease arises from healthy tissue, are associated with changes in these  networks. 
MONSTER is an approach to understanding these transitions. MONSTER models phenotypic-specific 
regulatory networks and then estimates a &quot;transition matrix&quot; that converts one state to another. 
By examining the properties of the transition matrix, we can gain insight into regulatory 
changes associated with phenotypic state transition.
Important note: the direct regulatory network observed from gene expression is currently
implemented as a regular correlation as opposed to the partial correlation described 
in the paper.
Citation: Schlauch, Daniel, et al. &quot;Estimating drivers of cell state transitions using gene regulatory network models.&quot; 
BMC systems biology 11.1 (2017): 139. https://doi.org/10.1186/s12918-017-0517-y"><title>MOdeling Network State Transitions from Expression and Regulatory data (MONSTER) — monster • netZooR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="MOdeling Network State Transitions from Expression and Regulatory data (MONSTER) — monster"><meta property="og:description" content="This function runs the MONSTER algorithm.  Biological states are characterized by distinct patterns 
of gene expression that reflect each phenotype's active cellular processes. 
Driving these phenotypes are gene regulatory networks in which transcriptions factors control 
when and to what degree individual genes are expressed. Phenotypic transitions, such as those that 
occur when disease arises from healthy tissue, are associated with changes in these  networks. 
MONSTER is an approach to understanding these transitions. MONSTER models phenotypic-specific 
regulatory networks and then estimates a &quot;transition matrix&quot; that converts one state to another. 
By examining the properties of the transition matrix, we can gain insight into regulatory 
changes associated with phenotypic state transition.
Important note: the direct regulatory network observed from gene expression is currently
implemented as a regular correlation as opposed to the partial correlation described 
in the paper.
Citation: Schlauch, Daniel, et al. &quot;Estimating drivers of cell state transitions using gene regulatory network models.&quot; 
BMC systems biology 11.1 (2017): 139. https://doi.org/10.1186/s12918-017-0517-y"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">netZooR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.17</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    <a class="dropdown-item" href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    <a class="dropdown-item" href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    <a class="dropdown-item" href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    <a class="dropdown-item" href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    <a class="dropdown-item" href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    <a class="dropdown-item" href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    <a class="dropdown-item" href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    <a class="dropdown-item" href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    <a class="dropdown-item" href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/netZoo/netZooR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>MOdeling Network State Transitions from Expression and Regulatory data (MONSTER)</h1>
      <small class="dont-index">Source: <a href="https://github.com/netZoo/netZooR/blob/HEAD/R/MONSTER.R" class="external-link"><code>R/MONSTER.R</code></a></small>
      <div class="d-none name"><code>monster.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function runs the MONSTER algorithm.  Biological states are characterized by distinct patterns 
of gene expression that reflect each phenotype's active cellular processes. 
Driving these phenotypes are gene regulatory networks in which transcriptions factors control 
when and to what degree individual genes are expressed. Phenotypic transitions, such as those that 
occur when disease arises from healthy tissue, are associated with changes in these  networks. 
MONSTER is an approach to understanding these transitions. MONSTER models phenotypic-specific 
regulatory networks and then estimates a "transition matrix" that converts one state to another. 
By examining the properties of the transition matrix, we can gain insight into regulatory 
changes associated with phenotypic state transition.
Important note: the direct regulatory network observed from gene expression is currently
implemented as a regular correlation as opposed to the partial correlation described 
in the paper.
Citation: Schlauch, Daniel, et al. "Estimating drivers of cell state transitions using gene regulatory network models." 
BMC systems biology 11.1 (2017): 139. https://doi.org/10.1186/s12918-017-0517-y</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">monster</span><span class="op">(</span></span>
<span>  <span class="va">expr</span>,</span>
<span>  <span class="va">design</span>,</span>
<span>  <span class="va">motif</span>,</span>
<span>  nullPerms <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  ni_method <span class="op">=</span> <span class="st">"BERE"</span>,</span>
<span>  ni.coefficient.cutoff <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  numMaxCores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  outputDir <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  alphaw <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  mode <span class="op">=</span> <span class="st">"buildNet"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>expr</dt>
<dd><p>Gene Expression dataset, can be matrix or data.frame of expression values or ExpressionSet.</p></dd>


<dt>design</dt>
<dd><p>Binary vector indicating case control partition. 1 for case and 0 for control.</p></dd>


<dt>motif</dt>
<dd><p>Regulatory data.frame consisting of three columns.  For each row, a transcription factor (column 1) 
regulates a gene (column 2) with a defined strength (column 3), usually taken to be 0 or 1</p></dd>


<dt>nullPerms</dt>
<dd><p>number of random permutations to run (default 100).  Set to 0 to only 
calculate observed transition matrix. When mode is is 'buildNet' it randomly permutes the case and control expression
samples, if mode is 'regNet' it will randomly permute the case and control networks.</p></dd>


<dt>ni_method</dt>
<dd><p>String to indicate algorithm method.  Must be one of "bere","pearson","cd","lda", or "wcd". Default is "bere"</p></dd>


<dt>ni.coefficient.cutoff</dt>
<dd><p>numeric to specify a p-value cutoff at the network
inference step.  Default is NA, indicating inclusion of all coefficients.</p></dd>


<dt>numMaxCores</dt>
<dd><p>requires doParallel, foreach.  Runs MONSTER in parallel computing 
environment.  Set to 1 to avoid parallelization, NA will take the default parallel pool in the computer.</p></dd>


<dt>outputDir</dt>
<dd><p>character vector specifying a directory or path in which 
which to save MONSTER results, default is NA and results are not saved.</p></dd>


<dt>alphaw</dt>
<dd><p>A weight parameter between 0 and 1 specifying proportion of weight 
to give to indirect compared to direct evidence. The default is 0.5 to give an 
equal weight to direct and indirect evidence.</p></dd>


<dt>mode</dt>
<dd><p>A parameter telling whether to build the regulatory networks ('buildNet') or to use provided regulatory networks
('regNet'). If set to 'regNet', then the parameters motif, ni_method, ni.coefficient.cutoff, and alphaw will be set to NA.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>An object of class "monsterAnalysis" containing results</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="co"># Example with the network reconstruction step</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">yeast</span><span class="op">)</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">20</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="cn">NA</span>,<span class="fl">10</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">20</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">yeast</span><span class="op">$</span><span class="va">exp.cc</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">yeast</span><span class="op">$</span><span class="va">exp.cc</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">yeast</span><span class="op">$</span><span class="va">exp.cc</span><span class="op">)</span>,na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#monsterRes &lt;- monster(yeast$exp.cc[1:500,], design, yeast$motif, nullPerms=10, numMaxCores=1)</span></span>
<span><span class="co"># Example with provided networks</span></span>
<span><span class="co"># \donttest{</span></span>
<span><span class="va">pandaResult</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pandaR/man/panda.html" class="external-link">panda</a></span><span class="op">(</span><span class="va">pandaToyData</span><span class="op">$</span><span class="va">motif</span>, <span class="va">pandaToyData</span><span class="op">$</span><span class="va">expression</span>, <span class="va">pandaToyData</span><span class="op">$</span><span class="va">ppi</span><span class="op">)</span></span>
<span><span class="va">case</span><span class="op">=</span><span class="fu">getRegNet</span><span class="op">(</span><span class="va">pandaResult</span><span class="op">)</span></span>
<span><span class="va">nelemReg</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu">getRegNet</span><span class="op">(</span><span class="va">pandaResult</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu">getRegNet</span><span class="op">(</span><span class="va">pandaResult</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">nGenes</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu">getRegNet</span><span class="op">(</span><span class="va">pandaResult</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">control</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Exponential.html" class="external-link">rexp</a></span><span class="op">(</span><span class="va">nelemReg</span>, rate<span class="op">=</span><span class="fl">.1</span><span class="op">)</span>, ncol<span class="op">=</span><span class="va">nGenes</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">control</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">case</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">control</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">case</span><span class="op">)</span> </span>
<span><span class="va">expr</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">control</span>,<span class="va">case</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">design</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="va">nGenes</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">nGenes</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">monsterRes</span> <span class="op">&lt;-</span> <span class="fu">monster</span><span class="op">(</span><span class="va">expr</span>, <span class="va">design</span>, motif<span class="op">=</span><span class="cn">NA</span>, nullPerms<span class="op">=</span><span class="fl">10</span>, numMaxCores<span class="op">=</span><span class="fl">1</span>, mode<span class="op">=</span><span class="st">'regNet'</span><span class="op">)</span></span>
<span><span class="co"># }</span></span></code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill, Kate Shutta.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

