<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Bipartite Edge Reconstruction from Expression data 
(composite method with direct/indirect) — monsterBereFull • netZooR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Bipartite Edge Reconstruction from Expression data 
(composite method with direct/indirect) — monsterBereFull" />
<meta property="og:description" content="This function generates a complete bipartite network from 
gene expression data and sequence motif data. This NI method
serves as a default method for inferring bipartite networks
in MONSTER.  Running monsterBereFull can generate these networks
independently from the larger MONSTER method." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.81</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    </li>
    <li>
      <a href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    </li>
    <li>
      <a href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    </li>
    <li>
      <a href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    </li>
    <li>
      <a href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    </li>
    <li>
      <a href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/netZoo/netZooR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Bipartite Edge Reconstruction from Expression data 
(composite method with direct/indirect)</h1>
    <small class="dont-index">Source: <a href='https://github.com/netZoo/netZooR/blob/master/R/MONSTER.R'><code>R/MONSTER.R</code></a></small>
    <div class="hidden name"><code>monsterBereFull.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function generates a complete bipartite network from 
gene expression data and sequence motif data. This NI method
serves as a default method for inferring bipartite networks
in MONSTER.  Running monsterBereFull can generate these networks
independently from the larger MONSTER method.</p>
    </div>

    <pre class="usage"><span class='fu'>monsterBereFull</span><span class='op'>(</span>
  <span class='va'>motif.data</span>,
  <span class='va'>expr.data</span>,
  alpha <span class='op'>=</span> <span class='fl'>0.5</span>,
  lambda <span class='op'>=</span> <span class='fl'>10</span>,
  score <span class='op'>=</span> <span class='st'>"motifincluded"</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>motif.data</th>
      <td><p>A motif dataset, a data.frame, matrix or exprSet 
containing 3 columns. Each row describes an motif associated 
with a transcription factor (column 1) a gene (column 2) 
and a score (column 3) for the motif.</p></td>
    </tr>
    <tr>
      <th>expr.data</th>
      <td><p>An expression dataset, as a genes (rows) by 
samples (columns) data.frame</p></td>
    </tr>
    <tr>
      <th>alpha</th>
      <td><p>A weight parameter specifying proportion of weight 
to give to indirect compared to direct evidence.  See documentation.</p></td>
    </tr>
    <tr>
      <th>lambda</th>
      <td><p>if using penalized, the lambda parameter in the penalized logistic regression</p></td>
    </tr>
    <tr>
      <th>score</th>
      <td><p>String to indicate whether motif information will 
be readded upon completion of the algorithm</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An matrix or data.frame</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>yeast</span><span class='op'>)</span>
<span class='va'>monsterRes</span> <span class='op'>&lt;-</span> <span class='fu'>monsterBereFull</span><span class='op'>(</span><span class='va'>yeast</span><span class='op'>$</span><span class='va'>motif</span>, <span class='va'>yeast</span><span class='op'>$</span><span class='va'>exp.cc</span>, alpha<span class='op'>=</span><span class='fl'>.5</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Using V3 as value column: use value.var to override.</span></div><div class='output co'>#&gt; .123456
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .123456
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .123456
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .123456
#&gt; .1234
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .1234
#&gt; .12345
#&gt; .123456
#&gt; .12345
#&gt; .12345
#&gt; .1234
#&gt; .1234
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .123456
#&gt; .12345
#&gt; .12345
#&gt; .123456
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .1234
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .12345
#&gt; .1234
#&gt; .12345
#&gt; .12345</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Tian Wang, Marouen Ben Guebila, John Platig, Marieke Kuijjer, Magha Padi, Rebekka Burkholz, Deborah Weighill.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


