<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='PANDA(Passing Attributes between Networks for Data Assimilation) is a message-passing model 
to reconstruct gene regulatory network, which integrates multiple sources of biological data-including protein-protein interaction data,
gene expression data, and transcription factor binding motifs data to reconstruct genome-wide, condition-specific regulatory networks.
[(Glass et al. 2013)])
This function is designed to run the a derived PANDA implementation in Python Library "netZooPy" netZooPy.'><title>Run Python implementation PANDA in R — pandaPy • netZooR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Run Python implementation PANDA in R — pandaPy"><meta property="og:description" content='PANDA(Passing Attributes between Networks for Data Assimilation) is a message-passing model 
to reconstruct gene regulatory network, which integrates multiple sources of biological data-including protein-protein interaction data,
gene expression data, and transcription factor binding motifs data to reconstruct genome-wide, condition-specific regulatory networks.
[(Glass et al. 2013)])
This function is designed to run the a derived PANDA implementation in Python Library "netZooPy" netZooPy.'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">netZooR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    <a class="dropdown-item" href="../articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    <a class="dropdown-item" href="../articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    <a class="dropdown-item" href="../articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    <a class="dropdown-item" href="../articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    <a class="dropdown-item" href="../articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    <a class="dropdown-item" href="../articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    <a class="dropdown-item" href="../articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    <a class="dropdown-item" href="../articles/pandaR.html">An Introduction to the pandaR Package</a>
    <a class="dropdown-item" href="../articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    <a class="dropdown-item" href="../articles/yarn.html">YARN: Robust Multi-Tissue RNA-Seq Preprocessing and Normalization</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/netZoo/netZooR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Run Python implementation PANDA in R</h1>
      <small class="dont-index">Source: <a href="https://github.com/netZoo/netZooR/blob/HEAD/R/PANDA.R" class="external-link"><code>R/PANDA.R</code></a></small>
      <div class="d-none name"><code>pandaPy.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><strong>PANDA</strong>(Passing Attributes between Networks for Data Assimilation) is a message-passing model 
to reconstruct gene regulatory network, which integrates multiple sources of biological data-including protein-protein interaction data,
gene expression data, and transcription factor binding motifs data to reconstruct genome-wide, condition-specific regulatory networks.
<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0064832" class="external-link">[(Glass et al. 2013)])</a>
This function is designed to run the a derived PANDA implementation in Python Library "netZooPy" <a href="https://github.com/netZoo/netZooPy" class="external-link">netZooPy</a>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">pandaPy</span><span class="op">(</span></span>
<span>  <span class="va">expr_file</span>,</span>
<span>  motif_file <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ppi_file <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  computing <span class="op">=</span> <span class="st">"cpu"</span>,</span>
<span>  precision <span class="op">=</span> <span class="st">"double"</span>,</span>
<span>  save_memory <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  save_tmp <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  keep_expression_matrix <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  modeProcess <span class="op">=</span> <span class="st">"union"</span>,</span>
<span>  remove_missing <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  with_header <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>expr_file</dt>
<dd><p>Character string indicating the file path of expression values file, with each gene(in rows) across samples(in columns).</p></dd>


<dt>motif_file</dt>
<dd><p>An optional character string indicating the file path of a prior transcription factor binding motifs dataset.
When this argument is not provided, analysis will continue with Pearson correlation matrix.</p></dd>


<dt>ppi_file</dt>
<dd><p>An optional character string indicating the file path of protein-protein interaction edge dataset.
Also, this can be generated with a list of proteins of interest by <code><a href="sourcePPI.html">sourcePPI</a></code>.</p></dd>


<dt>computing</dt>
<dd><p>'cpu' uses Central Processing Unit (CPU) to run PANDA; 'gpu' use the Graphical Processing Unit (GPU) to run PANDA. The default value is "cpu".</p></dd>


<dt>precision</dt>
<dd><p>'double' computes the regulatory network in double precision (15 decimal digits); 'single' computes the regulatory network in single precision (7 decimal digits) which is fastaer, requires half the memory but less accurate. The default value is 'double'.</p></dd>


<dt>save_memory</dt>
<dd><p>'TRUE' removes temporary results from memory. The result network is weighted adjacency matrix of size (nTFs, nGenes); 'FALSE' keeps the temporary files in memory. The result network has 4 columns in the form gene - TF - weight in motif prior - PANDA edge. PANDA indegree/outdegree of panda network, only if save_memory = FALSE. The default value is 'FALSE'.</p></dd>


<dt>save_tmp</dt>
<dd><p>'TRUE' saves middle data like expression matrix and normalized networks; 'FALSE' deletes the middle data. The default value is 'TURE'.</p></dd>


<dt>keep_expression_matrix</dt>
<dd><p>'TRUE' keeps the input expression matrix as an attribute in the result Panda object.'FALSE' deletes the expression matrix attribute in the Panda object. The default value is 'FALSE'.</p></dd>


<dt>modeProcess</dt>
<dd><p>'legacy' refers to the processing mode in netZooPy&lt;=0.5, 'union': takes the union of all TFs and genes across priors and fills the missing genes in the priors with zeros; 'intersection': intersects the input genes and TFs across priors and removes the missing TFs/genes. Default values is 'union'.</p></dd>


<dt>remove_missing</dt>
<dd><p>Only when modeProcess='legacy': remove_missing='TRUE' removes all unmatched TF and genes; remove_missing='FALSE' keeps all tf and genes. The default value is 'FALSE'.</p></dd>


<dt>with_header</dt>
<dd><p>Boolean to read gene expression file with a header for sample names</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>When save_memory=FALSE(default), this function will return a list of three items: 
         Use <code>$panda</code> to access the standard output of PANDA as data frame, which consists of four columns: 
         "TF", "Gene", "Motif" using 0 or 1 to indicate if this edge belongs to prior motif dataset, and "Score".</p>


<p>Use <code>$indegree</code> to access the indegree of PANDA network as data frame, which consists of two columns: "Gene", "Score".</p>


<p>Use <code>$outdegree</code> to access the outdegree of PANDA network as data frame, which consists of two columns: "TF", "Score".</p>
         

<p>When save_memory=TRUE, this function will return a weigheted adjacency matirx of size (nTFs, nGenes), use <code>$WAMpanda</code> to access.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="co"># take the treated TB dataset as example here.</span></span>
<span><span class="co"># refer to the datasets files path in inst/extdat</span></span>
<span></span>
<span><span class="va">treated_expression_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"expr4_matched.txt"</span>, </span>
<span>package <span class="op">=</span> <span class="st">"netZooR"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">treated_expression_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"expr4_matched.txt"</span>,</span>
<span> package <span class="op">=</span> <span class="st">"netZooR"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">motif_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"chip_matched.txt"</span>, package <span class="op">=</span> <span class="st">"netZooR"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">ppi_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ppi_matched.txt"</span>, package <span class="op">=</span> <span class="st">"netZooR"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Run PANDA for treated and control network</span></span>
<span><span class="co"># \donttest{</span></span>
<span><span class="va">treated_all_panda_result</span> <span class="op">&lt;-</span> <span class="fu">pandaPy</span><span class="op">(</span>expr_file <span class="op">=</span> <span class="va">treated_expression_file_path</span>, </span>
<span>motif_file <span class="op">=</span> <span class="va">motif_file_path</span>, ppi_file <span class="op">=</span> <span class="va">ppi_file_path</span>, </span>
<span>modeProcess<span class="op">=</span><span class="st">"legacy"</span>, remove_missing <span class="op">=</span> <span class="cn">TRUE</span> <span class="op">)</span></span>
<span></span>
<span><span class="co"># access PANDA regulatory network</span></span>
<span><span class="va">treated_net</span> <span class="op">&lt;-</span> <span class="va">treated_all_panda_result</span><span class="op">$</span><span class="va">panda</span></span>
<span></span>
<span><span class="co"># access PANDA regulatory indegree network.</span></span>
<span><span class="va">indegree_net</span> <span class="op">&lt;-</span> <span class="va">treated_all_panda_result</span><span class="op">$</span><span class="va">indegree</span></span>
<span></span>
<span><span class="co"># access PANDA regulatory outdegree networks</span></span>
<span><span class="va">outdegree_net</span> <span class="op">&lt;-</span> <span class="va">treated_all_panda_result</span><span class="op">$</span><span class="va">outdegree</span></span>
<span><span class="co"># }</span></span>
<span></span></code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Marouen Ben Guebila, Tian Wang, John Platig, Marieke Kuijjer, Megha Padi, Rebekka Burkholz, Des Weighill, Kate Shutta.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

